{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "nRJwvbotYSr8"
      },
      "source": [
        "# Семинар №11 - Вопросно-ответная система"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "7M760SwoYSr9"
      },
      "source": [
        "Сделаем поисковую систему на основе BM25 и BERT, чтобы отфильтровать научные работы по запросу."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "I-7ISvd9YSr9"
      },
      "source": [
        "## Bert for QA\n",
        "![bert.png](./imgs/bert.png)\n",
        "\n",
        "BERT (Bidirectional Encoder Representations from Transformers) - недавняя статья, опубликованная исследователями Google AI Language. Она вызвала ажиотаж в сообществе машинного обучения, представив самые современные результаты в широком спектре задач NLP, включая ответы на вопросы (SQuAD v1.1), вывод на естественном языке (MNLI) и другие.\n",
        "\n",
        "Ключевым техническим новшеством БЕРТА является применение двунаправленного обучения Transformer, популярной модели внимания, к языковому моделированию. Это контрастирует с предыдущими попытками, в которых рассматривалась последовательность текста либо слева направо, либо комбинированное обучение слева направо и справа налево. Результаты работы показывают, что языковая модель, которая обучается в двух направлениях, может иметь более глубокое представление о языковом контексте и потоке, чем однонаправленные языковые модели. В статье исследователи подробно описывают новую технику под названием Masked LM (MLM), которая позволяет проводить двунаправленное обучение в моделях, в которых ранее это было невозможно.\n",
        "\n",
        "Question answering (QA) - дисциплина компьютерных наук в области поиска информации и обработки естественного языка (NLP), которая занимается созданием систем, автоматически отвечающих на вопросы, задаваемые людьми на естественном языке.\n",
        "\n",
        "## BM25\n",
        "![image.png](./imgs/bm25_formula.jpg)\n",
        "\n",
        "Здесь:\n",
        "- D - документ.\n",
        "- Q - запрос.\n",
        "- f(q_i, D) - число вхождений слова q_i в документ.\n",
        "- |D| - длина документа.\n",
        "- avgdl - средняя длина документов в коллекции.\n",
        "- N - число документов в коллекции.\n",
        "- n(q_i) - число документов, содержащих слово q_i.\n",
        "- k_1 - параметр в диапазоне [1.2, 2.0].\n",
        "- b - параметр, обычно 0.75.\n",
        "- $\\delta$ - параметр, обычно 1.\n",
        "\n",
        "В области поиска информации Okapi BM25 (BM - сокращение от best matching) - это функция ранжирования, используемая поисковыми системами для оценки релевантности документов заданному поисковому запросу. Он основан на системе вероятностного поиска, разработанной в 1970-х и 1980-х годах Стивеном Э. Робертсоном, Карен Сперк Джонс и другими.\n",
        "\n",
        "Название фактической функции ранжирования - BM25. Более полное название, Okapi BM25, включает название первой системы, которая его использовала, - информационно-поисковой системы Okapi, внедренной в Лондонском городском университете в 1980-х и 1990-х годах. BM25 и его более новые варианты, например BM25F (версия BM25, которая может учитывать структуру документа и текст привязки), представляют собой современные функции поиска, подобные TF-IDF, используемые при поиске документов.\n",
        "\n",
        "Почитать про BM25: https://habr.com/ru/articles/823568/"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "nyuTCRYJYSr-"
      },
      "source": [
        "## Install and import packages"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "collapsed": true,
        "jupyter": {
          "outputs_hidden": true
        },
        "scrolled": true,
        "tags": [],
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "h0zQdyRlYSr-",
        "outputId": "afee9f0a-2b6a-4d74-c350-758439f5d606"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting rank_bm25\n",
            "  Downloading rank_bm25-0.2.2-py3-none-any.whl.metadata (3.2 kB)\n",
            "Requirement already satisfied: nltk in /usr/local/lib/python3.11/dist-packages (3.9.1)\n",
            "Requirement already satisfied: transformers in /usr/local/lib/python3.11/dist-packages (4.48.3)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from rank_bm25) (1.26.4)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.11/dist-packages (from nltk) (8.1.8)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.11/dist-packages (from nltk) (1.4.2)\n",
            "Requirement already satisfied: regex>=2021.8.3 in /usr/local/lib/python3.11/dist-packages (from nltk) (2024.11.6)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.11/dist-packages (from nltk) (4.67.1)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.11/dist-packages (from transformers) (3.17.0)\n",
            "Requirement already satisfied: huggingface-hub<1.0,>=0.24.0 in /usr/local/lib/python3.11/dist-packages (from transformers) (0.28.1)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.11/dist-packages (from transformers) (24.2)\n",
            "Requirement already satisfied: pyyaml>=5.1 in /usr/local/lib/python3.11/dist-packages (from transformers) (6.0.2)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.11/dist-packages (from transformers) (2.32.3)\n",
            "Requirement already satisfied: tokenizers<0.22,>=0.21 in /usr/local/lib/python3.11/dist-packages (from transformers) (0.21.0)\n",
            "Requirement already satisfied: safetensors>=0.4.1 in /usr/local/lib/python3.11/dist-packages (from transformers) (0.5.3)\n",
            "Requirement already satisfied: fsspec>=2023.5.0 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub<1.0,>=0.24.0->transformers) (2024.10.0)\n",
            "Requirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.11/dist-packages (from huggingface-hub<1.0,>=0.24.0->transformers) (4.12.2)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.11/dist-packages (from requests->transformers) (3.4.1)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests->transformers) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.11/dist-packages (from requests->transformers) (2.3.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests->transformers) (2025.1.31)\n",
            "Downloading rank_bm25-0.2.2-py3-none-any.whl (8.6 kB)\n",
            "Installing collected packages: rank_bm25\n",
            "Successfully installed rank_bm25-0.2.2\n",
            "Requirement already satisfied: ipywidgets in /usr/local/lib/python3.11/dist-packages (7.7.1)\n",
            "Requirement already satisfied: ipykernel>=4.5.1 in /usr/local/lib/python3.11/dist-packages (from ipywidgets) (6.17.1)\n",
            "Requirement already satisfied: ipython-genutils~=0.2.0 in /usr/local/lib/python3.11/dist-packages (from ipywidgets) (0.2.0)\n",
            "Requirement already satisfied: traitlets>=4.3.1 in /usr/local/lib/python3.11/dist-packages (from ipywidgets) (5.7.1)\n",
            "Requirement already satisfied: widgetsnbextension~=3.6.0 in /usr/local/lib/python3.11/dist-packages (from ipywidgets) (3.6.10)\n",
            "Requirement already satisfied: ipython>=4.0.0 in /usr/local/lib/python3.11/dist-packages (from ipywidgets) (7.34.0)\n",
            "Requirement already satisfied: jupyterlab-widgets>=1.0.0 in /usr/local/lib/python3.11/dist-packages (from ipywidgets) (3.0.13)\n",
            "Requirement already satisfied: debugpy>=1.0 in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (1.8.0)\n",
            "Requirement already satisfied: jupyter-client>=6.1.12 in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (6.1.12)\n",
            "Requirement already satisfied: matplotlib-inline>=0.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (0.1.7)\n",
            "Requirement already satisfied: nest-asyncio in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (1.6.0)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (24.2)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (5.9.5)\n",
            "Requirement already satisfied: pyzmq>=17 in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (24.0.1)\n",
            "Requirement already satisfied: tornado>=6.1 in /usr/local/lib/python3.11/dist-packages (from ipykernel>=4.5.1->ipywidgets) (6.4.2)\n",
            "Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.11/dist-packages (from ipython>=4.0.0->ipywidgets) (75.1.0)\n",
            "Collecting jedi>=0.16 (from ipython>=4.0.0->ipywidgets)\n",
            "  Downloading jedi-0.19.2-py2.py3-none-any.whl.metadata (22 kB)\n",
            "Requirement already satisfied: decorator in /usr/local/lib/python3.11/dist-packages (from ipython>=4.0.0->ipywidgets) (4.4.2)\n",
            "Requirement already satisfied: pickleshare in /usr/local/lib/python3.11/dist-packages (from ipython>=4.0.0->ipywidgets) (0.7.5)\n",
            "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in /usr/local/lib/python3.11/dist-packages (from ipython>=4.0.0->ipywidgets) (3.0.50)\n",
            "Requirement already satisfied: pygments in /usr/local/lib/python3.11/dist-packages (from ipython>=4.0.0->ipywidgets) (2.18.0)\n",
            "Requirement already satisfied: backcall in /usr/local/lib/python3.11/dist-packages (from ipython>=4.0.0->ipywidgets) (0.2.0)\n",
            "Requirement already satisfied: pexpect>4.3 in /usr/local/lib/python3.11/dist-packages (from ipython>=4.0.0->ipywidgets) (4.9.0)\n",
            "Requirement already satisfied: notebook>=4.4.1 in /usr/local/lib/python3.11/dist-packages (from widgetsnbextension~=3.6.0->ipywidgets) (6.5.5)\n",
            "Requirement already satisfied: parso<0.9.0,>=0.8.4 in /usr/local/lib/python3.11/dist-packages (from jedi>=0.16->ipython>=4.0.0->ipywidgets) (0.8.4)\n",
            "Requirement already satisfied: jupyter-core>=4.6.0 in /usr/local/lib/python3.11/dist-packages (from jupyter-client>=6.1.12->ipykernel>=4.5.1->ipywidgets) (5.7.2)\n",
            "Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.11/dist-packages (from jupyter-client>=6.1.12->ipykernel>=4.5.1->ipywidgets) (2.8.2)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (3.1.6)\n",
            "Requirement already satisfied: argon2-cffi in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (23.1.0)\n",
            "Requirement already satisfied: nbformat in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (5.10.4)\n",
            "Requirement already satisfied: nbconvert>=5 in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (7.16.6)\n",
            "Requirement already satisfied: Send2Trash>=1.8.0 in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.8.3)\n",
            "Requirement already satisfied: terminado>=0.8.3 in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.18.1)\n",
            "Requirement already satisfied: prometheus-client in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.21.1)\n",
            "Requirement already satisfied: nbclassic>=0.4.7 in /usr/local/lib/python3.11/dist-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.2.0)\n",
            "Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.11/dist-packages (from pexpect>4.3->ipython>=4.0.0->ipywidgets) (0.7.0)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.11/dist-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython>=4.0.0->ipywidgets) (0.2.13)\n",
            "Requirement already satisfied: platformdirs>=2.5 in /usr/local/lib/python3.11/dist-packages (from jupyter-core>=4.6.0->jupyter-client>=6.1.12->ipykernel>=4.5.1->ipywidgets) (4.3.6)\n",
            "Requirement already satisfied: notebook-shim>=0.2.3 in /usr/local/lib/python3.11/dist-packages (from nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.2.4)\n",
            "Requirement already satisfied: beautifulsoup4 in /usr/local/lib/python3.11/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (4.13.3)\n",
            "Requirement already satisfied: bleach!=5.0.0 in /usr/local/lib/python3.11/dist-packages (from bleach[css]!=5.0.0->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (6.2.0)\n",
            "Requirement already satisfied: defusedxml in /usr/local/lib/python3.11/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.7.1)\n",
            "Requirement already satisfied: jupyterlab-pygments in /usr/local/lib/python3.11/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.3.0)\n",
            "Requirement already satisfied: markupsafe>=2.0 in /usr/local/lib/python3.11/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (3.0.2)\n",
            "Requirement already satisfied: mistune<4,>=2.0.3 in /usr/local/lib/python3.11/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (3.1.2)\n",
            "Requirement already satisfied: nbclient>=0.5.0 in /usr/local/lib/python3.11/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.10.2)\n",
            "Requirement already satisfied: pandocfilters>=1.4.1 in /usr/local/lib/python3.11/dist-packages (from nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.5.1)\n",
            "Requirement already satisfied: fastjsonschema>=2.15 in /usr/local/lib/python3.11/dist-packages (from nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (2.21.1)\n",
            "Requirement already satisfied: jsonschema>=2.6 in /usr/local/lib/python3.11/dist-packages (from nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (4.23.0)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.1->jupyter-client>=6.1.12->ipykernel>=4.5.1->ipywidgets) (1.17.0)\n",
            "Requirement already satisfied: argon2-cffi-bindings in /usr/local/lib/python3.11/dist-packages (from argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (21.2.0)\n",
            "Requirement already satisfied: webencodings in /usr/local/lib/python3.11/dist-packages (from bleach!=5.0.0->bleach[css]!=5.0.0->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.5.1)\n",
            "Requirement already satisfied: tinycss2<1.5,>=1.1.0 in /usr/local/lib/python3.11/dist-packages (from bleach[css]!=5.0.0->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.4.0)\n",
            "Requirement already satisfied: attrs>=22.2.0 in /usr/local/lib/python3.11/dist-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (25.2.0)\n",
            "Requirement already satisfied: jsonschema-specifications>=2023.03.6 in /usr/local/lib/python3.11/dist-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (2024.10.1)\n",
            "Requirement already satisfied: referencing>=0.28.4 in /usr/local/lib/python3.11/dist-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.36.2)\n",
            "Requirement already satisfied: rpds-py>=0.7.1 in /usr/local/lib/python3.11/dist-packages (from jsonschema>=2.6->nbformat->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (0.23.1)\n",
            "Requirement already satisfied: jupyter-server<3,>=1.8 in /usr/local/lib/python3.11/dist-packages (from notebook-shim>=0.2.3->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.24.0)\n",
            "Requirement already satisfied: cffi>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.17.1)\n",
            "Requirement already satisfied: soupsieve>1.2 in /usr/local/lib/python3.11/dist-packages (from beautifulsoup4->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (2.6)\n",
            "Requirement already satisfied: typing-extensions>=4.0.0 in /usr/local/lib/python3.11/dist-packages (from beautifulsoup4->nbconvert>=5->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (4.12.2)\n",
            "Requirement already satisfied: pycparser in /usr/local/lib/python3.11/dist-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (2.22)\n",
            "Requirement already satisfied: anyio<4,>=3.1.0 in /usr/local/lib/python3.11/dist-packages (from jupyter-server<3,>=1.8->notebook-shim>=0.2.3->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (3.7.1)\n",
            "Requirement already satisfied: websocket-client in /usr/local/lib/python3.11/dist-packages (from jupyter-server<3,>=1.8->notebook-shim>=0.2.3->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.8.0)\n",
            "Requirement already satisfied: idna>=2.8 in /usr/local/lib/python3.11/dist-packages (from anyio<4,>=3.1.0->jupyter-server<3,>=1.8->notebook-shim>=0.2.3->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (3.10)\n",
            "Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.11/dist-packages (from anyio<4,>=3.1.0->jupyter-server<3,>=1.8->notebook-shim>=0.2.3->nbclassic>=0.4.7->notebook>=4.4.1->widgetsnbextension~=3.6.0->ipywidgets) (1.3.1)\n",
            "Downloading jedi-0.19.2-py2.py3-none-any.whl (1.6 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.6/1.6 MB\u001b[0m \u001b[31m51.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hInstalling collected packages: jedi\n",
            "Successfully installed jedi-0.19.2\n"
          ]
        }
      ],
      "source": [
        "!pip install rank_bm25 nltk transformers\n",
        "!pip install ipywidgets"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "_cell_guid": "79c7e3d0-c299-4dcb-8224-4455121ee9b0",
        "_uuid": "d629ff2d2480ee46fbb7e2d37f6b5fab8052498a",
        "tags": [],
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "P4SFhCr2YSr-",
        "outputId": "1f962778-d8c6-409c-bb95-9ec0c2e473e3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "[nltk_data] Downloading package punkt to /root/nltk_data...\n",
            "[nltk_data]   Unzipping tokenizers/punkt.zip.\n",
            "[nltk_data] Downloading package stopwords to /root/nltk_data...\n",
            "[nltk_data]   Unzipping corpora/stopwords.zip.\n",
            "[nltk_data] Downloading package punkt_tab to /root/nltk_data...\n",
            "[nltk_data]   Unzipping tokenizers/punkt_tab.zip.\n"
          ]
        }
      ],
      "source": [
        "import re\n",
        "import os\n",
        "import json\n",
        "from pathlib import Path, PurePath\n",
        "\n",
        "import requests\n",
        "from requests.exceptions import HTTPError, ConnectionError\n",
        "\n",
        "import numpy as np\n",
        "import pandas as pd\n",
        "from tqdm import tqdm\n",
        "from rank_bm25 import BM25Okapi\n",
        "\n",
        "import nltk\n",
        "from nltk.corpus import stopwords\n",
        "nltk.download(\"punkt\")\n",
        "nltk.download('stopwords')\n",
        "nltk.download('punkt_tab')\n",
        "\n",
        "import torch\n",
        "from transformers import BertTokenizer, BertForQuestionAnswering\n",
        "\n",
        "import warnings\n",
        "warnings.filterwarnings(\"ignore\")"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "iRPikPh9YSr-"
      },
      "source": [
        "# Предварительная обработка данных\n",
        "\n",
        "[Загрузите](https://disk.yandex.ru/d/Z3MfKtbtCwdZkw) полнотекстовые публикации и очистите их."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "tags": [],
        "id": "EKloENXDYSr-"
      },
      "outputs": [],
      "source": [
        "merged = pd.read_csv(r'processed_data_v8_2 2.csv', sep=';')"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "tags": [],
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 689
        },
        "id": "2YyJdg2RYSr_",
        "outputId": "a41d829f-1884-4792-b36c-86fe781cb0ab"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "   paper_id                                            authors  \\\n",
              "0  sl79r65r  Zhang, Zhonghua; Jiang, Shan; Liu, Yun; Sun, Y...   \n",
              "1  wvx6q999                                                NaN   \n",
              "2  z9jnljrt  Cao, Yurou; Gao, Lulu; Zhang, Li; Zhou, Lixian...   \n",
              "3  dqpujzo8  Li, Hong-Ying; Zhu, Guang-Jian; Zhang, Yun-Zhi...   \n",
              "4  q6mhqcho  Zhou, Zhen; Zhang, Pan; Cui, Yuxia; Zhang, Yon...   \n",
              "\n",
              "                                               title  \\\n",
              "0  Identification of ireA, 0007, 0008, and 2235 a...   \n",
              "1  Note from the editors: novel coronavirus (2019...   \n",
              "2  Genome-wide screening of lipoproteins in Actin...   \n",
              "3  A qualitative study of zoonotic risk factors a...   \n",
              "4  Experiments Investigating the Competitive Grow...   \n",
              "\n",
              "                                            abstract  \\\n",
              "0  Avian pathogenic Escherichia coli (APEC), a pa...   \n",
              "1                                                NaN   \n",
              "2  Actinobacillus pleuropneumoniae is an importan...   \n",
              "3  BACKGROUND: Strategies are urgently needed to ...   \n",
              "4  Human metapneumovirus (hMPV) is an important p...   \n",
              "\n",
              "                                                text        date  \\\n",
              "0  Avian pathogenic Escherichia coli (APEC), a pa...  2020-01-23   \n",
              "1  At the end of 2019, on 31 December, the World ...  2020-01-23   \n",
              "2  Actinobacillus pleuropneumoniae is a Gram-nega...  2020-02-11   \n",
              "3  Emerging and re-emerging zoonotic diseases are...  2020-02-10   \n",
              "4  Human metapneumovirus (hMPV), which was first ...  2020-02-18   \n",
              "\n",
              "                                                 url  risk_factor_age  \\\n",
              "0  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...            False   \n",
              "1  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...            False   \n",
              "2  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...            False   \n",
              "3  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...             True   \n",
              "4  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...             True   \n",
              "\n",
              "   risk_factor_sex  risk_factor_overweight  ...  \\\n",
              "0            False                   False  ...   \n",
              "1            False                   False  ...   \n",
              "2            False                   False  ...   \n",
              "3             True                   False  ...   \n",
              "4            False                   False  ...   \n",
              "\n",
              "   tag_design_retrospective_cohort  tag_design_cross_sectional_case_control  \\\n",
              "0                            False                                    False   \n",
              "1                            False                                    False   \n",
              "2                            False                                    False   \n",
              "3                            False                                    False   \n",
              "4                            False                                    False   \n",
              "\n",
              "   tag_design_matched_case_control  tag_design_prevalence_survey  \\\n",
              "0                            False                         False   \n",
              "1                            False                         False   \n",
              "2                            False                         False   \n",
              "3                            False                         False   \n",
              "4                            False                         False   \n",
              "\n",
              "   tag_design_time_series_analysis  tag_design_systematic_review  \\\n",
              "0                            False                         False   \n",
              "1                            False                         False   \n",
              "2                            False                         False   \n",
              "3                            False                         False   \n",
              "4                            False                         False   \n",
              "\n",
              "   tag_design_randomized_control  tag_design_pseudo_randomized_control  \\\n",
              "0                          False                                 False   \n",
              "1                          False                                 False   \n",
              "2                          False                                 False   \n",
              "3                          False                                 False   \n",
              "4                          False                                 False   \n",
              "\n",
              "   tag_design_case_study  tag_design_simulation  \n",
              "0                  False                  False  \n",
              "1                  False                  False  \n",
              "2                  False                  False  \n",
              "3                  False                  False  \n",
              "4                   True                  False  \n",
              "\n",
              "[5 rows x 36 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-e33ff4de-8964-408d-8991-dca1933128fa\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>paper_id</th>\n",
              "      <th>authors</th>\n",
              "      <th>title</th>\n",
              "      <th>abstract</th>\n",
              "      <th>text</th>\n",
              "      <th>date</th>\n",
              "      <th>url</th>\n",
              "      <th>risk_factor_age</th>\n",
              "      <th>risk_factor_sex</th>\n",
              "      <th>risk_factor_overweight</th>\n",
              "      <th>...</th>\n",
              "      <th>tag_design_retrospective_cohort</th>\n",
              "      <th>tag_design_cross_sectional_case_control</th>\n",
              "      <th>tag_design_matched_case_control</th>\n",
              "      <th>tag_design_prevalence_survey</th>\n",
              "      <th>tag_design_time_series_analysis</th>\n",
              "      <th>tag_design_systematic_review</th>\n",
              "      <th>tag_design_randomized_control</th>\n",
              "      <th>tag_design_pseudo_randomized_control</th>\n",
              "      <th>tag_design_case_study</th>\n",
              "      <th>tag_design_simulation</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>sl79r65r</td>\n",
              "      <td>Zhang, Zhonghua; Jiang, Shan; Liu, Yun; Sun, Y...</td>\n",
              "      <td>Identification of ireA, 0007, 0008, and 2235 a...</td>\n",
              "      <td>Avian pathogenic Escherichia coli (APEC), a pa...</td>\n",
              "      <td>Avian pathogenic Escherichia coli (APEC), a pa...</td>\n",
              "      <td>2020-01-23</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>wvx6q999</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Note from the editors: novel coronavirus (2019...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>At the end of 2019, on 31 December, the World ...</td>\n",
              "      <td>2020-01-23</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>z9jnljrt</td>\n",
              "      <td>Cao, Yurou; Gao, Lulu; Zhang, Li; Zhou, Lixian...</td>\n",
              "      <td>Genome-wide screening of lipoproteins in Actin...</td>\n",
              "      <td>Actinobacillus pleuropneumoniae is an importan...</td>\n",
              "      <td>Actinobacillus pleuropneumoniae is a Gram-nega...</td>\n",
              "      <td>2020-02-11</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>dqpujzo8</td>\n",
              "      <td>Li, Hong-Ying; Zhu, Guang-Jian; Zhang, Yun-Zhi...</td>\n",
              "      <td>A qualitative study of zoonotic risk factors a...</td>\n",
              "      <td>BACKGROUND: Strategies are urgently needed to ...</td>\n",
              "      <td>Emerging and re-emerging zoonotic diseases are...</td>\n",
              "      <td>2020-02-10</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>True</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>q6mhqcho</td>\n",
              "      <td>Zhou, Zhen; Zhang, Pan; Cui, Yuxia; Zhang, Yon...</td>\n",
              "      <td>Experiments Investigating the Competitive Grow...</td>\n",
              "      <td>Human metapneumovirus (hMPV) is an important p...</td>\n",
              "      <td>Human metapneumovirus (hMPV), which was first ...</td>\n",
              "      <td>2020-02-18</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>...</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>False</td>\n",
              "      <td>True</td>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>5 rows × 36 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-e33ff4de-8964-408d-8991-dca1933128fa')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-e33ff4de-8964-408d-8991-dca1933128fa button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-e33ff4de-8964-408d-8991-dca1933128fa');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-dea85fba-a18c-420c-977c-490c9b8bea32\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-dea85fba-a18c-420c-977c-490c9b8bea32')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-dea85fba-a18c-420c-977c-490c9b8bea32 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "merged"
            }
          },
          "metadata": {},
          "execution_count": 5
        }
      ],
      "source": [
        "merged.head()"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "f = merged.loc[0].to_frame().fillna('')\n",
        "f.columns = ['Value']\n",
        "f.iloc[:10]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 363
        },
        "id": "R8wiZesYttCQ",
        "outputId": "aaf66df9-3286-433e-a71a-02614eb97f17"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                                    Value\n",
              "paper_id                                                         sl79r65r\n",
              "authors                 Zhang, Zhonghua; Jiang, Shan; Liu, Yun; Sun, Y...\n",
              "title                   Identification of ireA, 0007, 0008, and 2235 a...\n",
              "abstract                Avian pathogenic Escherichia coli (APEC), a pa...\n",
              "text                    Avian pathogenic Escherichia coli (APEC), a pa...\n",
              "date                                                           2020-01-23\n",
              "url                     https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...\n",
              "risk_factor_age                                                     False\n",
              "risk_factor_sex                                                     False\n",
              "risk_factor_overweight                                              False"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-408b7a34-7f74-4adc-8a70-ad2cc67ae154\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Value</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>paper_id</th>\n",
              "      <td>sl79r65r</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>authors</th>\n",
              "      <td>Zhang, Zhonghua; Jiang, Shan; Liu, Yun; Sun, Y...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>title</th>\n",
              "      <td>Identification of ireA, 0007, 0008, and 2235 a...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>abstract</th>\n",
              "      <td>Avian pathogenic Escherichia coli (APEC), a pa...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>text</th>\n",
              "      <td>Avian pathogenic Escherichia coli (APEC), a pa...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>date</th>\n",
              "      <td>2020-01-23</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>url</th>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>risk_factor_age</th>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>risk_factor_sex</th>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>risk_factor_overweight</th>\n",
              "      <td>False</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-408b7a34-7f74-4adc-8a70-ad2cc67ae154')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-408b7a34-7f74-4adc-8a70-ad2cc67ae154 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-408b7a34-7f74-4adc-8a70-ad2cc67ae154');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-fa6a719b-f7e8-41bb-a54b-d117fd2270cf\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-fa6a719b-f7e8-41bb-a54b-d117fd2270cf')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-fa6a719b-f7e8-41bb-a54b-d117fd2270cf button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "summary": "{\n  \"name\": \"f\",\n  \"rows\": 10,\n  \"fields\": [\n    {\n      \"column\": \"Value\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 8,\n        \"samples\": [\n          \"Zhang, Zhonghua; Jiang, Shan; Liu, Yun; Sun, Yu; Yu, Peixin; Gong, Qianwen; Zeng, Hang; Li, Yihao; Xue, Feng; Zhuge, Xiangkai; Ren, Jianluan; Dai, Jianjun; Tang, Fang\",\n          \"2020-01-23\",\n          \"sl79r65r\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 6
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "bgkbRSFxYSr_"
      },
      "source": [
        "## Задача №1\n",
        "\n",
        "Создайте классную работу, чтобы упростить получение какого-либо поля, такого как аннотация, полный текст, doi и так далее."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "tags": [],
        "id": "9nQfwRnsYSr_"
      },
      "outputs": [],
      "source": [
        "class Paper:\n",
        "    def __init__(self, item):\n",
        "        self.paper = item.to_frame().fillna('')\n",
        "        # self.paper.columns = ['Value']\n",
        "    def url(self):\n",
        "        return self._get_field('url')\n",
        "    def text(self):\n",
        "        return self._get_field('text')\n",
        "    def abstract(self):\n",
        "        return self._get_field('abstract')\n",
        "    def title(self):\n",
        "        return self._get_field('title')\n",
        "    def authors(self, split=False):\n",
        "        return self._get_field('authors')\n",
        "    def _get_field(self, field_name):\n",
        "        return self.paper.get(field_name, '')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "78muyLFUYSr_"
      },
      "source": [
        "## Задача №2\n",
        "\n",
        "Некоторые функции предварительной обработки для очистки исходных данных. Мы обозначаем текст, удаляем знаки препинания и некоторые специальные символы."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "tags": [],
        "id": "0eyZEY9xYSr_"
      },
      "outputs": [],
      "source": [
        "english_stopwords = list(set(stopwords.words('english')))\n",
        "SEARCH_DISPLAY_COLUMNS = ['title', 'abstract', 'url', 'authors', 'text']"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "tags": [],
        "id": "WTS5zIv7YSr_"
      },
      "outputs": [],
      "source": [
        "def strip_characters(text):\n",
        "    return re.sub(r'[\\(\\)\\[\\]\\{\\}:;,\\\\.!?\\'\\\"-]', '', text)\n",
        "\n",
        "def clean(text):\n",
        "    text = text.lower()\n",
        "    text = strip_characters(text)\n",
        "    text = re.sub(r'\\s+', ' ', text).strip()\n",
        "    return text\n",
        "\n",
        "def tokenize(text):\n",
        "    words = nltk.word_tokenize(text)\n",
        "    return list(\n",
        "        set([word for word in words\n",
        "             if len(word) > 1\n",
        "             and not word in english_stopwords\n",
        "             and not (word.isnumeric() and len(word) is not 4)\n",
        "             and (not word.isnumeric() or word.isalpha())])\n",
        "    )\n",
        "\n",
        "def preprocess(text):\n",
        "    t = clean(text)\n",
        "    tokens = tokenize(t)\n",
        "    return tokens\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "n3K8b7h7YSr_"
      },
      "source": [
        "# Поисковая система BM25\n",
        "Определите класс для построения индекса и поиска по строкам."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "tags": [],
        "id": "auCWYXDxYSr_"
      },
      "outputs": [],
      "source": [
        "class SearchResults:\n",
        "    def __init__(self,\n",
        "                 data: pd.DataFrame,\n",
        "                 columns = None):\n",
        "        self.results = data\n",
        "        if columns:\n",
        "            self.results = self.results[columns]\n",
        "\n",
        "    def __getitem__(self, item):\n",
        "        return Paper(self.results.loc[item])\n",
        "\n",
        "    def __len__(self):\n",
        "        return len(self.results)\n",
        "\n",
        "    def set_ans(self, ans):\n",
        "        col_name = self.results.columns.tolist()\n",
        "        col_name.insert(2, 'Answer')\n",
        "        self.results = self.results.reindex(columns=col_name)\n",
        "        self.results['Answer'] = ans"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "id": "e7Bj-9-kYSr_"
      },
      "outputs": [],
      "source": [
        "class WordTokenIndex:\n",
        "    def __init__(self,\n",
        "                 corpus: pd.DataFrame,\n",
        "                 columns=SEARCH_DISPLAY_COLUMNS):\n",
        "        self.corpus = corpus\n",
        "\n",
        "        raw_search_str = self.corpus.abstract.fillna('') + ' ' + self.corpus.title.fillna('')\n",
        "        self.index = raw_search_str.apply(preprocess).to_frame()\n",
        "        self.index.columns = ['terms']\n",
        "        self.index.index = self.corpus.index\n",
        "        self.columns = columns\n",
        "\n",
        "    def search(self, search_string):\n",
        "        search_terms = preprocess(search_string)\n",
        "        # получить индексы, включающие строку search_string\n",
        "        result_index = self.index.terms.apply(lambda terms: any(i in terms for i in search_terms))\n",
        "        # получить найденные papers\n",
        "        results = self.corpus[result_index].copy().reset_index().rename(columns={'index':'paper'})\n",
        "\n",
        "        return SearchResults(results, self.columns + ['paper'])"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "tags": [],
        "id": "Ug4kUz3rYSr_"
      },
      "outputs": [],
      "source": [
        "class RankBM25Index(WordTokenIndex):\n",
        "    def __init__(self, corpus: pd.DataFrame, columns=SEARCH_DISPLAY_COLUMNS):\n",
        "        super().__init__(corpus, columns)\n",
        "        self.bm25 = BM25Okapi(self.index.terms.tolist())\n",
        "\n",
        "    def search(self, search_string, n=4):\n",
        "        search_terms = preprocess(search_string)\n",
        "        doc_scores = self.bm25.get_scores(search_terms)\n",
        "\n",
        "        ind = np.argsort(doc_scores)[::-1][:n]\n",
        "        results = self.corpus.iloc[ind][self.columns]\n",
        "        results['Score'] = doc_scores[ind]\n",
        "        results = results[results.Score > 0]\n",
        "\n",
        "        return SearchResults(results.reset_index(), self.columns + ['Score'])"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "D9LOEa1JYSr_"
      },
      "source": [
        "# Bert furtuer pre-train and fine\n",
        "Используйте реализацию transformers bert и загрузите предварительно обученную модель \"bert-base-uncased\". Мы используем все статьи в качестве корпуса для дальнейшего предварительного обучения bert задаче языковой модели. После дальнейшей предварительной подготовки мы используем набор данных SQuAD-2.0 для точной настройки модели bert. Наконец, мы сохраняем обученную модель, которая называется \"output_squad\"."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "tags": [],
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 212,
          "referenced_widgets": [
            "3c6fbd30273d4553ae106af589a3419d",
            "60564d4481b94a05bf47aba45ca668c7",
            "a3dc95c83f764fec90f65384d08b75df",
            "90049ac4ba384a3ea039ae1ab4bbee80",
            "8138efe3620048ff94265c671b7d2b1d",
            "3a25dc5c597f4879ab09341ce335975d",
            "008f8b0703c74a668a679319b798e2a5",
            "df687ca3da864c64862111254cedd54c",
            "8b6a0b1925b74f878f08c5fd2830619d",
            "0c43033fcabf4d36b7d7fe6a8abcd86e",
            "304a4909b3f844c2b41812257347ddb7",
            "1ccdf56406194717bde2687919967c43",
            "4e4a163366834a3787d9018c76540473",
            "2d7cac9eff8b463c8c6a6dc40d1ae84c",
            "4c277cf53ac640b2a6c4d87582bdddc8",
            "fb54ab2fadfb41158a7c2eb99a0de4de",
            "659d23c615ab4edba8222f4da2a7c0ed",
            "8217318c1ded4f1daf137b6d1c17da96",
            "1a2264de8a404bbf9cff8b9842ef647a",
            "da354a16760b49e385749cf2e3f402a4",
            "34bdb07c091f416ba769884597daf78d",
            "75c7a41b98084c1cba7f4a309ecc6622",
            "c337bb8619964924835c99744efa3240",
            "368f8f920e944852be18cada1bfd549e",
            "f33f65c2f84647d994427a86cbc8b00e",
            "770363869a4a43e899c8ae628102e031",
            "04f48f7b7b754d17a8946745c672338e",
            "3e57fe4104e44deb84cb7c1de76ec212",
            "b9919b88af1545278e13b05914542227",
            "abacb2cfc4c744b7a170af68278a4e19",
            "fb0e0740e00c4c27bdca2c6cbeafee07",
            "bf266f0cda1c4345be7b0a4c5757c35b",
            "dd30000507bd4b918a14822f36e95be6",
            "e5c0b47ee8164423b8885addab72fed9",
            "b52e7f99bd0c4b2785c8bf8026dd7c9d",
            "138da3d243bb46d8b4353213384e80d5",
            "a99e9ce48fe642aabd9682a5227b3e8f",
            "12cee6cefa954965bb01e3961996ffd4",
            "81988520ee4b43e69d566deaaf70d9ce",
            "934efc5c653c4b5f97db0ac87389a87f",
            "1211c83210b0494d857777b9b33d6e26",
            "24e3ad870e18400ca3111b3e304c81e4",
            "8e0bbbc666db4fd18ff968b66f8583de",
            "a9481c7ba0244215a9ac7aec04560519",
            "9527c1603ad34fb392a9dfca276b1607",
            "9fad40517da54635a78242d92e483bbc",
            "cfbf0c8439e64e82a9e4356aa7c48d5d",
            "79b57a556f3a4d6095be76c61afbbb59",
            "69bbf095ac64458aa968ea200e30cda8",
            "7a00990118494b50970505d484d25fb6",
            "a7f13ffcafb0493eb7614ef218b388e6",
            "f9895f853d0449b6af4ec2d7358ba22e",
            "adc6500a77904717852cd78d7d64cb21",
            "aadb941603f1422ab485a5287ea1aa7a",
            "d4a03a9ba3aa4d33bad070d014e572b7"
          ]
        },
        "id": "Dd4PxFqlYSr_",
        "outputId": "5c6a1ece-8bd6-4d0d-f6dd-c745603bc12d"
      },
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "tokenizer_config.json:   0%|          | 0.00/48.0 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "3c6fbd30273d4553ae106af589a3419d"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "vocab.txt:   0%|          | 0.00/232k [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "1ccdf56406194717bde2687919967c43"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "tokenizer.json:   0%|          | 0.00/466k [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "c337bb8619964924835c99744efa3240"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "config.json:   0%|          | 0.00/570 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "e5c0b47ee8164423b8885addab72fed9"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "model.safetensors:   0%|          | 0.00/440M [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "9527c1603ad34fb392a9dfca276b1607"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Some weights of BertForQuestionAnswering were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['qa_outputs.bias', 'qa_outputs.weight']\n",
            "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n"
          ]
        }
      ],
      "source": [
        "tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')\n",
        "model = BertForQuestionAnswering.from_pretrained('bert-base-uncased')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "tags": [],
        "id": "2YrhFSEJYSr_"
      },
      "source": [
        "# QA function\n",
        "Создайте функцию контроля качества, которая получает вопрос и фрагмент текста, передает их в модель bert и выводит возможный ответ в тексте. Если модель выдает неверный ответ, функция выдаст \"Нет ответа\"."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "naiFSQl-YSr_"
      },
      "source": [
        "## Постройте пайплайн поиска и покажите результаты\n",
        "\n",
        "### Задача №3\n",
        "Мы показываем выходные данные модели в таблицах и заключаем их в \"【】\", чтобы представить ответ в статье."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 97,
      "metadata": {
        "tags": [],
        "id": "JB5pu-4PYSsA"
      },
      "outputs": [],
      "source": [
        "max_question_len = 100\n",
        "\n",
        "def getAnswer(question, text):\n",
        "    input_ids = tokenizer.encode(question, text, max_length=512, truncation=True)\n",
        "    # mask text sequence  # 102 - SEP между question и text\n",
        "    token_type_ids = [0 if i <= input_ids.index(102) else 1 for i in range(len(input_ids))]\n",
        "\n",
        "    model_out = model(torch.tensor([input_ids]), token_type_ids=torch.tensor([token_type_ids]))\n",
        "    start_scores, end_scores = model_out['start_logits'], model_out['end_logits']\n",
        "\n",
        "    all_tokens = tokenizer.convert_ids_to_tokens(input_ids)\n",
        "\n",
        "    # выделить  часть за начало ответа и его конец\n",
        "    start_scores = start_scores[0, input_ids.index(102):]  # bs = 1\n",
        "    end_scores = end_scores[0, input_ids.index(102):]  # bs = 1\n",
        "\n",
        "    if torch.argmax(start_scores).item() >= torch.argmax(end_scores).item():\n",
        "        return \"No Answer\"\n",
        "\n",
        "    # выделить токены до ответа + [ начало контекста ответа\n",
        "    all_tokens = all_tokens[: input_ids.index(102) + torch.argmax(start_scores)] + [\"【\"] + all_tokens[input_ids.index(102) + torch.argmax(start_scores):]\n",
        "    # выделить токены конца ответа контекста ] + токены после ответа\n",
        "    all_tokens = all_tokens[: input_ids.index(102) + torch.argmax(end_scores) + 1] + [\"】\"] + all_tokens[input_ids.index(102) + torch.argmax(end_scores) + 1:]\n",
        "\n",
        "    # найти начало и конец ответа\n",
        "    start_span = max(input_ids.index(102) + torch.argmax(start_scores) - 5, input_ids.index(102))\n",
        "    end_span = min(input_ids.index(102) + torch.argmax(end_scores) + 6, len(all_tokens) + 1)\n",
        "\n",
        "    answer = tokenizer.convert_tokens_to_string(all_tokens[start_span:end_span])\n",
        "    answer = answer.replace(\"[SEP]\", \"\")\n",
        "    return answer"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "text = merged.iloc[0].text\n",
        "text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 123
        },
        "id": "-hI7w2GkycbX",
        "outputId": "481219ad-b2da-4447-8a92-4788dc93c2fa"
      },
      "execution_count": 47,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Avian pathogenic Escherichia coli (APEC), a pathotype of extraintestinal pathogenic E. coli (ExPEC), causes serious infectious diseases in poultry [1, 2]. Different serotypes of APEC cause local or systemic infectious diseases in poultry, including respiratory infections, sepsis, polyserositis, coligranuloma, cellulitis, yolk sac infection, omphalitis, and swollen head syndrome, resulting in significant economic losses to the poultry industry [3]. Strains of APEC and neonatal meningitis-associated E. coli (NMEC, a subpathotype of ExPEC), the latter of which causes infections in humans, reportedly share some common virulence genes [4–6]. It is thus particularly important to study the genes encoding virulence factors in APEC strains. These strains contain several virulence-associated genes that encode various virulence factors, including adhesins (fimC, ompA, papC), invasins (ibeA), avian haemolysins (hlyF), serum survival proteins (iss, ompT), and siderophores (iutA, fyuA, iroN); moreover, they show the presence of a pathogenicity island [7–9].\\nIron is a vital micronutrient that regulates enzyme activity and metabolism. This element plays a key role in basic cellular processes, such as cellular respiration, DNA replication, and electron transport and is accordingly essential for bacterial survival in host tissues [10–12]. Furthermore, iron is a necessary growth factor for bacteria and is reportedly involved in the expression of bacterial virulence factors [13]. Iron uptake factors evidently play a pivotal role in E. coli growth and pathogenesis. Bacteria employ different strategies to absorb iron from their environment, including siderophore-mediated iron uptake, which occurs on the cell surface [14]. In addition, bacteria either excessively reduce the external pH or dissolve iron oxide to meet their iron requirements by reducing ferric iron to a relatively soluble ferrous form. Another common strategy is to synthesize and secrete iron chelators, such as siderophores, as intracellular iron (Fe2+) is rarely found in natural conditions, and Fe2+ can be readily oxidized to Fe3+ in the presence of oxygen and water [15]. Siderophores then combine with the available iron (Fe3+) to form an iron–siderophore complex, which binds to specific receptor proteins on the bacterial cell surface, consequently entering cells via the TonB-dependent transport system, followed by iron release. Siderophore-mediated ferric uptake requires specific outer membrane (OM) receptors, such as FhuA, FecA, and FepA, which reportedly exist in E. coli [16]. These OM receptors share the same structural properties and acquire energy by coupling with TonB proteins located in the inner membrane, and they are referred to as TonB-dependent receptors (TBDRs) [17].\\nTBDRs are known to actively transport ferric–siderophore complexes in Gram-negative bacteria, and they also transport diverse antibiotics, vitamins, nickel complexes, and carbohydrates [18–20]. Transporters involved in iron uptake have very strict siderophore selectivity. A strong correlation exists between the amount of iron and siderophores that bacteria can use and the number of genes encoding iron-regulated TBDRs [19]. In simple terms, iron depletion triggers the upregulation of genes encoding TBDRs.\\nIn this study, upon analysing the whole genome of the APEC strain DE205B, we identified six putative TBDRs; however, the construction of two mutants failed. Thus, we eventually investigated the roles of four putative TBDRs—ireA, 0007, 0008, and 2235—in iron uptake.\\nThe bacterial strains and plasmids used in this study are listed in Table 1. The APEC strain DE205B (O2:K1), which was isolated from a duck with neurological symptoms, was previously determined to have no antibiotic (ampicillin, kanamycin, and chloramphenicol) resistance [21–25]. All the APEC strains used in this study were cultured in Luria–Bertani (LB, Oxoid, Thermo-Fisher, USA) or solid LB medium supplemented with 2% agar, followed by incubation in a shaker at 37 °C and 180 rpm unless otherwise stated. For culture of the mutant strain, appropriate antibiotics (ampicillin, 100 μg/mL; kanamycin, 50 μg/mL; chloramphenicol, 15 μg/mL) were added to the LB media. M9 medium was prepared as follows: 200 mL of 5 × M9 salt solution (12.8 g Na2PO4·7H2O, 3.0 g KH2PO4, 0.5 g NaCl, and 1.0 g NH4Cl) was mixed with 2 mL of 1 M MgSO4, 20 mL of 20% glucose, and 0.1 mL of 1 M CaCl2. This solution was then dissolved in 1000 mL double-distilled water and filtered through a 0.22-μm membrane. Fe (0.1 mM FeCl3·6H2O) was added or not added to M9 medium to establish iron-rich or iron-depleted conditions, respectively.\\n\\nFor analysis of the role of the four TBDRs in iron uptake, three genes encoding TBDRs (0007, 0008, 2235) were knocked out, and the ireA mutant strain, which was previously constructed [21], was used in this study. A single mutant strain for each of the three genes (0007, 0008, 2235) was constructed using the Red homologous recombination method in E. coli [26]. Briefly, for generation of the 0007 mutant strain, a gene-targeting fragment containing a homologous arm on both sides of 0007 was amplified by PCR using the plasmid pKD4 as the template, which contains the phage λ Red system under the control of an arabinose promoter. Competent DE205B cells containing pKD46 were then prepared; l-arabinose was added to induce the expression of the phage λ Red system, and the target fragments were transformed by electroporation into DE205B to replace 0007 with the resistance gene. The recombinant strain was screened by growth on LB plates supplemented with kanamycin and identified by cross-PCR. Details of the primers (0007-F/0007-R, K1, K2) used for amplification are listed in Additional file 1. Next, the temperature-sensitive plasmid pCP20 was transformed into the recombinant strain to remove the resistance gene. Finally, pCP20 was removed by growth at 42 °C for 24 h to obtain the mutant strain DE205BΔ0007 that did not show any resistance. For analysis of the effect of tonB deletion on the TBDRs, a tonB mutant strain was also constructed using the same method. Similarly, the 0008, 2235, and tonB mutant strains, namely, DE205BΔ0008, DE205BΔ2235, and DE205BΔtonB, respectively, were obtained.\\nThe construction of the complemented strain involved recovering the deleted gene. The genomic DNA of DE205B was used as the template to amplify 0007, including its putative promoter gene. The PCR product was then purified and inserted into pSTV28, and the composite vector PSTV28-0007 was transformed by electroporation into DE205BΔ0007 to produce the complemented strain DE205BΔ0007/0007*. In the same manner, the complemented strains DE205BΔ0008/0008*, DE205BΔ2235/2235*, and DE205BΔtonB/tonB* were obtained. The ireA deletion strain DE205BΔireA and the complemented strain DE205BΔireA/ireA* were constructed in an earlier study and used in this study as well [21].\\nGrowth curves of the wild-type (WT), mutant (DE205BΔireA, DE205BΔ0007, DE205BΔ0008, DE205BΔ2235) and complemented strains (DE205BΔireA/ireA*, DE205BΔ0007/0007*, DE205BΔ0008/0008*, DE205BΔ2235/2235*) cultured in LB, iron-depleted M9 and iron-rich M9 media were constructed. All strains were grown overnight in 5 mL of LB medium supplemented with antibiotics, as appropriate. The cultured strains were centrifuged at 5000 rpm for 10 min to remove the supernatant, and the cell pellets were washed twice with PBS containing 200 mM 2,2′-dipyridyl; the optical density at 600 nm (OD600) was adjusted to 1. Then, 100 μL of the suspension was inoculated into 50 mL of LB medium, and 1000 μL of this solution was inoculated into 50 mL of iron-depleted or iron-rich M9 medium. The OD600 of the LB culture was measured using a spectrophotometer (Philes, Nanjing, China) every hour for a total of 10 h and that of the iron-depleted and iron-rich M9 cultures was measured at 0, 4, 8, 12, 24, and 48 h [27].\\nAs ireA, 0007 (GI: MN239889), 0008 (GI: MN239890), and 2235 (GI: MN239892) encode putative TBDRs, their expression levels were determined under iron-rich and iron-depleted conditions. DE205B was cultured to the mid-log phase in both types of M9 media, and the gene expression levels were determined using quantitative real-time PCR (qRT-PCR). Briefly, total RNA was extracted from bacteria grown under different culture conditions using a bacterial RNA kit (Omega Bio-Tek, Beijing, China) and reverse transcribed into cDNA using the PrimeScript™ RT reagent Kit (TaKaRa, Perfect Real Time, Japan) according to the manufacturer’s instructions. qRT-PCR was performed in a 20 μL reaction volume containing SYBR Green PCR Master Mix (Vazyme Biotech, Nanjing, China) and 0.1 μM primers (Additional file 1) specific to the iron uptake-related genes (0007, 0008, 2235, ireA, fecA, fhuA, iutA, iroN1, iroN2); the quantification data were analysed with ABI StepOne Software, version 2.3 (USA). For analysis of the effect of gene deletion on the ferric uptake system, the expression levels of iron uptake-related genes (fecA, fhuA, fepA, fepC, feoB, fyuA, iutA, chuA, iroN1, iroN2) were tested in LB medium. Total RNA was extracted from the WT, mutant (DE205BΔireA, DE205BΔ0007, DE205BΔ0008, DE205BΔ2235), and complemented (DE205BΔireA/ireA*, DE205BΔ0007/0007*, DE205BΔ0008/0008*, DE205BΔ2235/2235*) strains using the same method. For analysis of the effect of tonB on the expression levels of the genes encoding the four TBDRs, the gene (0007, 0008, 2235, ireA, fecA, fhuA, fepA, iutA, feoB) expression levels of the WT strain and DE205BΔtonB cultured in iron-depleted M9 medium were examined. Primer details are listed in Additional file 1. dnaE was used as an internal reference gene [21, 22]. The relative gene expression levels were calculated using the 2−△△Ct method; the values are expressed as percentages [28–30].\\nThe pertinent gene fragments (ireA, 0007, 0008, 2235) were inserted into the plasmid pET32a, and tonB was cloned into pGEX-4t-1. Proteins were expressed in BL21 (Vazyme Biotech) via addition of 1 mM isopropyl β-d-1-thiogalactopyranoside in the mid-log phase. The recombinant proteins IreA, 0007, 0008, and 2235 expressed in vitro carried a His tag, and TonB carried a glutathione-S-transferase (GST) tag. The primer details are listed in Additional file 1. The His-tag fusion proteins were purified by Ni-column chromatography, and the biological activity of the proteins in inclusion bodies was restored via a conventional method that involves using a urea-containing protein refolding solution (including 20 mM Tris–HCl, 1 mM GSH, 0.2 mM GSSG, 0.5 M NaCl). Finally, we concentrated and collected the putative TBDRs (IreA, 0007, 0008, 2235) with His tags by centrifugation in an ultrafiltration tube (10 kDa) at 4 °C. The final fusion proteins were thus obtained (34 kDa for 0007-His, 66 kDa for 0008-His, 95 kDa for 2235-His, and 93 kDa for IreA-His).\\nThe functional domain of TonB (residues 150–239) is critical, as it directly interacts with the OM receptor [31]. As such, to further verify whether the four putative TBDRs directly interact with TonB, we performed GST pulldown assays. Cells expressing the GST fusion proteins were lysed using an ultrasonic cell disrupter system (Thermo Fisher Scientific, China). Unlysed cells and impurities were removed by centrifugation (5000 rpm, 10 min), and the supernatant was incubated with GST-binding beads (Enriching Biotechnology, Ltd., Shanghai, China) for 3 h, followed by stringent washing with PBS (2 mM KH2PO4, 2.6 mM KCl, 8 mM Na2HPO4, 136 mM NaCl) to minimize nonspecific binding. Next, the samples were incubated with the putative TBDRs (IreA, 0007, 0008, 2235) with His tags. Samples were collected after multiple washes with PBS to minimize nonspecific binding. Finally, protein separation was performed by SDS–PAGE on 12% protein gels (Warbio, Shanghai, China), followed by protein detection via a Western blot assay with His-tagged monoclonal antibodies [32, 33]. Further, pGEX-4t-1 (no gene fragment inserted) was incubated with the putative TBDRs, and this was used as the negative control to exclude nonspecific binding of prey proteins to the beads or to GST itself.\\nTo verify whether the four TBDRs are directly involved in iron uptake, we determined the Fe concentrations in the WT, mutant, and complemented strains of DE205B using an iron colorimetric assay kit (Elabscience Biotech, Wuhan, China, catalogue no. E-BC-K139-S). The WT, mutant (DE205BΔireA, DE205BΔtonB, DE205BΔ0007, DE205BΔ0008, DE205BΔ2235), and complemented (DE205BΔireA/ireA*, DE205BΔtonB/tonB*, DE205BΔ0007/0007*, DE205BΔ0008/0008*, DE205BΔ2235/2235*) strains were incubated overnight in 5 mL of LB medium supplemented with antibiotics, as appropriate. The cultured strains were then centrifuged (5000 rpm, 10 min), and the supernatant was discarded. The obtained cell pellets were washed twice with PBS containing 200 M 2,2′-dipyridyl, and the OD600 was adjusted to 1. Then, 1000 μL of the suspension was inoculated into 100 mL of LB media. Briefly, the strains were grown to the log phase and washed three times with 0.9% NaCl. Cell pellets were subsequently obtained and suspended in PBS, and the cells were lysed using an ultrasonic cell disrupter system (Thermo Fisher Scientific, USA) to release intracellular iron. Finally, impurities were removed by centrifugation (5000 rpm, 10 min); a sample of the supernatant was used for iron content determination [34]. Deionized water (0.5 mL), iron standard stock solution (0.5 mL), and sample (0.5 mL) were individually mixed with a chromogenic agent (1.5 mL), boiled for 5 min, and then centrifuged at 2300 rpm for 10 min; the supernatant was subsequently collected. According to the manufacturer’s instructions for the iron colorimetric assay kit, iron content was estimated by measuring the OD at 520 nm (OD520) of the supernatant. The following formula was used: \\\\documentclass[12pt]{minimal}\\n\\t\\t\\t\\t\\\\usepackage{amsmath}\\n\\t\\t\\t\\t\\\\usepackage{wasysym} \\n\\t\\t\\t\\t\\\\usepackage{amsfonts} \\n\\t\\t\\t\\t\\\\usepackage{amssymb} \\n\\t\\t\\t\\t\\\\usepackage{amsbsy}\\n\\t\\t\\t\\t\\\\usepackage{mathrsfs}\\n\\t\\t\\t\\t\\\\usepackage{upgreek}\\n\\t\\t\\t\\t\\\\setlength{\\\\oddsidemargin}{-69pt}\\n\\t\\t\\t\\t\\\\begin{document}$$ \\\\begin{aligned} {\\\\text{Fe concentration }}\\\\left( {\\\\upmu{\\\\text{g}}/{\\\\text{L}}} \\\\right)\\\\, = \\\\, & \\\\left( {{\\\\text{sample OD}}_{ 5 20} \\\\, - \\\\,{\\\\text{deionized water OD}}_{ 5 20} /{\\\\text{iron standard stock solution OD}}_{ 5 20} \\\\,} \\\\right)\\\\, \\\\\\\\ & & {\\\\kern 1pt} {\\\\kern 1pt} {\\\\kern 1pt} - \\\\,{\\\\text{deionized water OD}}_{ 5 20} \\\\, \\\\times \\\\,{\\\\text{standard iron concentration }}\\\\left( { 2000 \\\\upmu{\\\\text{g}}/{\\\\text{L}}} \\\\right). \\\\\\\\ \\\\end{aligned} $$\\\\end{document}Fe concentrationμg/L=sample OD520-deionized water OD520/iron standard stock solution OD520-deionized water OD520×standard iron concentration2000μg/L.\\n\\nStatistical analyses were performed in GraphPad Prism 7.0 using unpaired t-tests [35]. In the figures, the error bars indicate the standard deviations, “*” represents P < 0.05, “**” represents P < 0.01, and “***” represents P < 0.001. qRT-PCR data from three individual experiments were used to determine the differences (fold change) in gene transcription levels. Similarly, in the growth curve and iron uptake analyses, each reaction was performed three times to overcome any experimental errors.\\nThe expression levels of all four genes were significantly upregulated under iron-depleted conditions compared with iron-rich conditions; the expression levels of 0007, 0008 and 2235 were upregulated 1.76 times (P < 0.001), 1.43 times (P < 0.05), and 1.71 times (P < 0.01), respectively. The ireA gene was the most upregulated (1.87 times, P < 0.001) (Figure 1). In addition, the expression levels of the confirmed iron-uptake genes were upregulated under iron-depleted conditions compared to iron-rich conditions: the expression levels of fecA and fhuA were significantly upregulated (P < 0.01), but those of iutA, iroN1, and iroN2 were not significantly changed (Figure 1).\\n\\nGrowth curves of the WT and mutant strains cultured in LB, iron-depleted M9 and iron-rich M9 media were constructed. The obtained results indicated that the growth rates of the mutant strains (DE205BΔireA, DE205BΔ0007, DE205BΔ0008 and DE205BΔ2235) were comparable to that of the WT strain in LB medium (Figure 2A). Under iron-depleted conditions, the growth rates of the mutant strains were slightly lower than that of the WT strain (Figure 2B). However, under iron-rich conditions, the growth rates of the mutant strains were comparable to that of the WT strain (Figure 2C).\\n\\nMost iron uptake-related genes in DE205BΔireA, including fecA, fhuA, fepA, fepC, feoB, and fyuA, were significantly upregulated (P < 0.05); the expression levels of all these genes were restored to WT levels in the complemented strain DE205BΔireA/ireA* (Figure 3A). In contrast, relative to those of the WT strain, the expression levels of fecA, chuA, and iroN2 were higher only in DE205BΔ0007, DE205BΔ0008, and DE205BΔ2235, respectively (Figures 3B–D).\\n\\nProtein–protein interactions between the four putative TBDRs and TonB were analysed in vitro. The SDS-PAGE results showed that these proteins specifically bound to TonB; further, the Western blot assay results revealed the presence of a His tag on the TBDRs. These findings confirmed that TonB positively interacted with all four putative TBDRs (Figures 4A, B).\\n\\nWe found that at the mid-log phase of growth, Fe uptake by DE205BΔireA (Figure 5A) increased (P < 0.05) compared to that by the WT strain, whereas Fe uptake by DE205BΔtonB (Figure 5B) decreased (P < 0.05). Fe uptake by the complemented strains was restored to normal. There were no significant differences in Fe uptake between the 0007, 0008, and 2235 mutant strains and the WT strain (Figures 5D–F). In addition, compared to the medium for the WT and complemented strains, the medium in which DE205BΔtonB was cultured turned red (Figure 5C).\\n\\nThe expression levels of the genes encoding the TBDRs were examined by qRT-PCR in the WT strain, DE205BΔtonB, and DE205BΔtonB/tonB* under iron-depleted conditions. The ireA expression level in DE205BΔtonB was significantly upregulated (P < 0.01); furthermore, the expression levels of 0007, 0008, and 2235 were slightly upregulated (Figure 6). In addition, we observed that the expression level of the well-known TBDR fepA was significantly increased (P < 0.01), consistent with the results for ireA. Given that the TBDRs are under the control of the ferric uptake regulator fur, the expression levels of fur in the WT strain and DE205BΔtonB under iron-depleted conditions were examined. qRT-PCR showed that relative to that of the WT strain, the expression level of fur in DE205BΔtonB was downregulated (P < 0.01), indicating that the expression of the genes encoding the TBDRs was regulated by fur in the mutant strain DE205BΔtonB (Figure 7).\\n\\n\\nThe expression levels of the genes encoding the four TBDRs in DE205BΔtonB were further tested under iron-rich or iron-depleted conditions using qRT-PCR. The obtained results indicated that the expression level of ireA was upregulated 6.5 times (P < 0.05) under iron-depleted conditions compared to iron-rich conditions and that the expression levels of 0007, 0008, and 2235 were also slightly upregulated (Figure 8). The expression levels of the genes encoding known TBDRs, including fepA, iutA, and feoB, were also significantly increased (P < 0.05).\\n\\nHere, six putative TBDRs were predicted, but deletion or mutant strains could be obtained for only four genes encoding TBDRs (ireA, 0007, 0008, 2235). Motif analyses of ireA sequences have helped reveal the C-terminal region of the TBDRs [12, 13], but experimental verification is lacking. Herein, IreA and TonB were expressed in vitro, and their interaction was determined via GST pulldown assays, which confirmed that ireA is a TBDR. In addition to ireA, three genes (0007, 0008, and 2235) were confirmed to encode TBDRs; however, compared to the expression levels of 0007, 0008, and 2235, that of ireA was the most upregulated in response to iron depletion. Moreover, in the ireA mutant strain DE205BΔireA, iron uptake was increased, consistent with the excessive compensatory expression of other iron uptake-related genes. These findings are similar to those reported for the known TBDR fepA, indicating that ireA might function similarly to fepA [17].\\nTBDRs transport ferric siderophores on the OM with a lack of energy, relying on TonB to transduce energy from the proton motive force of the ExbB–ExbD complex in the inner membrane [36, 37]. In this study, a tonB mutant strain was constructed to analyse the effect of tonB deletion on TBDRs. The obtained results indicated that iron uptake by this mutant strain was significantly decreased; the medium used for culturing DE205BΔtonB turned red as the free iron in the culture medium was oxidized. In addition, the expression level of ireA in DE205BΔtonB was significantly upregulated and that of fur was downregulated. Due to severe iron uptake defects, an iron-depleted environment was created, causing fur to upregulate the expression of the TBDRs in DE205BΔtonB, which implies that the TBDRs are regulated by fur and not directly affected by the presence or absence of tonB.\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 47
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 84,
      "metadata": {
        "tags": [],
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 142
        },
        "id": "3O6acqTbYSsA",
        "outputId": "fb6ce047-4d75-462b-b0b5-8c3edd228e80"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Be aware, overflowing tokens are not returned for the setting you have chosen, i.e. sequence pairs with the 'longest_first' truncation strategy. So the returned list will always be empty even if some tokens have been removed.\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'to the poultry industry [ 【 3 ] . strains of apec and neonatal meningitis - associated e . coli ( nmec , a subpathotype of expec ) , the latter of which causes infections in humans , reportedly share some common virulence genes [ 4 – 6 ] . it is thus particularly important to study the genes encoding virulence factors in apec strains . these strains contain several virulence - associated genes that encode various virulence factors , including adhesins ( fimc , ompa , papc ) , invasins ( ibea ) , avian haemolysins ( hlyf ) , serum survival proteins ( iss , ompt ) , and siderophores ( iuta , fyua , iron ) ; moreover , they show the presence of a pathogenicity island [ 7 – 9 ] . iron is a vital micronutrient that regulates enzyme activity and metabolism . this element plays a key 】 role in basic cellular'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 84
        }
      ],
      "source": [
        "query = 'What is it omphalitis'\n",
        "text = merged.iloc[0].text\n",
        "getAnswer(query, text)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MRuYVm6QYSsA"
      },
      "source": [
        "### Задача №4\n",
        "Соберем весь процесс поиска в одной функции."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 90,
      "metadata": {
        "tags": [],
        "id": "i_JGhNpcYSsA"
      },
      "outputs": [],
      "source": [
        "bm25_index = RankBM25Index(merged)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def searchAndGetAnswer(question, top_bm=50, top_k=10):\n",
        "    assert top_bm > top_k and top_bm > 1 and top_k > 1, 'set top_bm > top_k > 1'\n",
        "    results = bm25_index.search(question, top_bm)\n",
        "    ans_list = []\n",
        "    ans_index = []\n",
        "    for i in range(len(results)):\n",
        "        text = results[i].text()\n",
        "        ans = getAnswer(question, text)\n",
        "        if ans != \"No Answer\":\n",
        "            ans_list.append(ans)\n",
        "            ans_index.append(i)\n",
        "            print(f\"{len(ans_index)}/{top_k}\")\n",
        "            if len(ans_index) >= top_k:\n",
        "                break\n",
        "    if not ans_index:\n",
        "        print(\"No valid answers found.\")\n",
        "        return results.results\n",
        "    final_results = results.results.iloc[ans_index].reset_index(drop=True)\n",
        "    final_results = final_results.copy()\n",
        "    final_results['Answer'] = ans_list\n",
        "    cols = ['title', 'abstract', 'Answer', 'Score', 'url', 'authors']\n",
        "    res = final_results.loc[:, cols]\n",
        "    return res"
      ],
      "metadata": {
        "id": "RuK9BsCELx-Z"
      },
      "execution_count": 91,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "N_SVts_4YSsA"
      },
      "source": [
        "### Q1: *Real-time tracking of whole genomes and a mechanism for coordinating the rapid dissemination of that information to inform the development of diagnostics and therapeutics and to track variations of the virus over time.*"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 92,
      "metadata": {
        "tags": [],
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "OKQxSQejYSsA",
        "outputId": "3f692d85-8812-4cc0-8925-6e9e8163b890"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No valid answers found.\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                title  \\\n",
              "0   Genome Detective Coronavirus Typing Tool for r...   \n",
              "1   Genome Detective Coronavirus Typing Tool for r...   \n",
              "2   Genotyping coronavirus SARS-CoV-2: methods and...   \n",
              "3   Genotyping coronavirus SARS-CoV-2: Methods and...   \n",
              "4   BioLaboro: A bioinformatics system for detecti...   \n",
              "5   MINERVA: A facile strategy for SARS-CoV-2 whol...   \n",
              "6   The ongoing COVID-19 epidemic in Minas Gerais,...   \n",
              "7   Differences in power-law growth over time and ...   \n",
              "8   Infection and Rapid Transmission of SARS-CoV-2...   \n",
              "9   Tracking COVID-19 in Europe: Infodemiology App...   \n",
              "10  Epidemiological data from the COVID-19 outbrea...   \n",
              "11  Current development of COVID-19 diagnostics, v...   \n",
              "12        Short-Term Covid-19 Forecast for Latecomers   \n",
              "13  Research on the Influence of Information Diffu...   \n",
              "14  An Interactive Online Dashboard for Tracking C...   \n",
              "15  A doubt of multiple introduction of SARS‐CoV‐2...   \n",
              "16  Internet Search Patterns Reveal Clinical Cours...   \n",
              "17  Dynamics of transmissible gastroenteritis viru...   \n",
              "18  Building an International Consortium for Track...   \n",
              "19  Artificial Intelligence (AI) applications for ...   \n",
              "20  People Behavior Changes in China during COVID-...   \n",
              "21  Artificial Intelligence (AI) applications for ...   \n",
              "22  Development and Evaluation of A CRISPR-based D...   \n",
              "23  The genomic variation landscape of globally-ci...   \n",
              "24  Diagnosing COVID-19: The Disease and Tools for...   \n",
              "25  Shared Antigen-specific CD8+ T cell Responses ...   \n",
              "26  Spike mutation pipeline reveals the emergence ...   \n",
              "27  Rapid, sensitive, full genome sequencing of Se...   \n",
              "28  Potential Rapid Diagnostics, Vaccine and Thera...   \n",
              "29  COVID-19 pandemic: an overview of epidemiology...   \n",
              "30  Structural and Functional Implications of Non-...   \n",
              "31  Rapid implementation of real-time SARS-CoV-2 s...   \n",
              "32  Testing on the Move South Korea’s rapid respon...   \n",
              "33  Rapid establishment of laboratory diagnostics ...   \n",
              "34  The Viral Infection and Respiratory Illness Un...   \n",
              "35  Failure of the cobas(R) SARS-CoV-2 (Roche) E-g...   \n",
              "36  Characterizing geographical and temporal dynam...   \n",
              "37  Where to buy face masks? Survey of application...   \n",
              "38  Machine learning using intrinsic genomic signa...   \n",
              "39  Racing Towards the Development of Diagnostics ...   \n",
              "40  Evolution and variation of 2019-novel coronavirus   \n",
              "41  Large scale genomic analysis of 3067 SARS-CoV-...   \n",
              "42  Intelligent classification of platelet aggrega...   \n",
              "43  Postintubation Dysphagia During COVID-19 Outbr...   \n",
              "44  Development of a Novel, Genome Subtraction-Der...   \n",
              "45  2019 novel coronavirus is undergoing active re...   \n",
              "46    COVID-19: A promising cure for the global panic   \n",
              "47  Estimation of COVID-19 spread curves integrati...   \n",
              "48  Maximum entropy method for estimating the repr...   \n",
              "49  COVID-19 for the Cardiologist: A Current Revie...   \n",
              "\n",
              "                                             abstract  \\\n",
              "0   SUMMARY: Genome Detective is a web-based, user...   \n",
              "1   Genome Detective is a web-based, user-friendly...   \n",
              "2   The emerging global infectious COVID-19 corona...   \n",
              "3   Abstract The emerging global infectious COVID-...   \n",
              "4   Background Emerging and reemerging infectious ...   \n",
              "5   The novel coronavirus disease 2019 (COVID-19) ...   \n",
              "6   The recent emergence of a previously unknown c...   \n",
              "7   An automated statistical and error analysis of...   \n",
              "8   The outbreak of coronavirus disease 2019 (COVI...   \n",
              "9   BACKGROUND: Infodemiology (ie, information epi...   \n",
              "10  Cases of a novel coronavirus were first report...   \n",
              "11  Abstract A novel coronavirus, designated as SA...   \n",
              "12  The number of Covid-19 cases is increasing dra...   \n",
              "13  With the rapid development of mobile Internet ...   \n",
              "14  OBJECTIVE: To create an online resource that i...   \n",
              "15  The emergence of the novel betacoronavirus, re...   \n",
              "16  Effective public health response to COVID-19 r...   \n",
              "17  Transmissible gastroenteritis virus (TGEV) is ...   \n",
              "18  Information is the most potent protective weap...   \n",
              "19  BACKGROUND AND AIMS: Healthcare delivery requi...   \n",
              "20  While most countries implemented measures to r...   \n",
              "21  Abstract Background and aims Healthcare delive...   \n",
              "22  Background: The recent outbreak of infections ...   \n",
              "23  We describe fifteen major mutation events from...   \n",
              "24  [Image: see text] COVID-19 has spread globally...   \n",
              "25  We report here on antigens from the SARS-CoV-2...   \n",
              "26  We have developed an analysis pipeline to faci...   \n",
              "27  SARS-CoV-2 recently emerged, resulting a globa...   \n",
              "28  Rapid diagnostics, vaccines and therapeutics a...   \n",
              "29  At the time of writing this review, severe acu...   \n",
              "30  SARS-CoV-2, the causative agent of COVID-2019 ...   \n",
              "31  Background The burden and impact of healthcare...   \n",
              "32  Abstract The emergence of COVID-19 in South Ko...   \n",
              "33  The need for timely establishment of diagnosti...   \n",
              "34  The coronavirus disease 2019 pandemic has disp...   \n",
              "35  Control of the ongoing severe acute respirator...   \n",
              "36  We propose an efficient framework for genetic ...   \n",
              "37  The coronavirus disease 2019 (COVID-19) had sp...   \n",
              "38  The 2019 novel coronavirus (renamed SARS-CoV-2...   \n",
              "39                                                NaN   \n",
              "40  The current outbreak caused by novel coronavir...   \n",
              "41  In late December 2019, an emerging viral infec...   \n",
              "42  Platelets are anucleate cells in blood whose p...   \n",
              "43  The COVID-19 is a global pandemic. Its rapid d...   \n",
              "44  The pandemic novel coronavirus infection, Coro...   \n",
              "45  The 2019 novel coronavirus (2019-nCoV) outbrea...   \n",
              "46  Abstract The novel Coronavirus disease 2019 (C...   \n",
              "47  Currently, novel coronavirus disease 2019 (COV...   \n",
              "48  The key parameter that characterizes the trans...   \n",
              "49  Summary The coronavirus disease-2019 (COVID-19...   \n",
              "\n",
              "                                                  url  \\\n",
              "0   https://doi.org/10.1093/bioinformatics/btaa145...   \n",
              "1           https://doi.org/10.1101/2020.01.31.928796   \n",
              "2              https://arxiv.org/pdf/2003.10965v1.pdf   \n",
              "3   https://api.elsevier.com/content/article/pii/S...   \n",
              "4           https://doi.org/10.1101/2020.04.08.031963   \n",
              "5           https://doi.org/10.1101/2020.04.25.060947   \n",
              "6   http://medrxiv.org/cgi/content/short/2020.05.0...   \n",
              "7         https://doi.org/10.1101/2020.03.31.20048827   \n",
              "8   https://www.ncbi.nlm.nih.gov/pubmed/32259477/;...   \n",
              "9   https://www.ncbi.nlm.nih.gov/pubmed/32250957/;...   \n",
              "10  https://www.ncbi.nlm.nih.gov/pubmed/32210236/;...   \n",
              "11  https://doi.org/10.1016/j.micinf.2020.05.001; ...   \n",
              "12             https://arxiv.org/pdf/2004.07977v1.pdf   \n",
              "13  http://medrxiv.org/cgi/content/short/2020.03.3...   \n",
              "14  https://doi.org/10.1093/jamia/ocaa071; https:/...   \n",
              "15  https://doi.org/10.1002/jmv.25773; https://www...   \n",
              "16  http://medrxiv.org/cgi/content/short/2020.05.0...   \n",
              "17  https://doi.org/10.1096/fj.201902455r; https:/...   \n",
              "18        https://doi.org/10.1101/2020.04.02.20051284   \n",
              "19  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   \n",
              "20  http://medrxiv.org/cgi/content/short/2020.05.1...   \n",
              "21  https://api.elsevier.com/content/article/pii/S...   \n",
              "22        https://doi.org/10.1101/2020.02.22.20025460   \n",
              "23          https://doi.org/10.1101/2020.04.21.054221   \n",
              "24  https://doi.org/10.1021/acsnano.0c02624; https...   \n",
              "25  http://medrxiv.org/cgi/content/short/2020.05.0...   \n",
              "26          https://doi.org/10.1101/2020.04.29.069054   \n",
              "27          https://doi.org/10.1101/2020.04.22.055897   \n",
              "28  https://doi.org/10.3390/jcm9030623; https://ww...   \n",
              "29  https://www.ncbi.nlm.nih.gov/pubmed/32397911/;...   \n",
              "30          https://doi.org/10.1101/2020.05.02.071811   \n",
              "31  http://medrxiv.org/cgi/content/short/2020.05.0...   \n",
              "32  https://api.elsevier.com/content/article/pii/S...   \n",
              "33  https://www.ncbi.nlm.nih.gov/pubmed/32156330/;...   \n",
              "34  https://www.ncbi.nlm.nih.gov/pubmed/32426754/;...   \n",
              "35  http://medrxiv.org/cgi/content/short/2020.04.2...   \n",
              "36          https://doi.org/10.1101/2020.04.07.030759   \n",
              "37  https://doi.org/10.1097/jcma.0000000000000325;...   \n",
              "38  https://doi.org/10.1371/journal.pone.0232391; ...   \n",
              "39  https://doi.org/10.1093/clinchem/hvaa038; http...   \n",
              "40          https://doi.org/10.1101/2020.01.30.926477   \n",
              "41          https://doi.org/10.1101/2020.05.03.074567   \n",
              "42  https://doi.org/10.7554/elife.52938; https://w...   \n",
              "43  https://doi.org/10.1007/s00455-020-10139-6; ht...   \n",
              "44  https://www.ncbi.nlm.nih.gov/pubmed/32276333/;...   \n",
              "45  https://doi.org/10.1093/cid/ciaa219; https://w...   \n",
              "46  https://www.ncbi.nlm.nih.gov/pubmed/32278175/;...   \n",
              "47             https://arxiv.org/pdf/2005.00662v3.pdf   \n",
              "48        https://doi.org/10.1101/2020.03.14.20035659   \n",
              "49  https://www.sciencedirect.com/science/article/...   \n",
              "\n",
              "                                              authors  \\\n",
              "0   Cleemput, Sara; Dumon, Wim; Fonseca, Vagner; K...   \n",
              "1   Cleemput, Sara; Dumon, Wim; Fonseca, Vagner; A...   \n",
              "2                                     Yin, Changchuan   \n",
              "3                                     Yin, Changchuan   \n",
              "4   Holland, Mitchell; Negrón, Daniel; Mitchell, S...   \n",
              "5   Chen, Chen; Li, Jizhou; Di, Lin; Jing, Qiuyu; ...   \n",
              "6   Xavier, J.; Giovanetti, M.; Adelino, T.; Fonse...   \n",
              "7                                        Merrin, Jack   \n",
              "8   Kim, Young-Il; Kim, Seong-Gyu; Kim, Se-Mi; Kim...   \n",
              "9                                Mavragani, Amaryllis   \n",
              "10  Xu, Bo; Gutierrez, Bernardo; Mekaru, Sumiko; S...   \n",
              "11  Zhang, Naru; Li, Chaoqun; Hu, Yue; Li, Kangche...   \n",
              "12  Medeiros, Marcelo; Street, Alexandre; Valladao...   \n",
              "13  Shanlang, L.; Chao, M.; Ruofei, L.; Junpei, H....   \n",
              "14  Wissel, Benjamin D; Van Camp, P J; Kouril, Mic...   \n",
              "15  Giovanetti, Marta; Angeletti, Silvia; Benvenut...   \n",
              "16                                Lu, T.; Reis, B. Y.   \n",
              "17   Wang, Jian; Li, Yangyang; Wang, Shouyu; Liu, Fei   \n",
              "18  Segal, Eran; Zhang, Feng; Lin, Xihong; King, G...   \n",
              "19  Vaishya, Raju; Javaid, Mohd; Khan, Ibrahim Hal...   \n",
              "20               Wu, Y.; Huang, M.; Xie, G.; Chen, X.   \n",
              "21  Vaishya, Raju; Javaid, Mohd; Khan, Ibrahim Hal...   \n",
              "22  Hou, Tieying; Zeng, Weiqi; Yang, Minling; Chen...   \n",
              "23  Guan, Qingtian; Sadykov, Mukhtar; Nugmanova, R...   \n",
              "24  Udugama, Buddhisha; Kadhiresan, Pranav; Kozlow...   \n",
              "25  Chour, W.; Xu, A. M.; Ng, A. H. C.; Choi, J.; ...   \n",
              "26  Korber, B; Fischer, WM; Gnanakaran, S; Yoon, H...   \n",
              "27  Paden, Clinton R; Tao, Ying; Queen, Krista; Zh...   \n",
              "28  Pang, Junxiong; Wang, Min Xian; Ang, Ian Yi Ha...   \n",
              "29  Amawi, Haneen; Abu Deiab, Ghina'a I; A Aljabal...   \n",
              "30  Nelson-Sathi, Shijulal; Umasankar, PK; Sreekum...   \n",
              "31  Meredith, L. W.; Hamilton, W. L.; Warne, B.; H...   \n",
              "32                           Lee, David; Lee, Jaehong   \n",
              "33  Konrad, Regina; Eberle, Ute; Dangel, Alexandra...   \n",
              "34  Walkey, Allan J.; Kumar, Vishakha K.; Harhay, ...   \n",
              "35  Artesi, M.; Bontems, S.; Gobbels, P.; Franckh,...   \n",
              "36  Zhao, Zhengqiao; Sokhansanj, Bahrad A.; Rosen,...   \n",
              "37  Yuan, Eunice J.; Hsu, Chia-An; Lee, Wui-Chiang...   \n",
              "38  Randhawa, Gurjit S.; Soltysiak, Maximillian P....   \n",
              "39                    Lo, Y M Dennis; Chiu, Rossa W K   \n",
              "40  Xiong, Chenglong; Jiang, Lufang; Chen, Yue; Ji...   \n",
              "41  Laamarti, Meriem; Alouane, Tarek; Kartti, Soua...   \n",
              "42  Zhou, Yuqi; Yasumoto, Atsushi; Lei, Cheng; Hua...   \n",
              "43  Frajkova, Zofia; Tedla, Miroslav; Tedlova, Eva...   \n",
              "44  Yip, Cyril Chik-Yan; Ho, Chi-Chun; Chan, Jaspe...   \n",
              "45                                       Yi, Huiguang   \n",
              "46  Vellingiri, Balachandar; Jayaramayya, Kaavya; ...   \n",
              "47         Lee, Se Yoon; Lei, Bowen; Mallick, Bani K.   \n",
              "48                                          Tao, Yong   \n",
              "49  Atri, Deepak; Siddiqi, Hasan K.; Lang, Joshua;...   \n",
              "\n",
              "                                                 text      Score  \n",
              "0   We are currently faced with a potential global...  22.262102  \n",
              "1   We are currently faced with a potential global...  21.113115  \n",
              "2   The novel coronavirus in humans, first discove...  20.446664  \n",
              "3   through nsp16 in all coronaviruses [8] . There...  20.102274  \n",
              "4   Using next generation sequencing, the whole ge...  18.086823  \n",
              "5   As of May 22, 2020, the ongoing COVID-19 viral...  16.010989  \n",
              "6   To date, more than 3.5 million cases of the di...  15.469203  \n",
              "7   Mathematics is essential to predict and contro...  15.163071  \n",
              "8   Coronaviruses (CoVs) are a large family of vir...  15.076527  \n",
              "9   In December 2019, Chinese researchers identifi...  14.799094  \n",
              "10  In December 2019 a number of novel coronavirus...  14.789665  \n",
              "11  In late December 2019, a pneumonia of unknown ...  14.670619  \n",
              "12  Being able to forecast accurately the number o...  14.261787  \n",
              "13  difficulty of epidemic prevention and control....  14.242511  \n",
              "14  As of April 13 th , the United States of Ameri...  14.233459  \n",
              "15  Coronaviruses (CoVs) are enveloped, positive‐s...  14.206456  \n",
              "16  Accurate real-time surveillance of local disea...  13.965029  \n",
              "17  Transmissible gastroenteritis virus (TGEV) is ...  13.855779  \n",
              "18  The CCC is a non-profit consortium open to any...  13.712987  \n",
              "19  In this worldwide health crisis, the medical i...  13.702983  \n",
              "20  As of 10 May 2020, there has been more than 4 ...  13.655200  \n",
              "21  In this worldwide health crisis, the medical i...  13.633815  \n",
              "22  Since the beginning of 2020, a surging number ...  13.571718  \n",
              "23  harbouring V483A and G476S mutations belong to...  13.516136  \n",
              "24  SARS-CoV-2 was first identified from patient s...  13.408559  \n",
              "25  In December 2019, an outbreak of unexplained p...  13.335027  \n",
              "26  KEYWORDS: COVID-19, pandemic, diversity, evolu...  13.234609  \n",
              "27  In December 2019, SARS-CoV-2, the etiological ...  13.217456  \n",
              "28  Since mid-December 2019 and as of early Februa...  13.172897  \n",
              "29  Coronaviruses (CoVs) belong to the genus Coron...  13.058392  \n",
              "30  COVID-19, the highly transmittable and pathoge...  12.998782  \n",
              "31  SARS-CoV-2 emerged in the human population in ...  12.910279  \n",
              "32  In early 2020, the world watched as a novel co...  12.835060  \n",
              "33  In connection with the ongoing outbreak of a n...  12.596926  \n",
              "34  The SCCM formed the Discovery, the Critical Ca...  12.544639  \n",
              "35  Coronavirus disease of 2019 (COVID-19) origina...  12.542551  \n",
              "36  For the aligned sequences, we merged the seque...  12.423850  \n",
              "37  The world was introduced to the coronavirus di...  12.373256  \n",
              "38  Coronaviruses are single-stranded positive-sen...  12.351881  \n",
              "39  In December 2019, a mysterious viral illness c...  12.340248  \n",
              "40  Coronaviruses (CoV) (order Nidovirales, family...  12.240991  \n",
              "41  We used Hyphy v2. 5.8 (13) to estimate synonym...  12.214958  \n",
              "42  Platelets are non-nucleated cells in blood who...  12.188046  \n",
              "43  The COVID-19 disease spread from Wuhan, China,...  12.177250  \n",
              "44  Coronaviruses are positive sense, single-stran...  12.141886  \n",
              "45  (shown as a pie chart node in Fig. 1 , see Tab...  12.099818  \n",
              "46  Most of the species under this head are enzoot...  12.090880  \n",
              "47  Since Thursday, March 26, 2020, the US leads t...  12.068897  \n",
              "48  In December, 2019, a cluster of pneumonia case...  12.053707  \n",
              "49  The coronavirus disease-2019 (COVID- 19) , a c...  12.039220  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-8e4f4aba-34bb-4b8e-9cc0-d6327a4104f8\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>title</th>\n",
              "      <th>abstract</th>\n",
              "      <th>url</th>\n",
              "      <th>authors</th>\n",
              "      <th>text</th>\n",
              "      <th>Score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Genome Detective Coronavirus Typing Tool for r...</td>\n",
              "      <td>SUMMARY: Genome Detective is a web-based, user...</td>\n",
              "      <td>https://doi.org/10.1093/bioinformatics/btaa145...</td>\n",
              "      <td>Cleemput, Sara; Dumon, Wim; Fonseca, Vagner; K...</td>\n",
              "      <td>We are currently faced with a potential global...</td>\n",
              "      <td>22.262102</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Genome Detective Coronavirus Typing Tool for r...</td>\n",
              "      <td>Genome Detective is a web-based, user-friendly...</td>\n",
              "      <td>https://doi.org/10.1101/2020.01.31.928796</td>\n",
              "      <td>Cleemput, Sara; Dumon, Wim; Fonseca, Vagner; A...</td>\n",
              "      <td>We are currently faced with a potential global...</td>\n",
              "      <td>21.113115</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Genotyping coronavirus SARS-CoV-2: methods and...</td>\n",
              "      <td>The emerging global infectious COVID-19 corona...</td>\n",
              "      <td>https://arxiv.org/pdf/2003.10965v1.pdf</td>\n",
              "      <td>Yin, Changchuan</td>\n",
              "      <td>The novel coronavirus in humans, first discove...</td>\n",
              "      <td>20.446664</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Genotyping coronavirus SARS-CoV-2: Methods and...</td>\n",
              "      <td>Abstract The emerging global infectious COVID-...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Yin, Changchuan</td>\n",
              "      <td>through nsp16 in all coronaviruses [8] . There...</td>\n",
              "      <td>20.102274</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>BioLaboro: A bioinformatics system for detecti...</td>\n",
              "      <td>Background Emerging and reemerging infectious ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.08.031963</td>\n",
              "      <td>Holland, Mitchell; Negrón, Daniel; Mitchell, S...</td>\n",
              "      <td>Using next generation sequencing, the whole ge...</td>\n",
              "      <td>18.086823</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>MINERVA: A facile strategy for SARS-CoV-2 whol...</td>\n",
              "      <td>The novel coronavirus disease 2019 (COVID-19) ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.25.060947</td>\n",
              "      <td>Chen, Chen; Li, Jizhou; Di, Lin; Jing, Qiuyu; ...</td>\n",
              "      <td>As of May 22, 2020, the ongoing COVID-19 viral...</td>\n",
              "      <td>16.010989</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>The ongoing COVID-19 epidemic in Minas Gerais,...</td>\n",
              "      <td>The recent emergence of a previously unknown c...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.0...</td>\n",
              "      <td>Xavier, J.; Giovanetti, M.; Adelino, T.; Fonse...</td>\n",
              "      <td>To date, more than 3.5 million cases of the di...</td>\n",
              "      <td>15.469203</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Differences in power-law growth over time and ...</td>\n",
              "      <td>An automated statistical and error analysis of...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.31.20048827</td>\n",
              "      <td>Merrin, Jack</td>\n",
              "      <td>Mathematics is essential to predict and contro...</td>\n",
              "      <td>15.163071</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>Infection and Rapid Transmission of SARS-CoV-2...</td>\n",
              "      <td>The outbreak of coronavirus disease 2019 (COVI...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32259477/;...</td>\n",
              "      <td>Kim, Young-Il; Kim, Seong-Gyu; Kim, Se-Mi; Kim...</td>\n",
              "      <td>Coronaviruses (CoVs) are a large family of vir...</td>\n",
              "      <td>15.076527</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>Tracking COVID-19 in Europe: Infodemiology App...</td>\n",
              "      <td>BACKGROUND: Infodemiology (ie, information epi...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32250957/;...</td>\n",
              "      <td>Mavragani, Amaryllis</td>\n",
              "      <td>In December 2019, Chinese researchers identifi...</td>\n",
              "      <td>14.799094</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>Epidemiological data from the COVID-19 outbrea...</td>\n",
              "      <td>Cases of a novel coronavirus were first report...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32210236/;...</td>\n",
              "      <td>Xu, Bo; Gutierrez, Bernardo; Mekaru, Sumiko; S...</td>\n",
              "      <td>In December 2019 a number of novel coronavirus...</td>\n",
              "      <td>14.789665</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>Current development of COVID-19 diagnostics, v...</td>\n",
              "      <td>Abstract A novel coronavirus, designated as SA...</td>\n",
              "      <td>https://doi.org/10.1016/j.micinf.2020.05.001; ...</td>\n",
              "      <td>Zhang, Naru; Li, Chaoqun; Hu, Yue; Li, Kangche...</td>\n",
              "      <td>In late December 2019, a pneumonia of unknown ...</td>\n",
              "      <td>14.670619</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>Short-Term Covid-19 Forecast for Latecomers</td>\n",
              "      <td>The number of Covid-19 cases is increasing dra...</td>\n",
              "      <td>https://arxiv.org/pdf/2004.07977v1.pdf</td>\n",
              "      <td>Medeiros, Marcelo; Street, Alexandre; Valladao...</td>\n",
              "      <td>Being able to forecast accurately the number o...</td>\n",
              "      <td>14.261787</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>Research on the Influence of Information Diffu...</td>\n",
              "      <td>With the rapid development of mobile Internet ...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.03.3...</td>\n",
              "      <td>Shanlang, L.; Chao, M.; Ruofei, L.; Junpei, H....</td>\n",
              "      <td>difficulty of epidemic prevention and control....</td>\n",
              "      <td>14.242511</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>An Interactive Online Dashboard for Tracking C...</td>\n",
              "      <td>OBJECTIVE: To create an online resource that i...</td>\n",
              "      <td>https://doi.org/10.1093/jamia/ocaa071; https:/...</td>\n",
              "      <td>Wissel, Benjamin D; Van Camp, P J; Kouril, Mic...</td>\n",
              "      <td>As of April 13 th , the United States of Ameri...</td>\n",
              "      <td>14.233459</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>A doubt of multiple introduction of SARS‐CoV‐2...</td>\n",
              "      <td>The emergence of the novel betacoronavirus, re...</td>\n",
              "      <td>https://doi.org/10.1002/jmv.25773; https://www...</td>\n",
              "      <td>Giovanetti, Marta; Angeletti, Silvia; Benvenut...</td>\n",
              "      <td>Coronaviruses (CoVs) are enveloped, positive‐s...</td>\n",
              "      <td>14.206456</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>Internet Search Patterns Reveal Clinical Cours...</td>\n",
              "      <td>Effective public health response to COVID-19 r...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.0...</td>\n",
              "      <td>Lu, T.; Reis, B. Y.</td>\n",
              "      <td>Accurate real-time surveillance of local disea...</td>\n",
              "      <td>13.965029</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>Dynamics of transmissible gastroenteritis viru...</td>\n",
              "      <td>Transmissible gastroenteritis virus (TGEV) is ...</td>\n",
              "      <td>https://doi.org/10.1096/fj.201902455r; https:/...</td>\n",
              "      <td>Wang, Jian; Li, Yangyang; Wang, Shouyu; Liu, Fei</td>\n",
              "      <td>Transmissible gastroenteritis virus (TGEV) is ...</td>\n",
              "      <td>13.855779</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>Building an International Consortium for Track...</td>\n",
              "      <td>Information is the most potent protective weap...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.02.20051284</td>\n",
              "      <td>Segal, Eran; Zhang, Feng; Lin, Xihong; King, G...</td>\n",
              "      <td>The CCC is a non-profit consortium open to any...</td>\n",
              "      <td>13.712987</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>Artificial Intelligence (AI) applications for ...</td>\n",
              "      <td>BACKGROUND AND AIMS: Healthcare delivery requi...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>Vaishya, Raju; Javaid, Mohd; Khan, Ibrahim Hal...</td>\n",
              "      <td>In this worldwide health crisis, the medical i...</td>\n",
              "      <td>13.702983</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>People Behavior Changes in China during COVID-...</td>\n",
              "      <td>While most countries implemented measures to r...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.1...</td>\n",
              "      <td>Wu, Y.; Huang, M.; Xie, G.; Chen, X.</td>\n",
              "      <td>As of 10 May 2020, there has been more than 4 ...</td>\n",
              "      <td>13.655200</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>Artificial Intelligence (AI) applications for ...</td>\n",
              "      <td>Abstract Background and aims Healthcare delive...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Vaishya, Raju; Javaid, Mohd; Khan, Ibrahim Hal...</td>\n",
              "      <td>In this worldwide health crisis, the medical i...</td>\n",
              "      <td>13.633815</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>Development and Evaluation of A CRISPR-based D...</td>\n",
              "      <td>Background: The recent outbreak of infections ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.02.22.20025460</td>\n",
              "      <td>Hou, Tieying; Zeng, Weiqi; Yang, Minling; Chen...</td>\n",
              "      <td>Since the beginning of 2020, a surging number ...</td>\n",
              "      <td>13.571718</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>The genomic variation landscape of globally-ci...</td>\n",
              "      <td>We describe fifteen major mutation events from...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.21.054221</td>\n",
              "      <td>Guan, Qingtian; Sadykov, Mukhtar; Nugmanova, R...</td>\n",
              "      <td>harbouring V483A and G476S mutations belong to...</td>\n",
              "      <td>13.516136</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>Diagnosing COVID-19: The Disease and Tools for...</td>\n",
              "      <td>[Image: see text] COVID-19 has spread globally...</td>\n",
              "      <td>https://doi.org/10.1021/acsnano.0c02624; https...</td>\n",
              "      <td>Udugama, Buddhisha; Kadhiresan, Pranav; Kozlow...</td>\n",
              "      <td>SARS-CoV-2 was first identified from patient s...</td>\n",
              "      <td>13.408559</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>Shared Antigen-specific CD8+ T cell Responses ...</td>\n",
              "      <td>We report here on antigens from the SARS-CoV-2...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.0...</td>\n",
              "      <td>Chour, W.; Xu, A. M.; Ng, A. H. C.; Choi, J.; ...</td>\n",
              "      <td>In December 2019, an outbreak of unexplained p...</td>\n",
              "      <td>13.335027</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>Spike mutation pipeline reveals the emergence ...</td>\n",
              "      <td>We have developed an analysis pipeline to faci...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.29.069054</td>\n",
              "      <td>Korber, B; Fischer, WM; Gnanakaran, S; Yoon, H...</td>\n",
              "      <td>KEYWORDS: COVID-19, pandemic, diversity, evolu...</td>\n",
              "      <td>13.234609</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>Rapid, sensitive, full genome sequencing of Se...</td>\n",
              "      <td>SARS-CoV-2 recently emerged, resulting a globa...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.22.055897</td>\n",
              "      <td>Paden, Clinton R; Tao, Ying; Queen, Krista; Zh...</td>\n",
              "      <td>In December 2019, SARS-CoV-2, the etiological ...</td>\n",
              "      <td>13.217456</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>Potential Rapid Diagnostics, Vaccine and Thera...</td>\n",
              "      <td>Rapid diagnostics, vaccines and therapeutics a...</td>\n",
              "      <td>https://doi.org/10.3390/jcm9030623; https://ww...</td>\n",
              "      <td>Pang, Junxiong; Wang, Min Xian; Ang, Ian Yi Ha...</td>\n",
              "      <td>Since mid-December 2019 and as of early Februa...</td>\n",
              "      <td>13.172897</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>COVID-19 pandemic: an overview of epidemiology...</td>\n",
              "      <td>At the time of writing this review, severe acu...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32397911/;...</td>\n",
              "      <td>Amawi, Haneen; Abu Deiab, Ghina'a I; A Aljabal...</td>\n",
              "      <td>Coronaviruses (CoVs) belong to the genus Coron...</td>\n",
              "      <td>13.058392</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>Structural and Functional Implications of Non-...</td>\n",
              "      <td>SARS-CoV-2, the causative agent of COVID-2019 ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.02.071811</td>\n",
              "      <td>Nelson-Sathi, Shijulal; Umasankar, PK; Sreekum...</td>\n",
              "      <td>COVID-19, the highly transmittable and pathoge...</td>\n",
              "      <td>12.998782</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>Rapid implementation of real-time SARS-CoV-2 s...</td>\n",
              "      <td>Background The burden and impact of healthcare...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.0...</td>\n",
              "      <td>Meredith, L. W.; Hamilton, W. L.; Warne, B.; H...</td>\n",
              "      <td>SARS-CoV-2 emerged in the human population in ...</td>\n",
              "      <td>12.910279</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>Testing on the Move South Korea’s rapid respon...</td>\n",
              "      <td>Abstract The emergence of COVID-19 in South Ko...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Lee, David; Lee, Jaehong</td>\n",
              "      <td>In early 2020, the world watched as a novel co...</td>\n",
              "      <td>12.835060</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>Rapid establishment of laboratory diagnostics ...</td>\n",
              "      <td>The need for timely establishment of diagnosti...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32156330/;...</td>\n",
              "      <td>Konrad, Regina; Eberle, Ute; Dangel, Alexandra...</td>\n",
              "      <td>In connection with the ongoing outbreak of a n...</td>\n",
              "      <td>12.596926</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>The Viral Infection and Respiratory Illness Un...</td>\n",
              "      <td>The coronavirus disease 2019 pandemic has disp...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32426754/;...</td>\n",
              "      <td>Walkey, Allan J.; Kumar, Vishakha K.; Harhay, ...</td>\n",
              "      <td>The SCCM formed the Discovery, the Critical Ca...</td>\n",
              "      <td>12.544639</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>Failure of the cobas(R) SARS-CoV-2 (Roche) E-g...</td>\n",
              "      <td>Control of the ongoing severe acute respirator...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.2...</td>\n",
              "      <td>Artesi, M.; Bontems, S.; Gobbels, P.; Franckh,...</td>\n",
              "      <td>Coronavirus disease of 2019 (COVID-19) origina...</td>\n",
              "      <td>12.542551</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>Characterizing geographical and temporal dynam...</td>\n",
              "      <td>We propose an efficient framework for genetic ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.07.030759</td>\n",
              "      <td>Zhao, Zhengqiao; Sokhansanj, Bahrad A.; Rosen,...</td>\n",
              "      <td>For the aligned sequences, we merged the seque...</td>\n",
              "      <td>12.423850</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>Where to buy face masks? Survey of application...</td>\n",
              "      <td>The coronavirus disease 2019 (COVID-19) had sp...</td>\n",
              "      <td>https://doi.org/10.1097/jcma.0000000000000325;...</td>\n",
              "      <td>Yuan, Eunice J.; Hsu, Chia-An; Lee, Wui-Chiang...</td>\n",
              "      <td>The world was introduced to the coronavirus di...</td>\n",
              "      <td>12.373256</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>Machine learning using intrinsic genomic signa...</td>\n",
              "      <td>The 2019 novel coronavirus (renamed SARS-CoV-2...</td>\n",
              "      <td>https://doi.org/10.1371/journal.pone.0232391; ...</td>\n",
              "      <td>Randhawa, Gurjit S.; Soltysiak, Maximillian P....</td>\n",
              "      <td>Coronaviruses are single-stranded positive-sen...</td>\n",
              "      <td>12.351881</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>Racing Towards the Development of Diagnostics ...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://doi.org/10.1093/clinchem/hvaa038; http...</td>\n",
              "      <td>Lo, Y M Dennis; Chiu, Rossa W K</td>\n",
              "      <td>In December 2019, a mysterious viral illness c...</td>\n",
              "      <td>12.340248</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>Evolution and variation of 2019-novel coronavirus</td>\n",
              "      <td>The current outbreak caused by novel coronavir...</td>\n",
              "      <td>https://doi.org/10.1101/2020.01.30.926477</td>\n",
              "      <td>Xiong, Chenglong; Jiang, Lufang; Chen, Yue; Ji...</td>\n",
              "      <td>Coronaviruses (CoV) (order Nidovirales, family...</td>\n",
              "      <td>12.240991</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>Large scale genomic analysis of 3067 SARS-CoV-...</td>\n",
              "      <td>In late December 2019, an emerging viral infec...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.03.074567</td>\n",
              "      <td>Laamarti, Meriem; Alouane, Tarek; Kartti, Soua...</td>\n",
              "      <td>We used Hyphy v2. 5.8 (13) to estimate synonym...</td>\n",
              "      <td>12.214958</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>42</th>\n",
              "      <td>Intelligent classification of platelet aggrega...</td>\n",
              "      <td>Platelets are anucleate cells in blood whose p...</td>\n",
              "      <td>https://doi.org/10.7554/elife.52938; https://w...</td>\n",
              "      <td>Zhou, Yuqi; Yasumoto, Atsushi; Lei, Cheng; Hua...</td>\n",
              "      <td>Platelets are non-nucleated cells in blood who...</td>\n",
              "      <td>12.188046</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>Postintubation Dysphagia During COVID-19 Outbr...</td>\n",
              "      <td>The COVID-19 is a global pandemic. Its rapid d...</td>\n",
              "      <td>https://doi.org/10.1007/s00455-020-10139-6; ht...</td>\n",
              "      <td>Frajkova, Zofia; Tedla, Miroslav; Tedlova, Eva...</td>\n",
              "      <td>The COVID-19 disease spread from Wuhan, China,...</td>\n",
              "      <td>12.177250</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>Development of a Novel, Genome Subtraction-Der...</td>\n",
              "      <td>The pandemic novel coronavirus infection, Coro...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32276333/;...</td>\n",
              "      <td>Yip, Cyril Chik-Yan; Ho, Chi-Chun; Chan, Jaspe...</td>\n",
              "      <td>Coronaviruses are positive sense, single-stran...</td>\n",
              "      <td>12.141886</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>2019 novel coronavirus is undergoing active re...</td>\n",
              "      <td>The 2019 novel coronavirus (2019-nCoV) outbrea...</td>\n",
              "      <td>https://doi.org/10.1093/cid/ciaa219; https://w...</td>\n",
              "      <td>Yi, Huiguang</td>\n",
              "      <td>(shown as a pie chart node in Fig. 1 , see Tab...</td>\n",
              "      <td>12.099818</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>COVID-19: A promising cure for the global panic</td>\n",
              "      <td>Abstract The novel Coronavirus disease 2019 (C...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32278175/;...</td>\n",
              "      <td>Vellingiri, Balachandar; Jayaramayya, Kaavya; ...</td>\n",
              "      <td>Most of the species under this head are enzoot...</td>\n",
              "      <td>12.090880</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>47</th>\n",
              "      <td>Estimation of COVID-19 spread curves integrati...</td>\n",
              "      <td>Currently, novel coronavirus disease 2019 (COV...</td>\n",
              "      <td>https://arxiv.org/pdf/2005.00662v3.pdf</td>\n",
              "      <td>Lee, Se Yoon; Lei, Bowen; Mallick, Bani K.</td>\n",
              "      <td>Since Thursday, March 26, 2020, the US leads t...</td>\n",
              "      <td>12.068897</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>48</th>\n",
              "      <td>Maximum entropy method for estimating the repr...</td>\n",
              "      <td>The key parameter that characterizes the trans...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.14.20035659</td>\n",
              "      <td>Tao, Yong</td>\n",
              "      <td>In December, 2019, a cluster of pneumonia case...</td>\n",
              "      <td>12.053707</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>49</th>\n",
              "      <td>COVID-19 for the Cardiologist: A Current Revie...</td>\n",
              "      <td>Summary The coronavirus disease-2019 (COVID-19...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Atri, Deepak; Siddiqi, Hasan K.; Lang, Joshua;...</td>\n",
              "      <td>The coronavirus disease-2019 (COVID- 19) , a c...</td>\n",
              "      <td>12.039220</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8e4f4aba-34bb-4b8e-9cc0-d6327a4104f8')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-8e4f4aba-34bb-4b8e-9cc0-d6327a4104f8 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-8e4f4aba-34bb-4b8e-9cc0-d6327a4104f8');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-bdf3563d-c013-4ecd-a6d3-97beb7ce2ae6\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-bdf3563d-c013-4ecd-a6d3-97beb7ce2ae6')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-bdf3563d-c013-4ecd-a6d3-97beb7ce2ae6 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "res",
              "summary": "{\n  \"name\": \"res\",\n  \"rows\": 50,\n  \"fields\": [\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 48,\n        \"samples\": [\n          \"COVID-19 pandemic: an overview of epidemiology, parthenogenesis, diagnostics and potential vaccines and therapeutics\",\n          \"Intelligent classification of platelet aggregates by agonist type\",\n          \"Potential Rapid Diagnostics, Vaccine and Therapeutics for 2019 Novel Coronavirus (2019-nCoV): A Systematic Review\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 49,\n        \"samples\": [\n          \"With the rapid development of mobile Internet in China, the information of the epidemic is full-time and holographic, and the role of information diffusion in epidemic control is increasingly prominent. At the same time, the publicity of all kinds of big data also provides the possibility to explore the impact of media information diffusion on disease transmission. This paper explores the mechanism of the influence of information diffusion on the spread of the novel coronavirus, develops a model of the interaction between information diffusion and disease transmission based on the SIR model, and empirically tests the role and mechanism of information diffusion in the spread of COCID-19 by using econometric method. The result shows that there was a significant negative correlation between the information diffusion and the spread of the novel coronavirus; The result of robust test shows that the spread of both epidemic information and protection information hindered the further spread of the epidemic.\",\n          \"Abstract The novel Coronavirus disease 2019 (COVID-19) is caused by SARS-CoV-2, which is the causative agent of a potentially fatal disease that is of great global public health concern. The outbreak of COVID-19 is wreaking havoc worldwide due to inadequate risk assessment regarding the urgency of the situation. The COVID-19 pandemic has entered a dangerous new phase. When compared with SARS and MERS, COVID-19 has spread more rapidly, due to increased globalization and adaptation of the virus in every environment. Slowing the spread of the COVID-19 cases will significantly reduce the strain on the healthcare system of the country by limiting the number of people who are severely sick by COVID-19 and need hospital care. Hence, the recent outburst of COVID-19 highlights an urgent need for therapeutics targeting SARS-CoV-2. Here, we have discussed the structure of virus; varying symptoms among COVID-19, SARS, MERS and common flu; the probable mechanism behind the infection and its immune response. Further, the current treatment options, drugs available, ongoing trials and recent diagnostics for COVID-19 have been discussed. We suggest traditional Indian medicinal plants as possible novel therapeutic approaches, exclusively targeting SARS-CoV-2 and its pathways.\",\n          \"The key parameter that characterizes the transmissibility of a disease is the reproduction number R. If it exceeds 1, the number of incident cases will inevitably grow over time, and a large epidemic is possible. To prevent the expansion of an epidemic, R must be reduced to a level below 1. To estimate the reproduction number, the probability distribution function of the generation interval of an infectious disease is required to be available; however, this distribution is often unknown. In this letter, given the incomplete information for the generation interval, we propose a maximum entropy method to estimate the reproduction number. Based on this method, given the mean value and variance of the generation interval, we first determine its probability distribution function and in turn estimate the real-time values of reproduction number of COVID-19 in China. By applying these estimated reproduction numbers into the susceptible-infectious-removed epidemic model, we simulate the evolutionary track of the epidemic in China, which is well in accordance with that of the real incident cases. The simulation results predict that China's epidemic will gradually tend to disappear by May 2020 if the quarantine measures can continue to be executed.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"http://medrxiv.org/cgi/content/short/2020.03.31.20048439v1?rss=1\",\n          \"https://doi.org/10.1093/clinchem/hvaa038; https://www.ncbi.nlm.nih.gov/pubmed/32031590/\",\n          \"https://doi.org/10.1101/2020.05.02.071811\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 48,\n        \"samples\": [\n          \"Amawi, Haneen; Abu Deiab, Ghina'a I; A Aljabali, Alaa A; Dua, Kamal; Tambuwala, Murtaza M\",\n          \"Zhou, Yuqi; Yasumoto, Atsushi; Lei, Cheng; Huang, Chun-Jung; Kobayashi, Hirofumi; Wu, Yunzhao; Yan, Sheng; Sun, Chia-Wei; Yatomi, Yutaka; Goda, Keisuke\",\n          \"Pang, Junxiong; Wang, Min Xian; Ang, Ian Yi Han; Tan, Sharon Hui Xuan; Lewis, Ruth Frances; Chen, Jacinta I-Pei; Gutierrez, Ramona A; Gwee, Sylvia Xiao Wei; Chua, Pearleen Ee Yong; Yang, Qian; Ng, Xian Yi; Yap, Rowena K.S.; Tan, Hao Yi; Teo, Yik Ying; Tan, Chorh Chuan; Cook, Alex R.; Yap, Jason Chin-Huat; Hsu, Li Yang\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"difficulty of epidemic prevention and control. In order to prevent the spread of the epidemic, the central government has taken unprecedented prevention and control measures, including setting up designated admission hospitals, expanding the supply of beds in the hospital, coordinating the dispatch of medical prevention and control materials, extending the Spring Festival holiday, implementing peak staggering return, measuring the temperature of vehicles and stations, disinfecting, ventilating, etc. The National Health Commission has also sent a number of supervision teams to hospitals and disease control agencies to conduct on-site supervision. Wuhan also announced the closure of the city on January 23, suspending urban public transport and strictly controlling the access of people inside and outside Wuhan. Subsequently, 31 provinces, regions and cities in the country successively launched the level 1 emergency response to public health emergencies, strictly controlled the transmission of the virus, and made every effort to prevent the further spread of the epidemic. At the early stage of the epidemic (December 31 to January 20), due to the lack of public reports in Wuhan's official media, the information of the epidemic was mainly disclosed by the media, so the epidemic was not paid attention by the public, which delayed the best time to expand the social impact, thus leading to the further spread of the epidemic. After January 20, with the outbreak of the epidemic and the release of information, COVID-19 epidemic information became the most concerned information of the public, and media reports entered a white-hot stage. The emergence of the novel coronavirus pneumonia has also caused widespread panic among the public. The official media, micro-blog, WeChat and other media have followed up the reports of real-time epidemic, new symptoms and prevention measures, and timely conveyed clear and positive information to the society and advised the public to protect themselves and to view the epidemic more objectively and objectively. Will the information diffusion help to eliminate rumors and guide the public to do a good job in protection and further inhibit the spread of the epidemic? Therefore, understanding the impact of information diffusion on epidemic transmission can help improve the prediction of epidemics and find preventive measures to slow down the spread of diseases. Therefore, this paper studies the problems above. The innovation points of the research are as follows: First, combing the epidemic theory, complex network analysis and the temporal and spatial background of the COVID-19 spread, we comb the mechanism of epidemic transmission; Second, we use econometric method to conduct a regress, and get the basic conclusion that information diffusion can effectively reduce the spread of COVID-19; third, using big data mining technology, Baidu search index, Baidu population migration, prevention and control data during the epidemic were mined through Baidu and government information websites at all levels. The remaining sections of the paper are as follows: the second section combs the research literature and puts forward the hypothesis; the third section is the research design, which describes the model construction, data source, main variable calculation, and statistical description; the fourth section is empirical analysis. The fifth section is a conclusion and discussion. \\nSince the beginning of the 21st century, due to the outbreak of SARS, avian influenza, novel H1N1 influenza and Ebola cross the world, the public has been increasingly concerned about the emerging infectious diseases, and the problem of disease transmission has been widely studied (Mao & Yang, 2012) . In general, the spread of an epidemic is considered to be a dynamic process in which the disease passes from one individual to another through contact between individuals on the contact network (kleczkowski et al., 2011) . Disease transmission often occurs in a dynamic social environment, and individual health behavior decision-making is guided by cultural norms, peer behavior and media reports (Kim et al., 2019) . Although vaccination is a major strategy to protect individuals from infection, the development, testing and production of new vaccines often take a long time (stohr &amp; esveld, 2004) . Before getting enough vaccines, the best protection for individuals is to take preventive actions, such as wearing masks, washing hands frequently, taking drugs, avoiding contact with patients, etc. (Centers for Disease Control and prevention, 2008) . The historical experience of SARS tells us that effective national control measures, such as early identification and isolation of SARS cases, tracking and isolation of the contacts, screening of travelers, and raising public awareness of risk, can help to contain the spread of the virus (Ahmad et al., 2009) . As the public gradually realized the importance of personal behavior in preventing the spread of infection, researchers began to explore the mathematical model of disease transmission including personal behavior. These models have been used to guide strategies for disease transmission control (vardavas et al., 2007) and quantify the role of individual protective measures in controlling several outbreaks, including the Ebola virus outbreak in West Africa in 2014 (fast et al., 2007) , the SARS outbreak in Hong Kong in 2003 and the H1N1 outbreak in central Mexico in 2009 Mexico in (springborn et al., 2007 . Saunders et al. (2017) also tested the effectiveness of personal protective measures in preventing the spread of pandemic influenza in humans. Recently, the research on the outbreak of COVID-19 from the perspective of transmission dynamics is also increasing. Sun et al. (2020) evaluated the epidemiological trend of COVID-19 based on the public epidemic data, and studied the outbreak progress in all parts of China. Understanding the impact of the media on the spread of the disease can help improve the prediction of epidemics and identify preventive measures to slow the spread of the disease. Many models also link the disease-related media transmission with the protection function, usually assuming that the influence of media will reduce the effective transmission rate and slow down the spread of diseases. These studies indicate that the impact of media increases with the number of people infected (Sun et al., 2011; , or both with the number and the rate of change (tchuenche &amp; bauch, 2012; Xiao et al., 2015) . When the number of cases is high or the prevalence of diseases increases rapidly, the information diffusion slows down the spread of diseases and creates interesting disease transmission dynamics, such as multi-wave outbreaks . However, it is not clear whether the media function formalization proposed by the model fully reflects the actual influence. The choice of media function directly affects the form of disease transmission, making the accurate parameterization of the media is the key (collison et al., 2014) . However, most of the current researches only focus on the development of the disease itself on the complex network, as well as the impact of protective measures on the spread of the disease. There are relatively few studies on the spread of disease-related information, and only a few of them are carried out through numerical simulation with the preconditions which are too idealized and too dependent on the setting of parameters. In addition, the model has just begun to consider how to combine the data from actual media reports (collison et al., 2015) , lacking the econometric analysis based on real-time data. In decade years, China's Internet has experienced unprecedented development, various online social media based on the Internet (such as major search engines, social networking sites, news sites, etc.) have been integrated into people's daily life, providing a broad platform for the dissemination of various information. Compared with the outbreak of SARS in 2003, the economic link between regions are increasingly close, and during the Spring Festival in China, the population flow is more frequent. Although the government has taken unprecedented measures to prevent and control the epidemic, and the official media and social media have timely released the latest epidemic information, the actual effect needs to be further tested. By combing and integrating the existing literature, we propose the hypothesis that information diffusion is helpful to curb the spread of new coronavirus.\\nAs shown in the figure 2, there is a clear correlation between the COVID-19-related information and the spread of the epidemic. This indicates that it is feasible to study the development of the epidemic through the Network public opinion information. Further, we visualized the geographic distribution of the epidemic and information (as shown in figure 3 ). Overall, there is a correlation between the two. It can be seen that the place with the most public opinion information is actually not the place where the epidemic is the most serious (such as Hubei Province, but it does not rule out that the epidemic area has no time to process the information), but the first-tier provinces such as Beijing, Shanghai and Guangdong (The situation in these cities is still severe, but it is relatively stable compared to Hubei Province). Therefore, this article proposes a hypothesis: the spread of epidemic information will inhibit the spread of the epidemic. \\nThe mechanism of the influence of information dissemination on the spread of COVID-19 is as follows: When a disease spreads among people, information about the disease spreads immediately. In the early days of the outbreak, a small amount of information spread, mainly about the popular science of the virus and the current infection dynamics. The information has improved the public's awareness of the virus. Many people choose to continue to pay attention to the epidemic-related information, and respond faster when the outbreak later occurs, reducing the probability of infection. In the stage of the outbreak, information about it will also be spread on major social media. When individuals learn about the existence of the disease, they will change their behavior, such as wearing a mask and vaccinating to avoid being infected, which can have an impact on the spread of the disease. The susceptible (or infected person) who knows the information of the epidemic will break the connection with the around infected person (or susceptible person) in order to prevent the further spread of the \\n\\u767e\\u4e07 \\u75ab\\u60c5\\u4fe1\\u606f \\u7d2f\\u8ba1\\u786e\\u8bca\\u4eba\\u6570 All rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10. 1101 /2020 disease. After receiving the epidemic-related information, governments at all levels also quickly launched emergency plans, and gradually adopted a series of measures such as early identification and isolation, traveler screening, closing of public places, and even \\\"lockdown\\\" the cities in accordance with the deterioration of the epidemic to limit traffic and population flow, greatly reducing further transmission of the epidemic. \\nIn the current literature, models of disease transmission and behavioral spread have been developed for decades, all are based on human networks ((Deffuant, Huet & Amblard, 2005; Keeling & Eames, 2005) . However, few people committed to combining information diffusion and human behavior, considering these two interactive processes. In fact, when a disease-related information spreads among people, people will naturally take some precautionary measures to counteract it, which in turn limits the spread of the disease. In order to study the impact of information diffusion on epidemic outbreaks, this paper draws on the conceptual framework of Mao & Yang (2012) to develop a model of the interaction between disease and information. We divided the population into two categories according to their health status: the susceptible (S) and the (I); according to the level of information obtained, they were divided into two categories: conscious (+) and unconscious (-). Therefore, the population in the entire society can be divided into four states: (1) S\\u2212: unconscious susceptible; (2) S +: conscious susceptible; (3) I\\u2212: unconscious infected; (4) I +: conscious infected. As shown in the figure 5, the transmission process of COVID-19 including the information diffusion is proposed. The entire propagation process of the model is described as follows: the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.03.31.20048439 doi: medRxiv preprint Since the reinfection process after recovery of patients with COVID-19 is not clear,here we don't take it into consideration. Therefore, the information diffusion can reduce the spread of the epidemic in two ways. First, information diffusion will cause some conscious and susceptible people ( + S \\uf062 ) to take proactive protective measures to prevent infection; second, information diffusion will change some unconsciously susceptible people ( 1 S \\uf071 \\uf02d ) into consciously susceptible people, and then take protective measures to reduce some\\nAccording to the hypothesis above, the empirical model of the influence of information diffusion on the spread of COVID-19 is as follow:\\nis measured by the number of cumulative and newly confirmed cases published by the All rights reserved. No reuse allowed without permission. the author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10.1101/2020.03.31.20048439 doi: medRxiv preprint National Health Commission each day; X is control variables, including traffic control ( _ traf con ), self-control ( _ self seg ), movement of population ( migration ), Population inflow rate of Wuhan ( ratio ), and GDP per capita ( pergdp ). i \\uf078 is time fixed effect, j \\uf078 is city fixed time, ij \\uf065 the random error term.\\nIn order to quantitatively explore the relationship between the spread of the COVID-19 and information diffusion, we first visited the Baidu Index website through Python to obtain the Baidu search index2 of the keywords related to the epidemic during the outbreak from January 19 to February 10, 2020 to measure the level of information diffusion. The data of infected case during the corresponding period mainly come from the daily epidemic data released by the National Health Commission. The self-control and traffic control data come from the public information of each city's Health Commission website and government website on taking preventive and control measures, and they are scored uniformly according to the degree of control, and the corresponding values are added up. The national migration data and Wuhan's outflow data come from Baidu Migration. The control variables at the city level come from China City Statistical Yearbook. In addition, cities without outbreaks were also excluded. After collation, 6417 observations from 301 cities were finally obtained.\\nCoronavirus transmission ( XGBD ): The number of the infected in this article is used to indicate the transmission of the virus. After the outbreak of the epidemic, the National Health Commission provided daily outbreak data. Therefore, we have compiled a list of the daily number of cumulative and newly confirmed in prefecture-level cities from January 19 to February 10, 2020.\\nInformation diffusion ( search ): The main explanatory variable---information diffusion in this paper is measured by the number of searches for epidemic-related information by the national people every day during the epidemic. Different from the media data (***) used in previous studies as the level of information diffusion, the search index can better reflect people's acceptance of information diffusion. Therefore, based on the search services provided by Baidu Index, six epidemic-related terms of \\\"the novel coronavirus\\\", \\\"pneumonia\\\", \\\"Zhong Nanshan\\\", \\\"pneumonia symptoms\\\", \\\"masks\\\" and \\\"correct wearing of masks\\\" were selected as search terms, and the search index during the epidemic period (January 19-February 10) was crawled by Python, and the daily level of information diffusion of prefecture-level cities was summed up.\\nIn addition, the search index is divided into two categories, one is about the information on epidemic with \\\"the novel coronavirus, pneumonia, Zhong Nanshan\\\"\\n( 1 search ), and the other is about the protection with \\\"pneumonia symptoms, correct wearing of masks and masks\\\" ( 2 search ). Traffic Control ( _ traf con ) and Self-control ( _ self seg ): This paper collects, summarizes and collates the epidemic prevention and control mechanisms published by the Emergency Command of the Novel Coronavirus Pneumonia Prevention and Control in each province and city, mainly including traffic control and self-control. According to these preventive and control measures taken by all prefectural administrative regions in the country, they are classified and scored into 15 items (see Table 1 ), each with a score of 1, starting from the time when each measure is implemented until the measure is cancelled. For example, Shanghai began to implement the \\\"isolation of close contacts of confirmed patients\\\" for 14 days on January 21. Since this measure belongs to \\\"social isolation\\\", the \\\"social isolation\\\" score of Shanghai was 1 from January 21. On January 24, Shanghai began to implement the \\\"partial cessation of public places in the city\\\", then the \\\"social isolation\\\" was added 1 point from January 24, and so on. Finally, traffic control was carried out separately. The scores of each measure of social alienation were summed up.\\nPopulation Flow ( migration ): As the epidemic occurred during the Spring Festival Movement in China, the large-scale population flow provided favorable conditions for the spread of the virus, and reasonable control of population flow helped to slow down the spread of the epidemic. Baidu Migration Big Data provides a migration index that reflects the scale of population migration into or out, and is comparable between cities. Therefore, the migration indexes of population moving in and out of prefecture-level cities were obtained respectively during the epidemic respectively and the indicators reflecting the overall population flow status of the city were summed up.\\nLaunching level 1 response Closing all the public places Suspending all the cross-city passenger transport Closing part the public places Suspending part of the cross-city passenger transport Closed management of all the community Monitoring all the cross-city passenger transport Closed management of part of the community Monitoring part of the cross-city passenger transport Quarantining returnees from key epidemic area (Hubei) for 14days\\nSuspending all the public transport Quarantining all the returnees for 14days\\nSuspending part of the public transport Quarantining the contact for 14days\\nAll rights reserved. No reuse allowed without permission.\\nthe author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nThe copyright holder for this preprint (which was not peer-reviewed) is . https://doi.org/10. 1101 /2020 Isolating and testing the suspected Table 1 Items of Traffic Control and Self-control Other control variables. (1) The influx of population in Wuhan ( ratio ). Wuhan is the city of the outbreak, and the influx of Wuhan population may lead to the cross-city transmission of the epidemic. The big data of Baidu Migration provides the destination and proportion of population migration in Wuhan every day. This paper selects the proportion of population flow in Wuhan to other cities to represent the population inflow in Wuhan; (2) GDP per capita ( pergdp ). GDP per capita reflects the level of urban social and economic development, while cities with high economic development tend to have more complete epidemic prevention facilities and stronger epidemic prevention capacity. \\nAccording to the econometric model constructed above, benchmark regression was performed by controlling time and urban fixed effects separately. Column (1) in Table  3 performed a direct regression on the spread of information and the spread of the COVID-19, and the results were statistically significant and negative. Column (2) showed results when added traffic control and personal control variables. Column (3) showed the results after adding further variables such as population flow, Wuhan inflow, and GDP per capita based on column (2). The regression results are also statistically significant and negative. This indicates that after controlling other factors affecting the COVID-19, the spread of information has significantly reduced the spread of COVID-19 in the country. The regression coefficients of traffic control and personal control were significantly negative, indicating that after implementing a first-level response measures such as urban traffic control, segregated observation, and closed communities adopted by local governments were significantly effective, reducing the spread of the COVID-19. In terms of variables reflecting population migration, the coefficient of urban population migration variable is significantly negative, indicating that the decrease in population inflows and outflows has also reduced the spread of COVID-19; the coefficient of Wuhan population inflow variables is significantly positive, indicating that the population inflow in Wuhan has accelerated the spread of the virus to a certain extent. Wuhan is the place where the disease first spread on a large scale. The timing of the initial confirmed cases is unknown, and the lack of medical equipment and supplies after the outbreak has caused the number of confirmed cases to be lower than the actual number of confirmed cases. Therefore, in column (4), we exclude the data of Wuhan City, and the regression results are still stable.\\nIn order to learn more about the impact of information diffusion on the spread of COVID-19, this article divides the information represented by the Baidu search index into two categories. One is information on COVID-19 that the public desires to know (Search 1). The other is the self-protection information (search2) during the epidemic search by the public in order to prevent themselves from being infected by the virus.\\nColumns (1) and (2) in Table 4 are the results of robustness test which shows the impacts of the spread of epidemic information on the spread of COVID-19, the regression coefficients are significantly negative. Columns (3) and (4) are the results of robustness test which shows the impacts of the self-protection information diffusion on the spread of COVID-19, and the regression coefficients are also significantly negative. It shows that both the spread of virus information and self-protection information has hindered the further spread of the COVID-19.\\n(1) Table 4 Regression results of different kinds of information t statistics in parentheses * p < 0.1, ** p < 0.05, *** p < 0.01\\n(2) Robustness test of new cases\\nThe data released by the National Health and Construction Commission includes the daily number of newly diagnosed patients, which can better reflect the spread of the epidemic every day. Therefore, we use the new confirmed number instead of the cumulative confirmed number in the previous model for further robustness testing. The regression results when use the comprehensive search index in column (2) in Table 5 are significantly negative, and the regression results when use the search index of disease information and protection information in column (3) and column (4) are also significantly negative, this further indicats that the regression results are robust and the spread of information can reduce the spread of the COVID-19.\\n(1) Table 5 Regression results of new diagnoses t statistics in parentheses * p < 0.1, ** p < 0.05, *** p < 0.01\\nThis paper draws on the model of behavioral dynamics, uses econometric methods and high-frequency data such as new coronavirus epidemic data published by the National Health and Medical Commission, Baidu search index and Baidu migration index to explore the relationship between information diffusion and the spread of COVID-19 Relationship. Studies show that: Firstly, after fixing time and cities and controlling other variables that affect the spread of New Coronavirus, the spread of information significantly reduces the spread of COVID-19. After excluding Wuhan from the sample, the regression results are still robust. Secondly, two robustness tests of information classification and new confirmed diagnoses show that both the spread of epidemic information and self-protection information have significantly reduced the further spread of the COVID-19. This shows that when the epidemic occurs, the timely and accurate spread of information plays an important role in the prevention and control of the epidemic.\\nExcessive and inaccurate spread of epidemic information may also bring unexpected counter-effects, such as causing panic among the people, causing snatching of living materials, and curbing economic activities. Therefore, while information diffusion has played a role in the practice of epidemic prevention and control, it may be possible to take interventions to reduce the adverse effects. This article puts forward the following suggestions: (1) The government convenes a press conference in a timely manner to disclose the epidemic situation information and make the information transmission more transparent. The state-run media played a role of weathervane, they need to timely follow up the epidemic report to let public learn about the virus. (2) Do a good job of information management and control. Official authority, hospital and well-known experts need to timely deny a rumor for various purposes to avoid unhealthy social impact. (3) In view of the differences in urban and rural internet penetration rates, governments at all levels need to issue official documents timely to transfer information to rural areas.\\n\",\n          \"In December 2019, a mysterious viral illness causing pneumonia broke out in the city of Wuhan, in Hubei Province in China. A proportion of the earlier cases were associated with a seafood market in the city, where exotic animals were also sold for food (1). Since then, this illness has been shown to be caused by a novel coronavirus (nCoV) that is named 2019-nCoV, and the disease named as coronavirus disease 2019 (COVID-19). As of February 26, 2020, the disease has been confirmed in over 80 900 cases, with over 2760 deaths. Geographically, the disease has spread beyond China to close to 40 other countries.\\nThe sensitive and specific detection of this virus is an important part of the global healthcare response to this outbreak. In this issue of the journal, Chu et al. (2) reported the development of two one-step real-time reverse transcription polymerase chain reaction assays for detecting 2019-nCoV. The 2 assays target the Orf1b and the N region of the viral genome. Due to the relative paucity of positive control materials when the authors developed these assays, the authors had designed the primers and probes such that they would also cross-react with the Severe Acute Respiratory Syndrome (SARS)-CoV. The authors then use SARS-CoV as one of their positive controls. The authors argued that this cross-reactivity would not cause any diagnostic ambiguity as SARS-CoV was no longer seen clinically following the resolution of the SARS epidemic in 2004.\\nThe authors found that the N assay was more sensitive than that targeting the Orf1b. The authors proposed using the former as a screening assay, and the latter as a diagnostic assay. On the basis of the data presented by the authors, one can perhaps argue that this two-tier test arrangement should be considered provisional. Further validation of this approach using a much larger sample cohort would be necessary.\\nWhen the authors designed their assays, only one 2019-nCoV sequence was publicly available. However, since then, several other 2019-nCoV sequences became available (3). It will be essential to reassess the specificity of the authors\\u2019 primers and probes as the sequence database and biobank for 2019-nCoV grow. It also will be important to compare the performance of the authors\\u2019 assays versus those developed by other workers in the field (1).\\nAdditionally, it will be necessary to assess the quantitative performance of the authors\\u2019 assays in a clinical context, using serial samples. Exploration of the viral kinetics in various sample types and looking for correlations with clinical outcome will be valuable. Viral kinetics studies will be beneficial in identifying treatment modalities that may be effective in inhibiting viral replication.\\nViral diagnostics is one important part of our armamentarium against COVID-19. Public health measures, such as decisions to place a person and his or her close contacts under medical isolation, surveillance, or quarantine are intimately related to whether a suspected case has been confirmed to be infected with 2019-nCoV. To combat an epidemic, time is of the essence and hence the rapid development of sensitive and specific diagnostic tests is crucial. It is hoped that new therapies and even vaccines might eventually become available for this disease. The emergence of such viral zoonoses is indeed testing modern global healthcare and collaboration to the extreme.\\n\",\n          \"COVID-19, the highly transmittable and pathogenic viral infection caused by SARS-CoV-2, belongs to betacoronavirus genus that is known to cause acute respiratory distress syndrome, coagulation dysfunction and septic shock (Yang, P. and Wang, X. 2020). The rate of COVID-19 spread appears to be more than SARS-CoV and MERS-CoV ( To study the evolutionary pattern of SARS-CoV-2 strains circulating globally and in Indian subcontinent and also to reveal structural and functional implications of amino acid variations present in the S proteins, we analysed 2,954 SARS-CoV-2 complete genomes. Our analysis reveals that there are at least 5 distinct clades of strains circulating in India. In addition, the global strains appear to have acquired 54 mutations that are non-synonymous in the gene encoding S-protein among which 4 important mutations are on the RBD that may have direct implications on the infectivity of SARS-CoV-2. Most importantly, we identify that one of these RBD mutations is unique to a set of strains currently circulating in India.\\nComplete and high coverage genomes were downloaded on April 6 th , 2020 from the GISAID database.\\nThis comprises a total of 3,060 genomes (>29,000 bp) and countries with less than ten genomes were discarded for further analysis. In addition, we also added 30 Indian genomes downloaded on April 15 th , 2020 for the analysis. The RefSeq genome (NC_045512) from Wuhan is taken as the reference. The genes were predicted using Prokka (Seemann T; , and the complete sequences of structural proteins such as Spike, Membrane and Nucleocapsid proteins were extracted. The alignments of the structural proteins were done using Mafft (maxiterate 1,000 and global pair-ginsi) (Katoh et al. 2002) .\\nThe alignments were visualized in Jalview (Waterhouse AM et al., 2009 ) and the amino acid substitutions in each position were extracted using custom python script. We ignored the substitutions that are present in only one genome and unidentified amino acid X. The mutations that are present in at least two independent genomes in a particular position were further considered. These two criteria were used to avoid mutations due to sequencing errors. The mutated amino acids were further tabulated and plotted as a matrix using R script. The number of amino acid substitutions observed in S proteins were compared against other structural proteins.\\nFor the Maximum likelihood phylogeny, we have used 30 Indian Genomes (as on April 15th, 2020)\\nwith the worldwide genomes sampled 10 per country and with Wuhan RefSeq strain as root. The phylogeny was reconstructed using IQ-Tree (Nguyen et al., 2015) , and the best model \\\"TIM2+F+I\\\" was picked according to the ModelFinder (Kalyaanamoorthy et al., 2017). All the data used for reconstructing the phylogeny was downloaded from the GISAID database. In addition, three random samplings were done to check the consistency of the phylogeny.\\nThe structural analysis of the mutated spike glycoprotein of SARS-CoV-2 was done to assess the impact of mutations on binding affinity towards human ACE2 receptor. \\nWe \\nSince mutations in S-protein have been linked with virus infectivity, we performed a high stringent, indepth mutational analysis to capture the global variations in the S-protein. Altogether, 54 mutations were identified that belong to 1176 strains from 29 countries. These mutations were found to be distributed across different domains of S-protein (Figure 2) .\\nThe N-terminally located signal peptide which helps S-protein in ER translocation during its Hong Kong and France. A unique S438F mutation is found in the RBD region of only two Indian strains, which is absent in global strains. G476S mutation is seen predominantly in 7 strains from the USA and one from Belgium whereas V483A is found in 16 strains unique to the USA. However, these four RBD mutations together represent only <2% of SARS-CoV-2 strains circulating globally.\\nFurther, to understand the evolutionary significance of these mutations, we compared RBDs from SARS-CoV-2, SARS-CoV (2002) and Bat coronavirus RaTG13 strain, a suspected precursor of SARS-CoV-2 (Figure 3) . We found that the RBD from SARS-CoV-2 is only 73.4 % identical to SARS-CoV but is 90.1 % identical to RaTG13. Interestingly, none of the amino acid changes observed in SARS-CoV-2 matches with either RaTG13 or SARS-CoV (Figure 3) . Hence, based on our findings, the possibility of origin of a novel sub-lineage cannot be neglected. \\nThe amino acid substitutions at the RBD can have an influence on the overall binding affinity of the Sprotein with ACE2 receptor. Thus, the structural changes and differences in the binding affinity corresponding to each mutation in the RBD (V367F, S438F, G476S and V483A) were analysed in detail (Figure 4a and b) . (Table-1 ). Although no changes were evident in the interactions during conservative substitution of V367F (Figure 4c, Supplementary Figure 1a) , the binding affinity towards ACE2 receptor decreased (Kd = 1.4E-09) when compared to the wild type protein complex (Kd = 4.3E-10). The amino acid substitution that is unique to Indian strains, S438F is associated with alterations in hydrogen bonding with D442. Also, a new hydrogen bond with N437 is formed in the mutant (Figure 4d, Supplementary Figure 1b) . These structural alterations are consistent with the reduced binding affinity observed in S438F mutant (Kd = 1.2E-09). The amino acid mutation at position 476, G to S also made some changes to its interacting residues. In wild type protein complex, G476 is forming two hydrogen bonds with T478 while this interaction is abolished in the mutant. Also, the single hydrogen bond interaction with Q474 in the wild type was replaced to double bond in mutant (Figure 4e, Supplementary Figure 1c) \\n\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.410516960764089,\n        \"min\": 12.039220176565594,\n        \"max\": 22.26210215914598,\n        \"num_unique_values\": 50,\n        \"samples\": [\n          14.2425107295992,\n          12.340247950822512,\n          12.998781583306965\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 92
        }
      ],
      "source": [
        "p = \"Real-time tracking of whole genomes and a mechanism for coordinating the rapid dissemination of that information to inform the development of diagnostics and therapeutics and to track variations of the virus over time.\"\n",
        "res = searchAndGetAnswer(p)\n",
        "res"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "GjlK-HdUYSsA"
      },
      "source": [
        "### Q2: *Access to geographic and temporal diverse sample sets to understand geographic distribution and genomic differences, and determine whether there is more than one strain in circulation. Multi-lateral agreements such as the Nagoya Protocol could be leveraged.*"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 98,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 710
        },
        "id": "DiIvKRTlYSsA",
        "outputId": "a11d60bb-4b22-46b4-951f-b02c845f8e33"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1/10\n",
            "2/10\n",
            "3/10\n",
            "4/10\n",
            "5/10\n",
            "6/10\n",
            "7/10\n",
            "8/10\n",
            "9/10\n",
            "10/10\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                               title  \\\n",
              "0  No-Test Medication Abortion: A Sample Protocol...   \n",
              "1  Evaluation of Rural Public Libraries to Addres...   \n",
              "2  Perfectionism and Perceived Control in Posttra...   \n",
              "3  Anatomic evidence shows that lymphatic drainag...   \n",
              "4  Quantifying antibody kinetics and RNA shedding...   \n",
              "5           Presence of SARS-Coronavirus-2 in sewage   \n",
              "6  Accelerated infection testing at scale: a prop...   \n",
              "7  Mobilization of Telepsychiatry in Response to ...   \n",
              "8      Genomic variance of the 2019‐nCoV coronavirus   \n",
              "9  Identification of IgG antibody response to SAR...   \n",
              "\n",
              "                                            abstract Answer      Score  \\\n",
              "0                                                NaN  【  】   18.587669   \n",
              "1  Introduction: In the United States, access to ...  【  】   16.814598   \n",
              "2  In this study, we sought to examine associatio...  【  】   16.680524   \n",
              "3  Respiratory infections can result in intracran...  【  】   16.471302   \n",
              "4  Our ability to understand and mitigate the spr...  【  】   15.147414   \n",
              "5  In the current COVID-19 pandemic, a significan...  【  】   14.390560   \n",
              "6  In pandemics or epidemics, public health autho...  【  】   13.906085   \n",
              "7  The COVID-19 pandemic threatens to disrupt the...  【  】   13.785966   \n",
              "8  There is a rising global concern for the recen...  【  】   13.749903   \n",
              "9  Diagnostic testing and evaluation of patient i...  【  】   13.572998   \n",
              "\n",
              "                                                 url  \\\n",
              "0  https://doi.org/10.1016/j.contraception.2020.0...   \n",
              "1  http://medrxiv.org/cgi/content/short/2020.05.2...   \n",
              "2  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   \n",
              "3  https://www.sciencedirect.com/science/article/...   \n",
              "4  http://medrxiv.org/cgi/content/short/2020.05.1...   \n",
              "5  http://medrxiv.org/cgi/content/short/2020.03.2...   \n",
              "6             https://arxiv.org/pdf/2003.13282v1.pdf   \n",
              "7  https://doi.org/10.1007/s10488-020-01044-z; ht...   \n",
              "8  https://www.ncbi.nlm.nih.gov/pubmed/32027036/;...   \n",
              "9  http://medrxiv.org/cgi/content/short/2020.05.0...   \n",
              "\n",
              "                                             authors  \n",
              "0  Raymond, Elizabeth G.; Grossman, Daniel; Mark,...  \n",
              "1  DeGuzman, P. B.; Siegfried, Z. C.; Leimkuhler,...  \n",
              "2  Molnar, Danielle S.; Flett, Gordon L.; Hewitt,...  \n",
              "3  Elham, Elzat; Wumaier, Reziya; Wang, Chengji; ...  \n",
              "4  Borremans, B.; Gamble, A.; Prager, K. C.; Helm...  \n",
              "5  Medema, G.; Heijnen, L.; Elsinga, G.; Italiaan...  \n",
              "6                   Jain, Tarun; Jain, Bijendra Nath  \n",
              "7  Kannarkat, Jacob T.; Smith, Noah N.; McLeod-Br...  \n",
              "8              Ceraolo, Carmine; Giorgi, Federico M.  \n",
              "9  McAndrews, K. M.; Dowlatshahi, D. P.; Hensel, ...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-6454596f-6902-44d6-abba-901fed8e28b0\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>title</th>\n",
              "      <th>abstract</th>\n",
              "      <th>Answer</th>\n",
              "      <th>Score</th>\n",
              "      <th>url</th>\n",
              "      <th>authors</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>No-Test Medication Abortion: A Sample Protocol...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>【  】</td>\n",
              "      <td>18.587669</td>\n",
              "      <td>https://doi.org/10.1016/j.contraception.2020.0...</td>\n",
              "      <td>Raymond, Elizabeth G.; Grossman, Daniel; Mark,...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Evaluation of Rural Public Libraries to Addres...</td>\n",
              "      <td>Introduction: In the United States, access to ...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>16.814598</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.2...</td>\n",
              "      <td>DeGuzman, P. B.; Siegfried, Z. C.; Leimkuhler,...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Perfectionism and Perceived Control in Posttra...</td>\n",
              "      <td>In this study, we sought to examine associatio...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>16.680524</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>Molnar, Danielle S.; Flett, Gordon L.; Hewitt,...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Anatomic evidence shows that lymphatic drainag...</td>\n",
              "      <td>Respiratory infections can result in intracran...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>16.471302</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Elham, Elzat; Wumaier, Reziya; Wang, Chengji; ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Quantifying antibody kinetics and RNA shedding...</td>\n",
              "      <td>Our ability to understand and mitigate the spr...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>15.147414</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.1...</td>\n",
              "      <td>Borremans, B.; Gamble, A.; Prager, K. C.; Helm...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>Presence of SARS-Coronavirus-2 in sewage</td>\n",
              "      <td>In the current COVID-19 pandemic, a significan...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>14.390560</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.03.2...</td>\n",
              "      <td>Medema, G.; Heijnen, L.; Elsinga, G.; Italiaan...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>Accelerated infection testing at scale: a prop...</td>\n",
              "      <td>In pandemics or epidemics, public health autho...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>13.906085</td>\n",
              "      <td>https://arxiv.org/pdf/2003.13282v1.pdf</td>\n",
              "      <td>Jain, Tarun; Jain, Bijendra Nath</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Mobilization of Telepsychiatry in Response to ...</td>\n",
              "      <td>The COVID-19 pandemic threatens to disrupt the...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>13.785966</td>\n",
              "      <td>https://doi.org/10.1007/s10488-020-01044-z; ht...</td>\n",
              "      <td>Kannarkat, Jacob T.; Smith, Noah N.; McLeod-Br...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>Genomic variance of the 2019‐nCoV coronavirus</td>\n",
              "      <td>There is a rising global concern for the recen...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>13.749903</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32027036/;...</td>\n",
              "      <td>Ceraolo, Carmine; Giorgi, Federico M.</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>Identification of IgG antibody response to SAR...</td>\n",
              "      <td>Diagnostic testing and evaluation of patient i...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>13.572998</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.0...</td>\n",
              "      <td>McAndrews, K. M.; Dowlatshahi, D. P.; Hensel, ...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-6454596f-6902-44d6-abba-901fed8e28b0')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-6454596f-6902-44d6-abba-901fed8e28b0 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-6454596f-6902-44d6-abba-901fed8e28b0');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-5f225558-f685-4f61-a798-22cd7c9d012d\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-5f225558-f685-4f61-a798-22cd7c9d012d')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-5f225558-f685-4f61-a798-22cd7c9d012d button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "res",
              "summary": "{\n  \"name\": \"res\",\n  \"rows\": 10,\n  \"fields\": [\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Genomic variance of the 2019\\u2010nCoV coronavirus\",\n          \"Evaluation of Rural Public Libraries to Address Telemedicine Inequities\",\n          \"Presence of SARS-Coronavirus-2 in sewage\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 9,\n        \"samples\": [\n          \"There is a rising global concern for the recently emerged novel coronavirus (2019\\u2010nCoV). Full genomic sequences have been released by the worldwide scientific community in the last few weeks to understand the evolutionary origin and molecular characteristics of this virus. Taking advantage of all the genomic information currently available, we constructed a phylogenetic tree including also representatives of other coronaviridae, such as Bat coronavirus (BCoV) and severe acute respiratory syndrome. We confirm high sequence similarity (>99%) between all sequenced 2019\\u2010nCoVs genomes available, with the closest BCoV sequence sharing 96.2% sequence identity, confirming the notion of a zoonotic origin of 2019\\u2010nCoV. Despite the low heterogeneity of the 2019\\u2010nCoV genomes, we could identify at least two hypervariable genomic hotspots, one of which is responsible for a Serine/Leucine variation in the viral ORF8\\u2010encoded protein. Finally, we perform a full proteomic comparison with other coronaviridae, identifying key aminoacidic differences to be considered for antiviral strategies deriving from previous anti\\u2010coronavirus approaches.\",\n          \"In this study, we sought to examine associations between dimensions of trait perfectionism and posttraumatic stress disorder (PTSD) symptoms among a community sample of adults who experienced at least one traumatic event and determine whether the associations between trait perfectionism dimensions and PTSD symptomatology varied as a function of perceived control. A sample of 161 (57% women; M = 33.24 years, SD = 10.84 years) community adults who experienced at least one traumatic event in their lives completed self-report measures of trait perfectionism, perceived control, personality, and PTSD symptomatology. Results from multiple regression analyses indicated that socially prescribed perfectionism was associated with higher levels of PTSD symptomatology. However, this finding was qualified by a significant interaction with perceived control such that higher levels of socially prescribed perfectionism were associated with greater PTSD symptomatology only under conditions of low perceived control. Findings underscore the importance of individual differences in socially prescribed perfectionism and perceived control for PTSD symptomatology.\",\n          \"In pandemics or epidemics, public health authorities need to rapidly test a large number of individuals, both to determine the line of treatment as well as to know the spread of infection to plan containment, mitigation and future responses. However, the lack of adequate testing kits could be a bottleneck, especially in the case of unanticipated new diseases, such as COVID-19, where the testing technology, manufacturing capability, distribution, human skills and laboratories might be unavailable or in short supply. In addition, the cost of the standard PCR test is approximately USD 48, which is prohibitive for poorer patients and most governments. We address this bottleneck by proposing a test methodology that pools the sample from two (or more) patients in a single test. The key insight is that a single negative result from a pooled sample likely implies negative infection of all the individual patients. and It thereby rules out further tests for the patients. This protocol, therefore, requires significantly fewer tests. This may, however, result in somewhat increased false negatives. Our simulations show that combining samples from two patients with 7% underlying likelihood of infection implies that 36% fewer test kits are required, with 14% additional units of time for testing.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Answer\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"\\u3010  \\u3011 \"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1.7272465905741388,\n        \"min\": 13.572998012892041,\n        \"max\": 18.587669163923284,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          13.749902901904047\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"https://www.ncbi.nlm.nih.gov/pubmed/32027036/; https://doi.org/10.1002/jmv.25700\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Ceraolo, Carmine; Giorgi, Federico M.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 98
        }
      ],
      "source": [
        "p = \"Access to geographic and temporal diverse sample sets to understand geographic distribution and genomic differences, and determine whether there is more than one strain in circulation. Multi-lateral agreements such as the Nagoya Protocol could be leveraged.\"\n",
        "res = searchAndGetAnswer(p)\n",
        "res"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "4iSUCnD1YSsA"
      },
      "source": [
        "### Q3: *Evidence that livestock could be infected (e.g., field surveillance, genetic sequencing, receptor binding) and serve as a reservoir after the epidemic appears to be over.*\n",
        "* Evidence of whether farmers are infected, and whether farmers could have played a role in the origin.\n",
        "* Surveillance of mixed wildlife- livestock farms for SARS-CoV-2 and other coronaviruses in Southeast Asia.\n",
        "* Experimental infections to test host range for this pathogen."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 99,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "RHMfrB3CYSsA",
        "outputId": "07ef4b88-1274-4049-ba9f-8c6736cb9c61"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No valid answers found.\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                title  \\\n",
              "0   Coronavirus surveillance of wildlife in the La...   \n",
              "1   SARS-CoV-2 spike protein predicted to form sta...   \n",
              "2   Spillover of SARS-CoV-2 into novel wild hosts ...   \n",
              "3   Are pangolins the intermediate host of the 201...   \n",
              "4   A potential role for integrins in host cell en...   \n",
              "5   Hypothesis: angiotensin-converting enzyme inhi...   \n",
              "6   Proteolytic cleavage of the SARS-CoV-2 spike p...   \n",
              "7   Predicting wildlife hosts of betacoronaviruses...   \n",
              "8   SARS-CoV-2, an evolutionary perspective of int...   \n",
              "9   Use of the informational spectrum methodology ...   \n",
              "10  The Story behind COVID-19: Animal Diseases at ...   \n",
              "11  Comparison of SARS-CoV-2 spike protein binding...   \n",
              "12  SARS-CoV-2: an Emerging Coronavirus that Cause...   \n",
              "13  Cross‐species transmission of the newly identi...   \n",
              "14  A pneumonia outbreak associated with a new cor...   \n",
              "15  Structural Basis for Potent Neutralization of ...   \n",
              "16  Isolation, quarantine, social distancing and c...   \n",
              "17  Potential host range of multiple SARS-like cor...   \n",
              "18  Monoclonal antibodies for the S2 subunit of sp...   \n",
              "19  Are pangolins the intermediate host of the 201...   \n",
              "20  Interactions of coronaviruses with ACE2, angio...   \n",
              "21  Failed detection of the full-length genome of ...   \n",
              "22  Genetic analysis of the novel SARS-CoV-2 host ...   \n",
              "23  Comparative seasonalities of influenza A, B an...   \n",
              "24  Emergence of SARS-CoV-2 through Recombination ...   \n",
              "25  Receptor Recognition by the Novel Coronavirus ...   \n",
              "26  Evidence of recombination in coronaviruses imp...   \n",
              "27  The Potential Role of Th17 Immune Responses in...   \n",
              "28  Structural Basis for Potent Neutralization of ...   \n",
              "29  Detection of novel coronaviruses in bats in My...   \n",
              "30  Diarrhea may be underestimated: a missing link...   \n",
              "31  SARS‐CoV‐2, an evolutionary perspective of int...   \n",
              "32  Analyses of spike protein from first deposited...   \n",
              "33  A novel bat coronavirus reveals natural insert...   \n",
              "34  Emerging novel coronavirus (2019-nCoV)—current...   \n",
              "35  False-Negative Results of Real-Time Reverse-Tr...   \n",
              "36  No evidence for allelic association between Co...   \n",
              "37  Development of a rapid test kit for SARS-CoV-2...   \n",
              "38  Genome-wide data inferring the evolution and p...   \n",
              "39  Genetic Variability of Human Angiotensin-Conve...   \n",
              "40  The COVID-19 Host Genetics Initiative, a globa...   \n",
              "41  COVID-19: is there a role for immunonutrition,...   \n",
              "42  “Amantadine disrupts lysosomal gene expression...   \n",
              "43            Zoonotic origins of human coronaviruses   \n",
              "44  Heterogeneous expression of the SARS-Coronavir...   \n",
              "45  Susceptible supply limits the role of climate ...   \n",
              "46  Could India be the origin of next COVID-19 lik...   \n",
              "47  “Amantadine disrupts lysosomal gene expression...   \n",
              "48  Deep phylogenetic analysis of Orthocoronavirin...   \n",
              "49  Prediction analysis of SARS-COV-2 entry in Liv...   \n",
              "\n",
              "                                             abstract  \\\n",
              "0   Coronaviruses can become zoonotic as in the ca...   \n",
              "1   SARS-CoV-2 has a zoonotic origin and was trans...   \n",
              "2   Abstract There is evidence that the current ou...   \n",
              "3   The outbreak of 2019-nCoV pneumonia (COVID-19)...   \n",
              "4   • Integrin may act as an alternative receptor ...   \n",
              "5   Intravenous infusions of angiotensin-convertin...   \n",
              "6   Abstract Severe acute respiratory syndrome cor...   \n",
              "7   Despite massive investment in research on rese...   \n",
              "8   The emergence of SARS-CoV-2 has resulted in mo...   \n",
              "9   A novel coronavirus recently identified in Wuh...   \n",
              "10  A number of virological, epidemiological and e...   \n",
              "11  The emergence of a novel coronavirus, SARS-CoV...   \n",
              "12  An ongoing outbreak of pneumonia caused by a n...   \n",
              "13  The current outbreak of viral pneumonia in the...   \n",
              "14  Since the outbreak of severe acute respiratory...   \n",
              "15  Summary Coronaviruses make use of a large enve...   \n",
              "16  Public health measures were decisive in contro...   \n",
              "17  The severe acute respiratory syndrome coronavi...   \n",
              "18  The emergence of a novel coronavirus, SARS-CoV...   \n",
              "19  The outbreak of a novel corona Virus Disease 2...   \n",
              "20  The rapid spread of a novel coronavirus, sever...   \n",
              "21  Over 10 percent of recovered and discharged pa...   \n",
              "22  The infection coronavirus disease 2019 (COVID-...   \n",
              "23  Comparative seasonalities of influenza A, B an...   \n",
              "24  COVID-19 has become a global pandemic caused b...   \n",
              "25  Recently, a novel coronavirus (2019-nCoV) has ...   \n",
              "26  A novel coronavirus (nCoV-2019) was the cause ...   \n",
              "27  Increasing evidence points to host Th17 inflam...   \n",
              "28  The pathogenic Middle East respiratory syndrom...   \n",
              "29  The recent emergence of bat-borne zoonotic vir...   \n",
              "30  The outbreak of pneumonia caused by the 2019 N...   \n",
              "31  The emergence of SARS‐CoV‐2 has resulted in ne...   \n",
              "32  India has recently started sequencing SARS-CoV...   \n",
              "33  The unprecedented epidemic of pneumonia caused...   \n",
              "34  Coronaviruses are the well-known cause of seve...   \n",
              "35  The epidemic of 2019 novel coronavirus, later ...   \n",
              "36  Background: Coronaviruses (CoV) are a large fa...   \n",
              "37  We present an example of applying ‘need-driven...   \n",
              "38  Since December 2019, coronavirus disease 2019 ...   \n",
              "39  There appears to be large regional variations ...   \n",
              "40                                                NaN   \n",
              "41  In late December 2019 severe acute respiratory...   \n",
              "42  SARS-coronavirus 2 is the causal agent of the ...   \n",
              "43  Mutation and adaptation have driven the co-evo...   \n",
              "44  Rationale Zoonotically transmitted coronavirus...   \n",
              "45  Preliminary evidence suggests that climate may...   \n",
              "46  Abstract The COVID-19 global pandemic is not e...   \n",
              "47  Abstract SARS-coronavirus 2 is the causal agen...   \n",
              "48  The outbreak of severe acute respiratory syndr...   \n",
              "49  Interaction between spike protein of SARS-CoV-...   \n",
              "\n",
              "                                                  url  \\\n",
              "0           https://doi.org/10.1101/2020.04.22.056218   \n",
              "1           https://doi.org/10.1101/2020.05.01.072371   \n",
              "2   https://api.elsevier.com/content/article/pii/S...   \n",
              "3           https://doi.org/10.1101/2020.02.18.954628   \n",
              "4   https://doi.org/10.1016/j.antiviral.2020.10475...   \n",
              "5   https://www.ncbi.nlm.nih.gov/pubmed/32186711/;...   \n",
              "6   https://api.elsevier.com/content/article/pii/S...   \n",
              "7           https://doi.org/10.1101/2020.05.22.111344   \n",
              "8           https://doi.org/10.1101/2020.03.21.001933   \n",
              "9   https://www.ncbi.nlm.nih.gov/pubmed/32419926/;...   \n",
              "10                https://doi.org/10.1017/err.2020.45   \n",
              "11          https://doi.org/10.1101/2020.05.08.084061   \n",
              "12  https://doi.org/10.7150/ijbs.45053; https://ww...   \n",
              "13  https://www.ncbi.nlm.nih.gov/pubmed/31967321/;...   \n",
              "14  https://www.ncbi.nlm.nih.gov/pubmed/32015507/;...   \n",
              "15  https://doi.org/10.1016/j.cell.2020.04.031; ht...   \n",
              "16  https://www.ncbi.nlm.nih.gov/pubmed/32052841/;...   \n",
              "17          https://doi.org/10.1101/2020.04.10.032342   \n",
              "18          https://doi.org/10.1101/2020.03.06.980037   \n",
              "19  https://www.ncbi.nlm.nih.gov/pubmed/32407364/;...   \n",
              "20  https://www.ncbi.nlm.nih.gov/pubmed/32341442/;...   \n",
              "21        https://doi.org/10.1101/2020.03.27.20043299   \n",
              "22          https://doi.org/10.1101/2020.04.23.057190   \n",
              "23  https://www.sciencedirect.com/science/article/...   \n",
              "24          https://doi.org/10.1101/2020.03.20.000885   \n",
              "25  https://doi.org/10.1128/jvi.00127-20; https://...   \n",
              "26          https://doi.org/10.1101/2020.02.07.939207   \n",
              "27  https://doi.org/10.1016/j.micinf.2020.04.005; ...   \n",
              "28          https://doi.org/10.1101/2020.03.26.010165   \n",
              "29  https://www.ncbi.nlm.nih.gov/pubmed/32271768/;...   \n",
              "30  http://medrxiv.org/cgi/content/short/2020.02.0...   \n",
              "31                  https://doi.org/10.1111/eva.12980   \n",
              "32          https://doi.org/10.1101/2020.04.28.066985   \n",
              "33          https://doi.org/10.1101/2020.03.02.974139   \n",
              "34  https://www.ncbi.nlm.nih.gov/pubmed/32036774/;...   \n",
              "35  https://doi.org/10.3348/kjr.2020.0146; https:/...   \n",
              "36  http://medrxiv.org/cgi/content/short/2020.05.2...   \n",
              "37  https://doi.org/10.1007/s42242-020-00075-7; ht...   \n",
              "38          https://doi.org/10.1101/2020.03.04.976662   \n",
              "39          https://doi.org/10.1101/2020.04.14.041434   \n",
              "40  https://www.ncbi.nlm.nih.gov/pubmed/32404885/;...   \n",
              "41  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   \n",
              "42          https://doi.org/10.1101/2020.04.05.026187   \n",
              "43  https://doi.org/10.7150/ijbs.45472; https://ww...   \n",
              "44          https://doi.org/10.1101/2020.04.22.056127   \n",
              "45        https://doi.org/10.1101/2020.04.03.20052787   \n",
              "46  https://www.sciencedirect.com/science/article/...   \n",
              "47  https://www.ncbi.nlm.nih.gov/pubmed/32361028/;...   \n",
              "48          https://doi.org/10.1101/2020.05.12.091199   \n",
              "49          https://doi.org/10.1101/2020.05.08.084327   \n",
              "\n",
              "                                              authors  \\\n",
              "0   McIver, David J.; Silithammavong, Soubanh; The...   \n",
              "1   Lam, SD; Bordin, N; Waman, VP; Scholes, HM; As...   \n",
              "2                 Franklin, Alan B.; Bevins, Sarah N.   \n",
              "3   Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...   \n",
              "4   Sigrist, Christian JA; Bridge, Alan; Le Mercie...   \n",
              "5                                       Diaz, James H   \n",
              "6   Jaimes, Javier A.; Millet, Jean K.; Whittaker,...   \n",
              "7   Becker, Daniel J.; Albery, Gregory F.; Sjodin,...   \n",
              "8   Armijos-Jaramillo, Vinicio; Yeager, Justin; Mu...   \n",
              "9   Veljkovic, Veljko; Vergara-Alert, Júlia; Segal...   \n",
              "10            DE SADELEER, Nicolas; GODFROID, Jacques   \n",
              "11  Zhai, Xiaofeng; Sun, Jiumeng; Yan, Ziqing; Zha...   \n",
              "12                                         Zheng, Jun   \n",
              "13  Ji, Wei; Wang, Wei; Zhao, Xiaofang; Zai, Junji...   \n",
              "14  Zhou, Peng; Yang, Xing-Lou; Wang, Xian-Guang; ...   \n",
              "15  Wrapp, Daniel; De Vlieger, Dorien; Corbett, Ki...   \n",
              "16                     Wilder-Smith, A; Freedman, D O   \n",
              "17  Li, Yujun; Wang, Haimin; Tang, Xiaojuan; Ma, D...   \n",
              "18  Zheng, Zhiqiang; Monteil, Vanessa M.; Maurer-S...   \n",
              "19  Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...   \n",
              "20                          Kai, Hisashi; Kai, Mamiko   \n",
              "21  Hu, Fengyu; Chen, Fengjuan; Wang, Yaping; Xu, ...   \n",
              "22  Russo, Roberta; Andolfo, Immacolata; Lasorsa, ...   \n",
              "23  Liu, Yang; Lam, Tommy TY; Lai, Florence YL; Kr...   \n",
              "24  Li, Xiaojun; Giorgi, Elena E.; Marichann, Manu...   \n",
              "25  Wan, Yushun; Shang, Jian; Graham, Rachel; Bari...   \n",
              "26  Wong, Matthew C.; Javornik Cregeen, Sara J.; A...   \n",
              "27  Hotez, Peter J.; Bottazzi, Maria Elena; Corry,...   \n",
              "28  Wrapp, Daniel; De Vlieger, Dorien; Corbett, Ki...   \n",
              "29  Valitutto, Marc T.; Aung, Ohnmar; Tun, Kyaw Ya...   \n",
              "30  Liang, W.; Rao, S.; Xiao, C.; Lin, Z.; Zhang, ...   \n",
              "31  Armijos‐Jaramillo, Vinicio; Yeager, Justin; Mu...   \n",
              "32  Begum, Feroza; Mukherjee, Debica; Thagriki, Dl...   \n",
              "33  Zhou, Hong; Chen, Xing; Hu, Tao; Li, Juan; Son...   \n",
              "34  Malik, Yashpal Singh; Sircar, Shubhankar; Bhat...   \n",
              "35  Li, Dasheng; Wang, Dawei; Dong, Jianping; Wang...   \n",
              "36  Novelli, A.; Biancolella, M.; Borgiani, P.; Co...   \n",
              "37  Cui, Zhanfeng; Chang, Hong; Wang, Hui; Lim, Bo...   \n",
              "38  Fang, Bin; Liu, Linlin; Yu, Xiao; Li, Xiang; Y...   \n",
              "39              Li, Quan; Cao, Zanxia; Rahman, Proton   \n",
              "40                                                NaN   \n",
              "41                  Derbyshire, Emma; Delange, Joanne   \n",
              "42  Smieszek, Sandra P.; Przychodzen, Bart P; Poly...   \n",
              "43  Ye, Zi-Wei; Yuan, Shuofeng; Yuen, Kit-San; Fun...   \n",
              "44  Bezara, Miguel Ortiz; Thurman, Andrew; Pezzulo...   \n",
              "45  Baker, Rachel E.; Yang, Wenchang; Vecchi, Gabr...   \n",
              "46                                     Minhas, Sachin   \n",
              "47  Smieszek, Sandra P.; Przychodzen, Bart P; Poly...   \n",
              "48                    Sharma, Amresh Kumar; Som, Anup   \n",
              "49  Praharaj, Manas Ranjan; Garg, Priyanka; Nabi K...   \n",
              "\n",
              "                                                 text      Score  \n",
              "0   The latest coronavirus (CoV) outbreak in human...  30.306554  \n",
              "1   Severe acute respiratory syndrome coronavirus ...  29.467318  \n",
              "2   J o u r n a l P r e -p r o o f 2 infected huma...  26.276919  \n",
              "3   In December 2019, there was an outbreak of pne...  25.001637  \n",
              "4   Since December 2019, a novel coronavirus (nCoV...  23.883754  \n",
              "5   Highlight Intravenous infusions of angiotensin...  23.743547  \n",
              "6   Since December 2019, human infections by a nov...  23.606242  \n",
              "7   sampling bias and can only make predictions fo...  23.308656  \n",
              "8   The recent emergence of the novel SARS coronav...  22.974869  \n",
              "9   Fears are mounting worldwide over the cross-bo...  22.492407  \n",
              "10  Today, the world is facing the ongoing COVID-1...  22.325828  \n",
              "11  infection of the respective animal. Some ACE2 ...  21.729573  \n",
              "12  Since December 2019, an increasing number of p...  21.488953  \n",
              "13  China has been the epicenter of emerging and r...  21.253602  \n",
              "14  Coronaviruses have caused two large-scale pand...  19.634429  \n",
              "15  Coronaviruses are enveloped, positive-sense RN...  19.554057  \n",
              "16  The novel coronavirus (2019-nCoV) that emerged...  19.552576  \n",
              "17  Two subgroups of coronaviruses, including alph...  19.498712  \n",
              "18  S1 subunit being responsible for attachment to...  19.405478  \n",
              "19  In December 2019, there was an outbreak of pne...  19.247204  \n",
              "20  A novel coronavirus, named severe acute respir...  19.016471  \n",
              "21  The novel coronavirus (SARS-CoV-2) infection s...  18.973756  \n",
              "22  In December 2019 a new infectious respiratory ...  18.915290  \n",
              "23  Dear Editor,\\nWe read with interest the recent...  18.860648  \n",
              "24  The severe respiratory disease COVID-19 was fi...  18.853971  \n",
              "25  A novel coronavirus (2019-nCoV) from Wuhan, Ch...  18.789114  \n",
              "26  An outbreak of respiratory illness caused by a...  18.610622  \n",
              "27  COVID19 caused by the SARS-2 coronavirus (SARS...  18.203589  \n",
              "28  The pathogenic Middle East respiratory syndrom...  18.150488  \n",
              "29  Infectious diseases are considered to be “emer...  18.096310  \n",
              "30  same cell entry receptor ACE2 with that of SAR...  18.081031  \n",
              "31  The recent emergence of the novel SARS coronav...  17.867609  \n",
              "32  SARS-CoV2 (a member of Coronaviruses) outbreak...  17.846854  \n",
              "33  The unprecedented epidemic of pneumonia caused...  17.844430  \n",
              "34  Coronaviruses (CoVs) are well-known causes of ...  17.775618  \n",
              "35  At the end of December 2019, pneumonia of unkn...  17.670776  \n",
              "36  Coronaviruses (CoV) are a large family of viru...  17.622174  \n",
              "37  The COVID-19 pandemic caught almost every coun...  17.576274  \n",
              "38  On this analysis technique, to precisely decod...  17.410927  \n",
              "39  As of March 24, 2020, the worldwide tracking f...  17.363501  \n",
              "40  The COVID-19 host genetics initiative is a bot...  17.270880  \n",
              "41  Coronaviruses are important pathogens to anima...  17.161800  \n",
              "42  Recently a novel type of highly virulent beta-...  17.066529  \n",
              "43  Coronaviruses (CoVs) belong to the family Coro...  17.014703  \n",
              "44  infection can be fatal, with the risk for incr...  16.993752  \n",
              "45  The SARS-CoV-2 pandemic represents a generatio...  16.963317  \n",
              "46  It all started from a wet market in Wuhan, the...  16.903808  \n",
              "47  Recently a novel type of highly virulent beta-...  16.896807  \n",
              "48  Coronaviruses are single-stranded RNA virus of...  16.787652  \n",
              "49  Three large-scale disease outbreaks during the...  16.754176  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-1225d3e9-1e16-4bec-9e93-f41bcf89071d\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>title</th>\n",
              "      <th>abstract</th>\n",
              "      <th>url</th>\n",
              "      <th>authors</th>\n",
              "      <th>text</th>\n",
              "      <th>Score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Coronavirus surveillance of wildlife in the La...</td>\n",
              "      <td>Coronaviruses can become zoonotic as in the ca...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.22.056218</td>\n",
              "      <td>McIver, David J.; Silithammavong, Soubanh; The...</td>\n",
              "      <td>The latest coronavirus (CoV) outbreak in human...</td>\n",
              "      <td>30.306554</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>SARS-CoV-2 spike protein predicted to form sta...</td>\n",
              "      <td>SARS-CoV-2 has a zoonotic origin and was trans...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.01.072371</td>\n",
              "      <td>Lam, SD; Bordin, N; Waman, VP; Scholes, HM; As...</td>\n",
              "      <td>Severe acute respiratory syndrome coronavirus ...</td>\n",
              "      <td>29.467318</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Spillover of SARS-CoV-2 into novel wild hosts ...</td>\n",
              "      <td>Abstract There is evidence that the current ou...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Franklin, Alan B.; Bevins, Sarah N.</td>\n",
              "      <td>J o u r n a l P r e -p r o o f 2 infected huma...</td>\n",
              "      <td>26.276919</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Are pangolins the intermediate host of the 201...</td>\n",
              "      <td>The outbreak of 2019-nCoV pneumonia (COVID-19)...</td>\n",
              "      <td>https://doi.org/10.1101/2020.02.18.954628</td>\n",
              "      <td>Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...</td>\n",
              "      <td>In December 2019, there was an outbreak of pne...</td>\n",
              "      <td>25.001637</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>A potential role for integrins in host cell en...</td>\n",
              "      <td>• Integrin may act as an alternative receptor ...</td>\n",
              "      <td>https://doi.org/10.1016/j.antiviral.2020.10475...</td>\n",
              "      <td>Sigrist, Christian JA; Bridge, Alan; Le Mercie...</td>\n",
              "      <td>Since December 2019, a novel coronavirus (nCoV...</td>\n",
              "      <td>23.883754</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>Hypothesis: angiotensin-converting enzyme inhi...</td>\n",
              "      <td>Intravenous infusions of angiotensin-convertin...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32186711/;...</td>\n",
              "      <td>Diaz, James H</td>\n",
              "      <td>Highlight Intravenous infusions of angiotensin...</td>\n",
              "      <td>23.743547</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>Proteolytic cleavage of the SARS-CoV-2 spike p...</td>\n",
              "      <td>Abstract Severe acute respiratory syndrome cor...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Jaimes, Javier A.; Millet, Jean K.; Whittaker,...</td>\n",
              "      <td>Since December 2019, human infections by a nov...</td>\n",
              "      <td>23.606242</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Predicting wildlife hosts of betacoronaviruses...</td>\n",
              "      <td>Despite massive investment in research on rese...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.22.111344</td>\n",
              "      <td>Becker, Daniel J.; Albery, Gregory F.; Sjodin,...</td>\n",
              "      <td>sampling bias and can only make predictions fo...</td>\n",
              "      <td>23.308656</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>SARS-CoV-2, an evolutionary perspective of int...</td>\n",
              "      <td>The emergence of SARS-CoV-2 has resulted in mo...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.21.001933</td>\n",
              "      <td>Armijos-Jaramillo, Vinicio; Yeager, Justin; Mu...</td>\n",
              "      <td>The recent emergence of the novel SARS coronav...</td>\n",
              "      <td>22.974869</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>Use of the informational spectrum methodology ...</td>\n",
              "      <td>A novel coronavirus recently identified in Wuh...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32419926/;...</td>\n",
              "      <td>Veljkovic, Veljko; Vergara-Alert, Júlia; Segal...</td>\n",
              "      <td>Fears are mounting worldwide over the cross-bo...</td>\n",
              "      <td>22.492407</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>The Story behind COVID-19: Animal Diseases at ...</td>\n",
              "      <td>A number of virological, epidemiological and e...</td>\n",
              "      <td>https://doi.org/10.1017/err.2020.45</td>\n",
              "      <td>DE SADELEER, Nicolas; GODFROID, Jacques</td>\n",
              "      <td>Today, the world is facing the ongoing COVID-1...</td>\n",
              "      <td>22.325828</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>Comparison of SARS-CoV-2 spike protein binding...</td>\n",
              "      <td>The emergence of a novel coronavirus, SARS-CoV...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.08.084061</td>\n",
              "      <td>Zhai, Xiaofeng; Sun, Jiumeng; Yan, Ziqing; Zha...</td>\n",
              "      <td>infection of the respective animal. Some ACE2 ...</td>\n",
              "      <td>21.729573</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>SARS-CoV-2: an Emerging Coronavirus that Cause...</td>\n",
              "      <td>An ongoing outbreak of pneumonia caused by a n...</td>\n",
              "      <td>https://doi.org/10.7150/ijbs.45053; https://ww...</td>\n",
              "      <td>Zheng, Jun</td>\n",
              "      <td>Since December 2019, an increasing number of p...</td>\n",
              "      <td>21.488953</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>Cross‐species transmission of the newly identi...</td>\n",
              "      <td>The current outbreak of viral pneumonia in the...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/31967321/;...</td>\n",
              "      <td>Ji, Wei; Wang, Wei; Zhao, Xiaofang; Zai, Junji...</td>\n",
              "      <td>China has been the epicenter of emerging and r...</td>\n",
              "      <td>21.253602</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>A pneumonia outbreak associated with a new cor...</td>\n",
              "      <td>Since the outbreak of severe acute respiratory...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32015507/;...</td>\n",
              "      <td>Zhou, Peng; Yang, Xing-Lou; Wang, Xian-Guang; ...</td>\n",
              "      <td>Coronaviruses have caused two large-scale pand...</td>\n",
              "      <td>19.634429</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>Structural Basis for Potent Neutralization of ...</td>\n",
              "      <td>Summary Coronaviruses make use of a large enve...</td>\n",
              "      <td>https://doi.org/10.1016/j.cell.2020.04.031; ht...</td>\n",
              "      <td>Wrapp, Daniel; De Vlieger, Dorien; Corbett, Ki...</td>\n",
              "      <td>Coronaviruses are enveloped, positive-sense RN...</td>\n",
              "      <td>19.554057</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>Isolation, quarantine, social distancing and c...</td>\n",
              "      <td>Public health measures were decisive in contro...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32052841/;...</td>\n",
              "      <td>Wilder-Smith, A; Freedman, D O</td>\n",
              "      <td>The novel coronavirus (2019-nCoV) that emerged...</td>\n",
              "      <td>19.552576</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>Potential host range of multiple SARS-like cor...</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.10.032342</td>\n",
              "      <td>Li, Yujun; Wang, Haimin; Tang, Xiaojuan; Ma, D...</td>\n",
              "      <td>Two subgroups of coronaviruses, including alph...</td>\n",
              "      <td>19.498712</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>Monoclonal antibodies for the S2 subunit of sp...</td>\n",
              "      <td>The emergence of a novel coronavirus, SARS-CoV...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.06.980037</td>\n",
              "      <td>Zheng, Zhiqiang; Monteil, Vanessa M.; Maurer-S...</td>\n",
              "      <td>S1 subunit being responsible for attachment to...</td>\n",
              "      <td>19.405478</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>Are pangolins the intermediate host of the 201...</td>\n",
              "      <td>The outbreak of a novel corona Virus Disease 2...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32407364/;...</td>\n",
              "      <td>Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...</td>\n",
              "      <td>In December 2019, there was an outbreak of pne...</td>\n",
              "      <td>19.247204</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>Interactions of coronaviruses with ACE2, angio...</td>\n",
              "      <td>The rapid spread of a novel coronavirus, sever...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32341442/;...</td>\n",
              "      <td>Kai, Hisashi; Kai, Mamiko</td>\n",
              "      <td>A novel coronavirus, named severe acute respir...</td>\n",
              "      <td>19.016471</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>Failed detection of the full-length genome of ...</td>\n",
              "      <td>Over 10 percent of recovered and discharged pa...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.27.20043299</td>\n",
              "      <td>Hu, Fengyu; Chen, Fengjuan; Wang, Yaping; Xu, ...</td>\n",
              "      <td>The novel coronavirus (SARS-CoV-2) infection s...</td>\n",
              "      <td>18.973756</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>Genetic analysis of the novel SARS-CoV-2 host ...</td>\n",
              "      <td>The infection coronavirus disease 2019 (COVID-...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.23.057190</td>\n",
              "      <td>Russo, Roberta; Andolfo, Immacolata; Lasorsa, ...</td>\n",
              "      <td>In December 2019 a new infectious respiratory ...</td>\n",
              "      <td>18.915290</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>Comparative seasonalities of influenza A, B an...</td>\n",
              "      <td>Comparative seasonalities of influenza A, B an...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Liu, Yang; Lam, Tommy TY; Lai, Florence YL; Kr...</td>\n",
              "      <td>Dear Editor,\\nWe read with interest the recent...</td>\n",
              "      <td>18.860648</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>Emergence of SARS-CoV-2 through Recombination ...</td>\n",
              "      <td>COVID-19 has become a global pandemic caused b...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.20.000885</td>\n",
              "      <td>Li, Xiaojun; Giorgi, Elena E.; Marichann, Manu...</td>\n",
              "      <td>The severe respiratory disease COVID-19 was fi...</td>\n",
              "      <td>18.853971</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>Receptor Recognition by the Novel Coronavirus ...</td>\n",
              "      <td>Recently, a novel coronavirus (2019-nCoV) has ...</td>\n",
              "      <td>https://doi.org/10.1128/jvi.00127-20; https://...</td>\n",
              "      <td>Wan, Yushun; Shang, Jian; Graham, Rachel; Bari...</td>\n",
              "      <td>A novel coronavirus (2019-nCoV) from Wuhan, Ch...</td>\n",
              "      <td>18.789114</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>Evidence of recombination in coronaviruses imp...</td>\n",
              "      <td>A novel coronavirus (nCoV-2019) was the cause ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.02.07.939207</td>\n",
              "      <td>Wong, Matthew C.; Javornik Cregeen, Sara J.; A...</td>\n",
              "      <td>An outbreak of respiratory illness caused by a...</td>\n",
              "      <td>18.610622</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>The Potential Role of Th17 Immune Responses in...</td>\n",
              "      <td>Increasing evidence points to host Th17 inflam...</td>\n",
              "      <td>https://doi.org/10.1016/j.micinf.2020.04.005; ...</td>\n",
              "      <td>Hotez, Peter J.; Bottazzi, Maria Elena; Corry,...</td>\n",
              "      <td>COVID19 caused by the SARS-2 coronavirus (SARS...</td>\n",
              "      <td>18.203589</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>Structural Basis for Potent Neutralization of ...</td>\n",
              "      <td>The pathogenic Middle East respiratory syndrom...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.26.010165</td>\n",
              "      <td>Wrapp, Daniel; De Vlieger, Dorien; Corbett, Ki...</td>\n",
              "      <td>The pathogenic Middle East respiratory syndrom...</td>\n",
              "      <td>18.150488</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>Detection of novel coronaviruses in bats in My...</td>\n",
              "      <td>The recent emergence of bat-borne zoonotic vir...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32271768/;...</td>\n",
              "      <td>Valitutto, Marc T.; Aung, Ohnmar; Tun, Kyaw Ya...</td>\n",
              "      <td>Infectious diseases are considered to be “emer...</td>\n",
              "      <td>18.096310</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>Diarrhea may be underestimated: a missing link...</td>\n",
              "      <td>The outbreak of pneumonia caused by the 2019 N...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.02.0...</td>\n",
              "      <td>Liang, W.; Rao, S.; Xiao, C.; Lin, Z.; Zhang, ...</td>\n",
              "      <td>same cell entry receptor ACE2 with that of SAR...</td>\n",
              "      <td>18.081031</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>SARS‐CoV‐2, an evolutionary perspective of int...</td>\n",
              "      <td>The emergence of SARS‐CoV‐2 has resulted in ne...</td>\n",
              "      <td>https://doi.org/10.1111/eva.12980</td>\n",
              "      <td>Armijos‐Jaramillo, Vinicio; Yeager, Justin; Mu...</td>\n",
              "      <td>The recent emergence of the novel SARS coronav...</td>\n",
              "      <td>17.867609</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>Analyses of spike protein from first deposited...</td>\n",
              "      <td>India has recently started sequencing SARS-CoV...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.28.066985</td>\n",
              "      <td>Begum, Feroza; Mukherjee, Debica; Thagriki, Dl...</td>\n",
              "      <td>SARS-CoV2 (a member of Coronaviruses) outbreak...</td>\n",
              "      <td>17.846854</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>A novel bat coronavirus reveals natural insert...</td>\n",
              "      <td>The unprecedented epidemic of pneumonia caused...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.02.974139</td>\n",
              "      <td>Zhou, Hong; Chen, Xing; Hu, Tao; Li, Juan; Son...</td>\n",
              "      <td>The unprecedented epidemic of pneumonia caused...</td>\n",
              "      <td>17.844430</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>Emerging novel coronavirus (2019-nCoV)—current...</td>\n",
              "      <td>Coronaviruses are the well-known cause of seve...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32036774/;...</td>\n",
              "      <td>Malik, Yashpal Singh; Sircar, Shubhankar; Bhat...</td>\n",
              "      <td>Coronaviruses (CoVs) are well-known causes of ...</td>\n",
              "      <td>17.775618</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>False-Negative Results of Real-Time Reverse-Tr...</td>\n",
              "      <td>The epidemic of 2019 novel coronavirus, later ...</td>\n",
              "      <td>https://doi.org/10.3348/kjr.2020.0146; https:/...</td>\n",
              "      <td>Li, Dasheng; Wang, Dawei; Dong, Jianping; Wang...</td>\n",
              "      <td>At the end of December 2019, pneumonia of unkn...</td>\n",
              "      <td>17.670776</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>No evidence for allelic association between Co...</td>\n",
              "      <td>Background: Coronaviruses (CoV) are a large fa...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.2...</td>\n",
              "      <td>Novelli, A.; Biancolella, M.; Borgiani, P.; Co...</td>\n",
              "      <td>Coronaviruses (CoV) are a large family of viru...</td>\n",
              "      <td>17.622174</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>Development of a rapid test kit for SARS-CoV-2...</td>\n",
              "      <td>We present an example of applying ‘need-driven...</td>\n",
              "      <td>https://doi.org/10.1007/s42242-020-00075-7; ht...</td>\n",
              "      <td>Cui, Zhanfeng; Chang, Hong; Wang, Hui; Lim, Bo...</td>\n",
              "      <td>The COVID-19 pandemic caught almost every coun...</td>\n",
              "      <td>17.576274</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>Genome-wide data inferring the evolution and p...</td>\n",
              "      <td>Since December 2019, coronavirus disease 2019 ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.04.976662</td>\n",
              "      <td>Fang, Bin; Liu, Linlin; Yu, Xiao; Li, Xiang; Y...</td>\n",
              "      <td>On this analysis technique, to precisely decod...</td>\n",
              "      <td>17.410927</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>Genetic Variability of Human Angiotensin-Conve...</td>\n",
              "      <td>There appears to be large regional variations ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.14.041434</td>\n",
              "      <td>Li, Quan; Cao, Zanxia; Rahman, Proton</td>\n",
              "      <td>As of March 24, 2020, the worldwide tracking f...</td>\n",
              "      <td>17.363501</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>The COVID-19 Host Genetics Initiative, a globa...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32404885/;...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>The COVID-19 host genetics initiative is a bot...</td>\n",
              "      <td>17.270880</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>COVID-19: is there a role for immunonutrition,...</td>\n",
              "      <td>In late December 2019 severe acute respiratory...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>Derbyshire, Emma; Delange, Joanne</td>\n",
              "      <td>Coronaviruses are important pathogens to anima...</td>\n",
              "      <td>17.161800</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>42</th>\n",
              "      <td>“Amantadine disrupts lysosomal gene expression...</td>\n",
              "      <td>SARS-coronavirus 2 is the causal agent of the ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.05.026187</td>\n",
              "      <td>Smieszek, Sandra P.; Przychodzen, Bart P; Poly...</td>\n",
              "      <td>Recently a novel type of highly virulent beta-...</td>\n",
              "      <td>17.066529</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>Zoonotic origins of human coronaviruses</td>\n",
              "      <td>Mutation and adaptation have driven the co-evo...</td>\n",
              "      <td>https://doi.org/10.7150/ijbs.45472; https://ww...</td>\n",
              "      <td>Ye, Zi-Wei; Yuan, Shuofeng; Yuen, Kit-San; Fun...</td>\n",
              "      <td>Coronaviruses (CoVs) belong to the family Coro...</td>\n",
              "      <td>17.014703</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>Heterogeneous expression of the SARS-Coronavir...</td>\n",
              "      <td>Rationale Zoonotically transmitted coronavirus...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.22.056127</td>\n",
              "      <td>Bezara, Miguel Ortiz; Thurman, Andrew; Pezzulo...</td>\n",
              "      <td>infection can be fatal, with the risk for incr...</td>\n",
              "      <td>16.993752</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>Susceptible supply limits the role of climate ...</td>\n",
              "      <td>Preliminary evidence suggests that climate may...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.03.20052787</td>\n",
              "      <td>Baker, Rachel E.; Yang, Wenchang; Vecchi, Gabr...</td>\n",
              "      <td>The SARS-CoV-2 pandemic represents a generatio...</td>\n",
              "      <td>16.963317</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>Could India be the origin of next COVID-19 lik...</td>\n",
              "      <td>Abstract The COVID-19 global pandemic is not e...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Minhas, Sachin</td>\n",
              "      <td>It all started from a wet market in Wuhan, the...</td>\n",
              "      <td>16.903808</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>47</th>\n",
              "      <td>“Amantadine disrupts lysosomal gene expression...</td>\n",
              "      <td>Abstract SARS-coronavirus 2 is the causal agen...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32361028/;...</td>\n",
              "      <td>Smieszek, Sandra P.; Przychodzen, Bart P; Poly...</td>\n",
              "      <td>Recently a novel type of highly virulent beta-...</td>\n",
              "      <td>16.896807</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>48</th>\n",
              "      <td>Deep phylogenetic analysis of Orthocoronavirin...</td>\n",
              "      <td>The outbreak of severe acute respiratory syndr...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.12.091199</td>\n",
              "      <td>Sharma, Amresh Kumar; Som, Anup</td>\n",
              "      <td>Coronaviruses are single-stranded RNA virus of...</td>\n",
              "      <td>16.787652</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>49</th>\n",
              "      <td>Prediction analysis of SARS-COV-2 entry in Liv...</td>\n",
              "      <td>Interaction between spike protein of SARS-CoV-...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.08.084327</td>\n",
              "      <td>Praharaj, Manas Ranjan; Garg, Priyanka; Nabi K...</td>\n",
              "      <td>Three large-scale disease outbreaks during the...</td>\n",
              "      <td>16.754176</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1225d3e9-1e16-4bec-9e93-f41bcf89071d')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-1225d3e9-1e16-4bec-9e93-f41bcf89071d button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-1225d3e9-1e16-4bec-9e93-f41bcf89071d');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-c7b38830-ece1-4470-a0d6-ffaeb842fcae\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-c7b38830-ece1-4470-a0d6-ffaeb842fcae')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-c7b38830-ece1-4470-a0d6-ffaeb842fcae button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "res",
              "summary": "{\n  \"name\": \"res\",\n  \"rows\": 50,\n  \"fields\": [\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"Cross\\u2010species transmission of the newly identified coronavirus 2019\\u2010nCoV\",\n          \"Genetic Variability of Human Angiotensin-Converting Enzyme 2 (hACE2) Among Various Ethnic Populations\",\n          \"Diarrhea may be underestimated: a missing link in 2019 novel coronavirus\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 49,\n        \"samples\": [\n          \"The current outbreak of viral pneumonia in the city of Wuhan, China, was caused by a novel coronavirus designated 2019\\u2010nCoV by the World Health Organization, as determined by sequencing the viral RNA genome. Many initial patients were exposed to wildlife animals at the Huanan seafood wholesale market, where poultry, snake, bats, and other farm animals were also sold. To investigate possible virus reservoir, we have carried out comprehensive sequence analysis and comparison in conjunction with relative synonymous codon usage (RSCU) bias among different animal species based on the 2019\\u2010nCoV sequence. Results obtained from our analyses suggest that the 2019\\u2010nCoV may appear to be a recombinant virus between the bat coronavirus and an origin\\u2010unknown coronavirus. The recombination may occurred within the viral spike glycoprotein, which recognizes a cell surface receptor. Additionally, our findings suggest that 2019\\u2010nCoV has most similar genetic information with bat coronovirus and most similar codon usage bias with snake. Taken together, our results suggest that homologous recombination may occur and contribute to the 2019\\u2010nCoV cross\\u2010species transmission.\",\n          \"Abstract The COVID-19 global pandemic is not even over yet but it has already taught us a lot of lessons - the hard way. The vast majority of the global community has blamed the Chinese Illegal wildlife markets for the origin of this pandemic. Through careful scientific analysis, I have explained in this article that we don't need such wildlife markets for these kinds of outbreaks to occur in the future. I have also explained how India which is the second-most populous country in the world, could be the origin of the next outbreak, even though such wildlife markets are either very rare or do not exist at all in India.\",\n          \"The outbreak of severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) in Wuhan city, China in December 2019 and thereafter its spillover across the world has created a global pandemic and public health crisis. Today, it appears as a threat to human civilization. Scientists and medical practitioners across the world are involved to trace out the origin and evolution of SARS-CoV-2 (also called 2019 novel coronavirus and referred as 2019-nCoV), its transmission route, cause of pathogenicity, and possible remedial action. In this work, we aim to find out the origin, evolutionary patternthat led to its pathogenicity and possible transmission pathway of 2019-nCoV. To achieve the aims we conducted a large-scale deep phylogenetic analysis on the 162 complete Orthocoronavirinae genomes consisting of four genera namely Alphacoronavirus, Betacoronavirus, Deltacoronavirus and Gammacoronavirus, their gene trees analysis and subsequently genome and gene recombination analyses. Our analyses revealed that i) bat, pangolin and anteater are the natural hosts of 2019-nCoV, ii) outbreak of 2019-nCoV took place via inter-intra species transmission mode, iii) host-specific adaptive mutation made 2019-nCoV more virulent, and the presence of widespread recombination events led to the evolution of new 2019-nCoV strain and/or could be determinant of its pathogenicity. Highlights Orthocoronavirinae genome phylogeny revealed that bat, pangolin and anteater are natural reservoir hosts of novel coronavirus (2019-nCoV/SARS-CoV-2). Host-specific adaptive mutation occurred among the coronaviruses. Transmission of 2019-nCoV to human took place by inter-intra species mode of transmission. Presence of widespread recombination events led to the evolution of new 2019-nCoV strain and/or could be determinant of its pathogenicity.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"https://www.ncbi.nlm.nih.gov/pubmed/31967321/; https://doi.org/10.1002/jmv.25682\",\n          \"https://doi.org/10.1101/2020.04.14.041434\",\n          \"http://medrxiv.org/cgi/content/short/2020.02.03.20020289v1?rss=1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 49,\n        \"samples\": [\n          \"Ji, Wei; Wang, Wei; Zhao, Xiaofang; Zai, Junjie; Li, Xingguang\",\n          \"Minhas, Sachin\",\n          \"Sharma, Amresh Kumar; Som, Anup\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"China has been the epicenter of emerging and re\\u2010emerging viral infections that continue to stir a global concern. In the last 20 years, China has witnessed several emerging viral diseases, including an avian influenza in 1997,\\n1\\n the severe acute respiratory syndrome (SARS) in 2003,\\n2\\n and a severe fever with thrombocytopenia syndrome (SFTS) in 2010.\\n3\\n The most recent crisis was the outbreak of an ongoing viral pneumonia with unknown etiology in the city of Wuhan, China. On 12 December 2019, Wuhan Municipal Health Commission (WMHC) reported 27 cases of viral pneumonia with 7 of them being critically ill. Most of them had a history of exposure to the virus at the Huanan Seafood Wholesale Market where poultry, bats, snakes; and other wildlife animals were also sold.\\n4\\n On 3 January 2020, WMHC updated the number of cases to a total of 44 with 11 of them in critical condition. On 5 January, the number of cases increased to 59 with 7 critically ill patients. The viral pneumonia outbreak was not caused by severe acute respiratory syndrome coronavirus (SARS\\u2010CoV), Middle East Respiratory Syndrome coronavirus (MERS\\u2010CoV), influenza virus, or adenovirus as determined by laboratory tests.\\n4\\n On 10 January, it was reported that a novel coronavirus designated 2019\\u2010nCoV by the World Health Organization (WHO)\\n5\\n was identified by high\\u2010throughput sequencing of the viral RNA genome, which was released through http://virological.org. More significantly, the newly identified 2019\\u2010CoV has also been isolated from one patient. The availability of viral RNA sequence has made it possible to develop reverse\\u2010transcription polymerase chain reaction (RT\\u2010PCR) methods for the detection of viral RNA in samples from patients and potential hosts.\\n6\\n As a result, 217 patients were confirmed to be infected with the 2019\\u2010nCoV, and 9 patients died as of 20 January 2020. Several patients from Wuhan were also reported in Thailand, Singapore, Hong Kong, South Korea, and Japan. High\\u2010throughput sequencing of viral RNA from patients\\u2019 samples has identified a novel coronavirus designated 2019\\u2010nCoV by the World Health Organization. Currently, a total of 14 full\\u2010length sequences of the 2019\\u2010nCoV were released to GISAID and GeneBank.\\nThe coronavirinae family consists of four genera based on their genetic properties, including genus Alphacoronavirus, genus Betacoronavirus, genus Gammacoronavirus, and genus Deltacoronavirus.\\n7\\n The coronavirus RNA genome (ranging from 26 to 32 kb) is the largest among all RNA viruses.\\n8\\n Coronavirus can infect humans and many different animal species, including swine, cattle, horses, camels, cats, dogs, rodents, birds, bats, rabbits, ferrets, mink, snake, and other wildlife animals.\\n7\\n, \\n9\\n Many coronavirus infections are subclinical.\\n7\\n, \\n9\\n SARS\\u2010CoV and MERS\\u2010CoV belong to the Betacoronavirus genus and are zoonotic pathogens that can cause severe respiratory diseases in humans.\\n7\\n\\n\\nThe outbreak of viral pneumonia in Wuhan is associated with history of exposure to virus reservoir at the Huanan seafood wholesale market, suggesting a possible zoonosis. The seafood market also sold live animals such as snakes, marmots, birds, frogs, and hedgehogs. Currently, there is no evidence suggesting a specific wildlife host as a virus reservoir. Studies of relative synonymous codon usage (RSCU) bias between viruses and their hosts suggested that viruses tends to evolve codon usage bias that is comparable to their hosts.\\n10\\n, \\n11\\n Results from our analysis suggest that 2019\\u2010nCoV has most similar genetic information with bat coronovirus and has most similar codon usage bias with snake. More interestingly, an origin\\u2010unknown homologous recombination may occured within the spike glycoprotein of the 2019\\u2010nCoV,\\n5\\n which may explain its cross\\u2010species transmission, and limited person\\u2010person spread.\\nThe newly sequenced Beta\\u2010coronavirus (MN908947) genome was downloaded from the GenBank database. Five hundred closely related sequences were also downloaded from GenBank. Out of them, 271 genome sequences (>19 000 bp in length) were used in this study together with the above\\u2010described Beta\\u2010coronavirus (2019\\u2010nCoV, MN908947) genome sequence (Table S1). The geographic origins of the sequences were from Bulgaria (n = 1), Canada (n = 2), China (n = 67), Germany (n = 1), Hong Kong (n = 5), Italy (n = 1), Kenya (n = 1), Russia (n = 1), Singapore (n = 24), South Korea (n = 1), Taiwan (n = 11), United Kingdom (n = 2), United States of America (n = 67), and unknown (n = 88). Sequences were aligned using MAFFT v7.222,\\n12\\n followed by manual adjustment using BioEdit v7.2.5.\\n13\\n\\n\\nPhylogenetic trees were constructed using maximum\\u2010likelihood methods and general time\\u2010reversible model of nucleotide substitution with gamma\\u2010distributed rates among sites (GTR+G substitution model) in RAxML v8.0.9.\\n14\\n Support for the inferred relationships was evaluated by a bootstrap analysis with 1000 replicates and trees were midpoint\\u2010rooted.\\nTo investigate the putative parents of the 2019\\u2010nCoV, we performed Similarity and Bootscanning plot analyses based on the Kimura two\\u2010parameter model with a window size of 500 bp, step size of 30 bp using SimPlot v.3.5.1.\\n15\\n We divided our data set into four clades, the newly discovered 2019\\u2010nCoV sequence was grouped as the query sequence. The closest relative coronaviruses (bat\\u2010SL\\u2010CoVZC45 and bat\\u2010SL\\u2010CoVZXC21) obtained from the city of Nanjing, China were grouped as \\u201cClade A.\\u201d The other two coronaviruses (BtCoV/BM48\\u201031/BGR/2008 and BtKY72) from Bulgaria and Kenya were grouped as \\u201cClade B.\\u201d The rest sequences were grouped as \\u201cClade C\\u201d (Figure 1).\\nTo estimate the RSCU bias of the 2019\\u2010nCoV and its potential host(s), All avaliable coding sequences (retaining coding sequences with ATG primer and mutiple of 3 nucleotides, excluding incorrect coding sequences) of the 2019\\u2010nCoV genome (1CDS's, 9672 codons), bat\\u2010SL\\u2010CoVZC45 genome (1CDS's, 9680 codons), Bungarus multicinctus genes (38 CDS's, 5381 codons), Naja atra genes (64 CDS's, 9587 codons), Erinaceus europaeus genome CDS (28947 CDS's, 16717458 codons), Marmota genes (36055CDS's, 21090600 codons), Manis javanica genome CDS (39192 CDS's, 22980491 codons), Rhinolophus sinicus genes (10 CDS's, 8081 codons) and Gallus gallus genome CDS (49453 CDS's, 36086657 codons) from GenBank were calculated with Codon W1.4.2.\\n16\\n, \\n17\\n The RSCU of human genes (40662582 codons) was retrieved from the Codon Usage Database (http://www.kazusa.or.jp/codon/). The relationship among these sequences was calculated using a squared Euclidean distance (dik=\\u2211j=1p(Xij\\u2212Xkj)2), as we previously reported.\\n18\\n A heat map of RSCU was drawn with MeV 4.9.0 software.\\n19\\n The coronavirus and their potential hosts were clustered using a Euclidean distance method.\\nPhylogenetic analysis of 276 coronavirus genomes revealed that the newly identified coronavirus 2019\\u2010nCoV sequence was monophyletic with 100% bootstrap support. The Clade A (bat\\u2010SL\\u2010CoVZC45 and bat\\u2010SL\\u2010CoVZXC21) derived from bats in the city of Nanjing, China between 2015 and 2017 represents the sister lineage to 2019\\u2010nCoV. The Clade B (BtCoV/BM48\\u201031/BGR/2008 and BtKY72) obtained from bats in Bulgaria and Kenya between 2005 and 2007 formed a distinct monophyletic cluster with 100% bootstrap support. The Clade C including 267 coronavirus strains was clustered together with 63% bootstrap support (Figure 1). This suggest that 2019\\u2010nCoV has most similar genetic information with bat coronovirus.\\nHomologous recombination is an important evolutionary force and previous studies have found that homologous recombination occurred in many viruses, including Dengue virus,\\n20\\n human immunodeficiency virus,\\n21\\n hepatitis B virus,\\n22\\n hepatitis C virus,\\n23\\n and classical swine fever virus.\\n18\\n Similarity plot analysis of the 2019\\u2010nCoV revealed that homologous recombination may occurred between Clade A strains (bat\\u2010coronaviruses) and the origin\\u2010unknown isolates, located within the spike glycoprotein that recognizes cell surface receptor (Figure 2). These characteristics indicate that cross\\u2010species transmission may be caused by homologous recombination.\\nAs parasitic microorganism, virus codon usage pattern resembles its host to some extent. The RSCU bias shows that the 2019\\u2010nCoV, bat\\u2010SL\\u2010CoVZC45, and snakes from China have similar synonymous codon usage bias (Figure 3A, Table 1). The squared euclidean distance indicates that the 2019\\u2010nCoV and snakes from China have the highest similarity in synonymous codon usage bias compared to those of bat, bird, Marmota, human, Manis, and hedgehog and (Figure 3B). Two types of snakes, containing B. multicinctus (many\\u2010banded krait) and N. atra (Chinese cobra) were used for RSCU analysis. Squared Euclidean distance between the 2019\\u2010nCoV and B. multicinctus is 13.54. The distance between the 2019\\u2010nCoV and another snake N. atra is 16.69. The distance between the 2019\\u2010nCoV and Rhinolophus sinicus is 23.46. However, the distance between the 2019\\u2010nCoV and other animals is greater than 26, specifically 26.93 for bird, 34.79 for Marmota, 35.36 for human, 36.71 for Manis, and 37.96 for hedgehog. These data suggest that the 2019\\u2010nCoV might more effectively use snake's translation machinery than that of other animals.\\nTwo types of snakes are common in Southeastern China including the city of Wuhan (Figure 4). Geographical distributions of B. multicinctus include Taiwan, the Central and Southern China, Hong Kong, Myanmar (Burma), Laos, and Northern Vietnam.\\n24\\n\\nN. atra is found in Southeastern China, Hong Kong, Northern Laos, Northern Vietnam, and Taiwan.\\n25\\n Snakes were also sold at the Huanan Seafood Wholesale Market where many patients worked or had a history of exposure to wildlife or farm animals.\\nIn this study, we have performed an evolutionary analysis using 272 genomic sequences of coronaviruses obtained from various geographic locations. Our results show that the novel coronavirus sequence obtained from the viral pneumonia outbreak occurring in the city of Wuhan forms a separate group that is highly distinctive to SARS\\u2010CoV. The SARS\\u2010CoV first emerged in China in 2002 and then spread to 37 countries/regions in 2003 and caused a travel\\u2010related global outbreak with 9.6% mortality rate.\\n26\\n More importantly, results from our analysis reveal a homologous recombination may occurred between the bat coronavirus and an origin\\u2010unknown coronavirus within the viral spike glycoprotein gene. Sequence homology analysis of the partial spike glycoprotein genes (1\\u2010783 bp) from the 2019\\u2010nCoV was done through BLAST at the NCBI website. Interestingly, no similar sequence was found with known sequence in the database, suggesting that a putative recombination parent virus was still unknown. Previous study suggested that the recombination of SARS in the spike glycoprotein genes might have mediated the initial cross\\u2010species transmission event from bats to other mammals.\\n27\\n Bootscanning plot analysis (data not shown) suggested that the major parents of the 2019\\u2010nCoV originated from Clade A (bat\\u2010SL\\u2010CoVZC45 and bat\\u2010SL\\u2010CoVZXC21) but formed a monophyletic cluster different from them. Overall, the ancestral origin of the 2019\\u2010nCoV was more likely from divergent host species rather than SARS\\u2010CoV.\\nThe host range of some animal coronaviruses was promiscuous.\\n7\\n They caught our attention only when they caused human diseases such as SARS, MERS, and 2019\\u2010nCoV pneumonia.\\n4\\n, \\n9\\n, \\n28\\n It is critical to determine the animal reservoir of the 2019\\u2010nCoV to understand the molecular mechanism of its cross\\u2010species spread. Homologous recombination within viral structural proteins between coronaviruses from different hosts may be responsible for \\u201ccross\\u2010species\\u201d transmission.\\n27\\n Information obtained from RSCU analysis provides some insights to the question of wildlife animal reservoir although it requires further validation by experimental studies in animal models. Currently, the 2019\\u2010nCoV has not been isolated from animal species although it was obtained from one patient. Identifying and characterizing the animal reservoir for 2019\\u2010nCoV will be helpful for investigation of the recombination and for a better understanding of its person\\u2010to\\u2010person spread among human populations.\\nThe 2019\\u2010nCoV has caused a total of 217 confirmed cases of pneumonia in China as of 20 January 2020 with new patients also reported in Hong Kong, Thailand, Singapore, South Korea, and Japan. Unlike SARS\\u2010CoV, the 2019\\u2010nCoV appeared to initially cause a mild form of viral pneumonia and have limited capability for person\\u2010person spread. This might be due to the recombination occurred within the receptor\\u2010binding glycoprotein. However, there is a concern about its adaptation in humans that may acquire the capability to replicate more efficiently and spread more rapidly via close person\\u2010person contact.\\nIn summary, results derived from our evolutionary analysis suggest that 2019\\u2010nCoV has most similar genetic information with bat coronovirus and has most similar codon usage bias with snake. Additionally, a homologous recombination may occured within the viral receptor\\u2010binding spike glycoprotein, which may determine cross\\u2010species transmission. These novel findings warrant future investigation to experimentally determine if homologous recombination within the spike glycoprotein determine the tropism of the 2019\\u2010nCoV in viral transmission and replication. New information obtained from our evolutionary analysis is highly significant for effective control of the outbreak caused by the 2019\\u2010nCoV\\u2010induced pneumonia.\\nThe authors declare that there are no conflict of interests.\\nWriting: WJ and XL. Data collection: JZ, WW, and XZ. Data analysis: WJ and XL.\\n\",\n          \"As of March 24, 2020, the worldwide tracking for Covid-19 infections reported 1,603,433 cases including 95,716 deaths1. Among the 18 countries that have reported more than 20,000 Covid-19 cases, the total number of cases per population ranged from 3,277 per million people in Spain to 57 per million cases in China and the total death rate among these countries ranged from 330 per million in Spain to 2 per million in China1.\\nThere are numerous potential factors to explain the wide variability in the number of infections and death among the countries including stringency of social isolation and contact tracing, the importation of the SARS-Cov-2, intensity of testing for the virus, as well as the preparedness and capacity of health care system to cope with the pandemic.\\nDifferences among the host, particularly the sex of the patient and presence of selected co-morbid diseases and immunosuppressive state have been identified as risk factors from severe Covid-192,3. Pre-existing medical conditions have been associated with increased prevalence of death including: cardiovascular disease (13.2%); diabetes (9.2%); chronic respiratory disease (8.0%); hypertension (8.4%) and recent diagnosis of cancer (7.6%) 4. Specifically, hypertensive patients had a hazard ratio of 1.70 for death 5 and 3.05 for in-hospital mortality 4. An increased association with the use of angiotensinconverting enzyme inhibitors (ACEIs) and angiotensin II receptor blockers (ARBs) has been reported for severe Covid-19 cases6. A biologically plausible link between SARS-CoV-2 infections and ACE inhibition has been proposed 7.\\nAngiotensin-converting enzyme 2 (ACE2) cleaves peptides within the reninangiotensin system 8. ACE2 is mostly expressed in kidneys and the GI tract, and smaller amounts are found in type 2 pneumocytes in the lung and peripheral blood. The ACE2 receptor serves as the host cell entry for SARS-CoV and more recently it has been shown to be receptor for cellular entry for SARS-CoV-2 9,10. As recently reported by Hoffman et al 11, the cellular entry of SARS-CoV-2 can be blocked by an inhibitor of the cellular serine protease TMPRSS2, which is employed by SARS-CoV-2 for S protein priming. Further attention has been drawn to the ACE2 receptor as antimalarials, which can interfere with ACE2 expression, have resulted in increased clearing of the SARS-CoV-2 clearance among Covid-19 patients 12.\\nTo explore the variability in genetic polymorphisms and expression in human ACE2 (hACE2), we set out to determine if there were any differences between the Asian and Caucasian populations for ACE2 polymorphisms and compare the variability of hACE2 expression in peripheral blood among eight different populations.\\nIn order to investigate whether differences in genetic variations exist between Caucasians and Asians and if these variants can influence the efficiency of cell entry of SARS-CoV-2, we retrieved the variants in the hACE2 from gnomAD v2.1 exomes13. We We also showed these four mutations in the hACE2 X-ray structure as Figure1. Ewald (PME) method was used for calculation the long-range electrostatic interactions.\\nThe LINCS algorithm was used for constraining all covalent bonds. Finally, a 100ns long MD simulation was carried out for each system. The stability of the simulation was checked by computing root mean square deviations (RMSDs). RMSDs of C\\u03b1 were calculated for the mutant K26R/I468V and compared to the wild-type shown in Figure 2 .\\nRMSDs for the complexes with K26R changed little during the 100ns MD simulations.\\nMutant I468V showed RMSD increase during time 20-30 ns, but no obvious RMSD change comparing with wild type after 40ns. The RMSD values for the K26R mutant are slightly higher and structure change compared with wild type.\\nThe effect of missense on protein-ligand complex can be assessed by various experimental technologies19, but are very time-consuming. The in silico binding free energies calculation can be performed to determine the effects of mutations. Here, the binding free energies between S protein and hACE2 (wild type or mutant) were calculated using the g_mmpbsa (Table 2 ). These two mutations K26R/I468V were both predicted to slightly increase the binding free energy and may slightly decrease the binding affinity.\\nWhile, K26R mutated more frequently in Caucasian, and I468V mutated more frequently among Asians. \\nThe and Yoruba in Ibadan, Nigeria (YRI). Raw expression data from Sentrix Human-6\\nExpression BeadChip version 2 were extracted, background-corrected, log2 scaled and quantile normalized. Expression data from the populations with admixture (GIH, LWK, MEX, and MKK) were also normalized for population admixture using EIGENSTRAT22.\\nThe normalized expression data were used for the assessing the expression level for different sets of analysis. The hACE2 expression levels of these 8 populations were showed in Figure- 3.\\nThe hACE2 expression level of CHB population (42 females and 38 males) was then compared to other cohorts using ANOVA analysis, after stratification for sex (Table-3 ).\\nMarginal differences were noted among the populations. There was a statistically higher hACE2 expression in men for LWK and GIH and JPT populations, LWK and YRI populations for females. However the effect sizes were small and its clinical relevance is not clear. \\nThis study that examined the genetic polymorphisms in the hACE2 gene between Caucasians and Asians. We noted that there were four missense mutations, of which K26R and I468V were of most interest based on their location. The propensity for these variants to mutate were different among these two polymorphisms. Among Caucasians, K26R mutated more, while among Asians I468V mutated more frequently. In silico studies have noted that these two mutations may affecting the binding characteristics between S protein and hACE2, however in the absence of further functional studies, the significance of these alternations is not clear.\\nWe also found that there was some variability of genetic expression of hACE2 among various populations, but the magnitude of the differences was small and so it is unclear if this has any impact based on this subtle differences in susceptibility or severity of Covid-19 among various ethnicities. We also did not notice a differential expression of hACE2 between the sexes, even though it is widely reported that males have a worse prognosis than females for complications of Covid-19 4. Our findings are generally consistent with a recently published study by Chen et al 23, that reported the ACE2 expression in Asians was similar to that of other races. Our expression study was conducted in peripheral blood and the results for site specific expression (alveolar, GI or renal tissue) may yield different results.\\nIn conclusion, our results do reveal some differences in genetic polymorphisms between Asians and Caucasians which may potentially alter the binding of the virus to the hACE2 receptor and some subtle variation in genetic expression of hACE2 among different populations. However our findings need to replicated and further in vitro studies should performed, before the significance of these findings can be determined. At present the reasons behind the variability in Covid-19 presentation among the various countries needs to be further explored, as we were unable to provide convincing evidence to suggest there are differences in allele frequency or expression of hACE2 among Caucasians and Asians.\\nConflict of Interest: None of the authors have no conflicts of interest to disclose regarding this manuscript.\\nData used for this study was obtained from public databases.\\n\",\n          \"same cell entry receptor ACE2 with that of SARS-CoV. Consistently, four independent groups have recently provided evidence to support this hypothesis [7] [8] [9] [10] .\\nIn the current study, we found that the incidence of diarrhea differed significantly in three recent reports. As 2019-nCoV receptor ACE2 plays a role in intestinal inflammation and diarrhea, we subsequently evaluated the expression profiles of ACE2 by analyzing the RNA-sequencing and single-cell-sequencing data. We found that ACE2 was highly expressed in the small intestine, especially in the proximal and distal enterocytes. To sum up, we proposed that ACE2-expressing small intestinal epithelium cells might be are more vulnerable to attack by 2019-nCoV and clinicians should be careful when their patients complain of diarrhea since the incidence of diarrhea may be underestimated.\\nThe original information of patients infected by 2019-nCoV infection was retrieved from three recent publications [1] [2] [3] . The 2019-nCoV infection was laboratory-confirmed by real-time RT-PCR and next-generation sequencing. Categorical variables were expressed as number with percentage in parenthesis. All the variables were compared by Fisher's exact test for the three studies as one study has a relatively small sample size 1 .\\nContinuous variables were expressed as median with inter-quartile range (IQR) in parenthesis. A two-sided \\u03b1 of less than 0.05 was considered statistically significant.\\nStatistical analyses were performed using the R software, version 3.5.3.\\nThe expression profiles of ACE2 from normal human tissues were obtained from public database NCBI (https://www.ncbi.nlm.nih.gov/). The original expression data was collected and then plotted by GraphPad Prism 5.\\nThe public single-cell RNA-seq sequencing data (GSE92332) was downloaded from the GEO database. The 10X matrix file of GSE92332_atlas_UMIcounts.txt was used for subsequent analysis based on R package Seurat (Version 3.1.2). In order to filter out low-quality cells and low-quality genes, strict parameters, \\\"min.featrue=1000\\\" and \\\"min.cell=20\\\", were used in the function CreateSeuratObject. The data was subsequently log-normalized by the function NormalizeData with the default parameters. The genes with highly variable expression were identified by the function FindVariableGenes. After the data were processed by the function ScaleData, PCA dimensionality reduction was performed utilizing the function RunPCA. Based on the analysis by the function of JackStraw and ScoreJackStraw, the first 15 PCA components were selected for further . CC-BY 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\nis the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.03.20020289 doi: medRxiv preprint two-dimensional t-distributed stochastic neighbor embedding (tSNE). The setting of k.param was 30 in the function FindNeighbors, and the setting of the resolution was 1.5 in the function FindClusters. In the downloaded original gene-barcode matrix, Adam L.Haber and his coworkers had annotated the cell identities of each barcode. In our analysis, the cell types were identified by the expression of marker genes and the annotation by the original gene-barcode matrix. Gene expression of different cell types was illustrated by the functions of FeaturePlot and VlnPlot.\\nBased on the recent publications, the most common symptoms in patients infected by 2019-nCoV were fever, cough, myalgia, and fatigue. A few infected people have symptoms like diarrhea and headache. However, the incidence of clinical features differs in different reports. As the clinical symptoms may affect the preliminary diagnostic accuracy, in order to establish accurate preliminary diagnostic criteria, we compared the incidence of these common symptoms. We first collected the data from three recent publications and compared the incidence 1-3 . Interestingly, we found that the incidence of leucopenia, fever, and diarrhea in the three studies showed a statistically significant difference (Table 1) . Among these symptoms, diarrhea displayed the smallest p-value (p value= 0.016), suggesting that the criteria for diagnosing diarrhea may differ in different hospitals. Due to the different criteria, clinicians may underestimate the value of this symptom in clinical practice, and it may affect the preliminary diagnostic accuracy.\\nBy using the next-generation sequencing technology, scientists successfully identified and characterized a novel RNA coronavirus with around 29.8 kilobases in length 5 . Phylogenetic analysis of the viral genome indicates that the genome organization of this novel coronavirus is most closely related to that of the bat SARS-CoV. Interestingly, subsequent results showed that 2019-nCoV shared the same cellular entry receptor ACE2 as SARS-CoV 7 .\\nIn terms of the importance of ACE2 in modulating intestinal inflammation and diarrhea 11 , we examined the expression profiles of ACE2 in various human tissues.\\nAccording to the RNA-sequencing data, we found that ACE2 was highly expressed in the human small intestine ( Figure 1 ). Intriguingly, the RNA level of ACE2 was quite low in lung tissues.\\ndata . CC-BY 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity. is the (which was not peer-reviewed) The copyright holder for this preprint . https://doi.org/10.1101/2020.02.03.20020289 doi: medRxiv preprint Given that the distribution of ACE2 receptor may determine the route of 2019-nCoV infection and ACE2 is highly expressed in the small intestine, we next evaluated the expression profiles of ACE2 in different cell populations of the small intestine. Currently, single-cell RNA sequencing (scRNA-Seq) technology is a powerful tool that enables us to investigate the ACE2 expression in various cell types and provide quantitative information at single-cell resolution. Based on the scRNA-Seq data (GSE92332) 12 , we utilized the bioinformatics tools to re-analyze the data. In total, we analyzed 7,216 individual cells derived from the small intestine of normal C57BL/6 mice. Using the unsupervised graph-based clustering, we found that the small intestine tissues consisted of at least eight distinct cell clusters according to their corresponding marker gene expression profiles ( Figure 2A&B ). For instance, the LGR5 gene was highly expressed in the stem cell cluster of the small intestine, and it was significantly reduced in other cell clusters ( Figure 2B ).\\nWe then focused on ACE2 and found that ACE2 was highly expressed in proximal and distal enterocytes ( Figure 2C&D ). Enterocytes, also called intestinal absorptive cells, are simple columnar epithelial cells located in the inner surface of the small and large intestines. Thus, enterocytes are directly exposed to food and foreign pathogens.\\nInterestingly, when we examined expression profiles of another two virus receptors (ANPEP receptor for HCoV-229E virus and DPP4 receptor for MERS-CoV virus), we found that the RNA levels of these two virus entry receptors were also highly expressed in In this study, we displayed that ACE2 was highly expressed in the small intestine, especially in proximal and distal enterocytes. Interestingly, another group has recently reported similar expression pattern in human small intestine one day before our submission 13 . It has been characterized that enterocytes are directly exposed to food and foreign pathogens. Interestingly, other virus receptors like DPP4 displayed similar . CC-BY 4.0 International license It is made available under a author/funder, who has granted medRxiv a license to display the preprint in perpetuity.\\n/n<html><body><table><tr><td>\\u00a0</td><td>Huang et al. </td><td>Chan et al. </td><td>Chen et al. </td><td>Difference </td></tr><tr><td>Characteristics </td><td>\\u00a0</td><td>\\u00a0</td><td>\\u00a0</td><td>\\u00a0</td></tr><tr><td>(41 cases) </td><td>(6 cases) </td><td>(99 cases) </td><td>between groups (p value) </td></tr><tr><td>Age </td><td>49.0(17.0) </td><td>46.2(28.3) </td><td>55.5(17.7) </td><td>-- </td></tr><tr><td>Sex(female) </td><td>11/41(27%) </td><td>3(50%) </td><td>32(32%) </td><td>0.4591 </td></tr><tr><td>Comorbidity </td><td>13/41(32%) </td><td>4(67%) </td><td>50(51%) </td><td>0.0818 </td></tr><tr><td>Fever </td><td>40/41(98%) </td><td>5(83%) </td><td>82(83%) </td><td>0.0345 a </td></tr><tr><td>Cough </td><td>31/41(76%) </td><td>4(67%) </td><td>81(82%) </td><td>0.3829 </td></tr><tr><td>Diarrhoea </td><td>1/38(3%) </td><td>2(33%) </td><td>2(2%) </td><td>0.016 b </td></tr><tr><td>Shortness of breath/ </td><td>\\u00a0</td><td>\\u00a0</td><td>\\u00a0</td><td>\\u00a0</td></tr><tr><td>22/40(55%) </td><td>NA </td><td>31(31%) </td><td>-- </td></tr><tr><td>difficulty in breath </td><td>\\u00a0</td><td>\\u00a0</td><td>\\u00a0</td><td>\\u00a0</td></tr><tr><td>Haemoptysis </td><td>2/39(5%) </td><td>NA </td><td>NA </td><td>-- </td></tr><tr><td>Sputum production </td><td>11/39(28%) </td><td>2(33%) </td><td>NA </td><td>-- </td></tr><tr><td>Myalgia or fatigue </td><td>18/41(44%) </td><td>3(50%) d </td><td>NA </td><td>-- </td></tr><tr><td>Headache </td><td>3/38(8%) </td><td>NA </td><td>8(8%) </td><td>-- </td></tr><tr><td>Leucopenia </td><td>10/40(25%) </td><td>0(0%) </td><td>9(9%) </td><td>0.0443 c </td></tr><tr><td>Platelet count </td><td>2/40(5%) </td><td>0(0%) </td><td>NA </td><td>-- </td></tr></table></body></html>\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 3.2050271670783017,\n        \"min\": 16.75417638247887,\n        \"max\": 30.30655367420809,\n        \"num_unique_values\": 50,\n        \"samples\": [\n          21.253602341141377,\n          17.363501349824503,\n          18.08103119084638\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 99
        }
      ],
      "source": [
        "p = '''\n",
        "Evidence that livestock could be infected (e.g., field surveillance, genetic sequencing, receptor binding) and serve as a reservoir after the epidemic appears to be over.\n",
        "Evidence of whether farmers are infected, and whether farmers could have played a role in the origin.\n",
        "Surveillance of mixed wildlife- livestock farms for SARS-CoV-2 and other coronaviruses in Southeast Asia.\n",
        "Experimental infections to test host range for this pathogen.\n",
        "'''\n",
        "res = searchAndGetAnswer(p)\n",
        "res"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "JxXPtytNYSsA"
      },
      "source": [
        "### Q4: *Animal host(s) and any evidence of continued spill-over to humans*"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 100,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "VENWzTvKYSsA",
        "outputId": "0fd542f8-2849-4f2d-9d78-f71237462901"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No valid answers found.\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                title  \\\n",
              "0   Spillover of SARS-CoV-2 into novel wild hosts ...   \n",
              "1   Are pangolins the intermediate host of the 201...   \n",
              "2   Are pangolins the intermediate host of the 201...   \n",
              "3   Exceptional diversity and selection pressure o...   \n",
              "4   COVID-19: Epidemiology, Evolution, and Cross-D...   \n",
              "5   Severe acute respiratory syndrome-related coro...   \n",
              "6              Horses as a Crucial Part of One Health   \n",
              "7   Emerging novel coronavirus (2019-nCoV)—current...   \n",
              "8   The species Severe acute respiratory syndrome-...   \n",
              "9   Emergence of SARS-CoV-2 through Recombination ...   \n",
              "10  Detection of novel coronaviruses in bats in My...   \n",
              "11  A tug-of-war between severe acute respiratory ...   \n",
              "12            A deadly spillover: SARS-CoV-2 outbreak   \n",
              "13  COVID-19: the First Documented Coronavirus Pan...   \n",
              "14  Coronaviruses: a paradigm of new emerging zoon...   \n",
              "15  Can the Coronavirus Disease 2019 (COVID-19) Af...   \n",
              "16  Personal protective equipment (PPE) for vitreo...   \n",
              "17  Global mapping of landscape fragmentation, hum...   \n",
              "18  Potential host range of multiple SARS-like cor...   \n",
              "19  A novel bat coronavirus closely related to SAR...   \n",
              "20            Zoonotic origins of human coronaviruses   \n",
              "21  Ecological processes underlying the emergence ...   \n",
              "22  Origin and Evolution of the 2019 Novel Coronav...   \n",
              "23  Interactions of coronaviruses with ACE2, angio...   \n",
              "24  Role of changes in SARS-CoV-2 spike protein in...   \n",
              "25  Predicting the angiotensin converting enzyme 2...   \n",
              "26  Human coronaviruses with emphasis on the COVID...   \n",
              "27  Bioaerosol Sampling at a Live Animal Market in...   \n",
              "28  Extracellular superoxide dismutase, a molecula...   \n",
              "29  Deep phylogenetic analysis of Orthocoronavirin...   \n",
              "30  Evolutionary origins of the SARS-CoV-2 sarbeco...   \n",
              "31  Development of remdesivir repositioning as a n...   \n",
              "32  VOLATILITY SPILLOVER AROUND PRICE LIMITS IN AN...   \n",
              "33  Receptor Recognition by the Novel Coronavirus ...   \n",
              "34  Identification of 2019-nCoV related coronaviru...   \n",
              "35  Susceptibility of ferrets, cats, dogs, and oth...   \n",
              "36  Susceptibility of ferrets, cats, dogs, and dif...   \n",
              "37  Structural and biochemical characterization of...   \n",
              "38  Structural and biochemical characterization of...   \n",
              "39  A Multibasic Cleavage Site in the Spike Protei...   \n",
              "40  Comparative computational analysis of SARS-CoV...   \n",
              "41  Synergistic China–US Ecological Research is Es...   \n",
              "42  A new coronavirus associated with human respir...   \n",
              "43  SARS‐CoV‐2 spike protein favors ACE2 from Bovi...   \n",
              "44  Novel Immunoglobulin Domain Proteins Provide I...   \n",
              "45  Novel NGS Pipeline for Virus Discovery from a ...   \n",
              "46                      The hard problem of AI rights   \n",
              "47  Broad and differential animal ACE2 receptor us...   \n",
              "48  Pangolins Lack IFIH1/MDA5, a Cytoplasmic RNA S...   \n",
              "49                           New Virus, New Challenge   \n",
              "\n",
              "                                             abstract  \\\n",
              "0   Abstract There is evidence that the current ou...   \n",
              "1   The outbreak of 2019-nCoV pneumonia (COVID-19)...   \n",
              "2   The outbreak of a novel corona Virus Disease 2...   \n",
              "3   Pandemics originating from pathogen transmissi...   \n",
              "4   The recent outbreak of COVID-19 in Wuhan turne...   \n",
              "5   The present outbreak of lower respiratory trac...   \n",
              "6   One Health (OH) is a crucial concept, where th...   \n",
              "7   Coronaviruses are the well-known cause of seve...   \n",
              "8   The present outbreak of a coronavirus-associat...   \n",
              "9   COVID-19 has become a global pandemic caused b...   \n",
              "10  The recent emergence of bat-borne zoonotic vir...   \n",
              "11  World Health Organization has declared the ong...   \n",
              "12                                                NaN   \n",
              "13  The novel human coronavirus disease COVID-19 h...   \n",
              "14  A novel type of coronavirus (2019-nCoV) infect...   \n",
              "15  In December 2019, a novel coronavirus (CoV) ep...   \n",
              "16  SARS-CoV-2 is the recently discovered virus wh...   \n",
              "17  Everyone wants to know when the next pandemic ...   \n",
              "18  The severe acute respiratory syndrome coronavi...   \n",
              "19  Summary The unprecedented pandemic of pneumoni...   \n",
              "20  Mutation and adaptation have driven the co-evo...   \n",
              "21  Pathogens originating from wildlife (zoonoses)...   \n",
              "22  As of today, the intermediate host of 2019-nCo...   \n",
              "23  The rapid spread of a novel coronavirus, sever...   \n",
              "24  Many human viral diseases are a consequence of...   \n",
              "25  Abstract SARS-CoV-2, the newly identified huma...   \n",
              "26  Human coronaviruses are known as respiratory p...   \n",
              "27  From October to December 2018, periodic bioaer...   \n",
              "28  Extracellular superoxide dismutase (EcSOD) is ...   \n",
              "29  The outbreak of severe acute respiratory syndr...   \n",
              "30  There are outstanding evolutionary questions o...   \n",
              "31  Severe acute respiratory syndrome coronavirus ...   \n",
              "32  Abstract The intraday volatility effects of pr...   \n",
              "33  Recently, a novel coronavirus (2019-nCoV) has ...   \n",
              "34  The ongoing outbreak of viral pneumonia in Chi...   \n",
              "35  Severe acute respiratory syndrome coronavirus ...   \n",
              "36  Severe acute respiratory syndrome coronavirus ...   \n",
              "37  Summary The ongoing global pandemic of coronav...   \n",
              "38  The ongoing global pandemic of coronavirus dis...   \n",
              "39  Summary The pandemic coronavirus SARS-CoV-2 th...   \n",
              "40  Abstract Several research lines are currently ...   \n",
              "41  The risk of a zoonotic pandemic disease threat...   \n",
              "42  Emerging infectious diseases, such as severe a...   \n",
              "43  Severe acute respiratory syndrome coronavirus ...   \n",
              "44  A novel coronavirus (SARS-CoV-2) is the causat...   \n",
              "45  The study of the microbiome data holds great p...   \n",
              "46  In the past few years, the subject of AI right...   \n",
              "47  The COVID-19 pandemic has caused an unpreceden...   \n",
              "48  Zoonotic infections are an imminent threat to ...   \n",
              "49  The ongoing COVID-19 pandemic has made us to c...   \n",
              "\n",
              "                                                  url  \\\n",
              "0   https://api.elsevier.com/content/article/pii/S...   \n",
              "1           https://doi.org/10.1101/2020.02.18.954628   \n",
              "2   https://www.ncbi.nlm.nih.gov/pubmed/32407364/;...   \n",
              "3           https://doi.org/10.1101/2020.04.20.051656   \n",
              "4   https://www.ncbi.nlm.nih.gov/pubmed/32359479/;...   \n",
              "5           https://doi.org/10.1101/2020.02.07.937862   \n",
              "6   https://www.ncbi.nlm.nih.gov/pubmed/32121327/;...   \n",
              "7   https://www.ncbi.nlm.nih.gov/pubmed/32036774/;...   \n",
              "8   https://doi.org/10.1038/s41564-020-0695-z; htt...   \n",
              "9           https://doi.org/10.1101/2020.03.20.000885   \n",
              "10  https://www.ncbi.nlm.nih.gov/pubmed/32271768/;...   \n",
              "11  https://doi.org/10.1080/22221751.2020.1736644;...   \n",
              "12  https://www.ncbi.nlm.nih.gov/pubmed/32321324/;...   \n",
              "13  https://www.ncbi.nlm.nih.gov/pubmed/32387617/;...   \n",
              "14  https://doi.org/10.1093/femspd/ftaa006; https:...   \n",
              "15  https://doi.org/10.1080/09273948.2020.1738501;...   \n",
              "16  https://www.ncbi.nlm.nih.gov/pubmed/32398849/;...   \n",
              "17  https://doi.org/10.1007/s10460-020-10104-x; ht...   \n",
              "18          https://doi.org/10.1101/2020.04.10.032342   \n",
              "19  https://api.elsevier.com/content/article/pii/S...   \n",
              "20  https://doi.org/10.7150/ijbs.45472; https://ww...   \n",
              "21          https://doi.org/10.1101/2020.04.24.057430   \n",
              "22  https://www.ncbi.nlm.nih.gov/pubmed/32011673/;...   \n",
              "23  https://www.ncbi.nlm.nih.gov/pubmed/32341442/;...   \n",
              "24  https://www.ncbi.nlm.nih.gov/pubmed/32210742/;...   \n",
              "25  https://www.ncbi.nlm.nih.gov/pubmed/32199943/;...   \n",
              "26  https://www.ncbi.nlm.nih.gov/pubmed/32399479/;...   \n",
              "27  https://doi.org/10.1093/ofid/ofaa134; https://...   \n",
              "28  https://doi.org/10.1016/j.redox.2020.101508; h...   \n",
              "29          https://doi.org/10.1101/2020.05.12.091199   \n",
              "30          https://doi.org/10.1101/2020.03.30.015008   \n",
              "31  https://doi.org/10.1080/07391102.2020.1767210;...   \n",
              "32  https://www.sciencedirect.com/science/article/...   \n",
              "33  https://doi.org/10.1128/jvi.00127-20; https://...   \n",
              "34          https://doi.org/10.1101/2020.02.13.945485   \n",
              "35  https://doi.org/10.1126/science.abb7015; https...   \n",
              "36          https://doi.org/10.1101/2020.03.30.015347   \n",
              "37  https://www.sciencedirect.com/science/article/...   \n",
              "38          https://doi.org/10.1101/2020.04.23.057265   \n",
              "39  https://www.sciencedirect.com/science/article/...   \n",
              "40  https://www.ncbi.nlm.nih.gov/pubmed/32302675/;...   \n",
              "41  https://doi.org/10.1007/s10393-020-01471-2; ht...   \n",
              "42  https://www.ncbi.nlm.nih.gov/pubmed/32015508/;...   \n",
              "43  https://doi.org/10.1002/jmv.25817; https://www...   \n",
              "44          https://doi.org/10.1101/2020.03.04.977736   \n",
              "45          https://doi.org/10.1101/2020.05.07.082107   \n",
              "46  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   \n",
              "47          https://doi.org/10.1101/2020.04.19.048710   \n",
              "48           https://doi.org/10.3389/fimmu.2020.00939   \n",
              "49  https://www.sciencedirect.com/science/article/...   \n",
              "\n",
              "                                              authors  \\\n",
              "0                 Franklin, Alan B.; Bevins, Sarah N.   \n",
              "1   Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...   \n",
              "2   Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...   \n",
              "3      Frank, Hannah K.; Enard, David; Boyd, Scott D.   \n",
              "4   Sun, Jiumeng; He, Wan-Ting; Wang, Lifang; Lai,...   \n",
              "5   Gorbalenya, Alexander E.; Baker, Susan C.; Bar...   \n",
              "6   Lönker, Nelly Sophie; Fechner, Kim; Abd El Wah...   \n",
              "7   Malik, Yashpal Singh; Sircar, Shubhankar; Bhat...   \n",
              "8                                                 NaN   \n",
              "9   Li, Xiaojun; Giorgi, Elena E.; Marichann, Manu...   \n",
              "10  Valitutto, Marc T.; Aung, Ohnmar; Tun, Kyaw Ya...   \n",
              "11  Fung, Sin-Yee; Yuen, Kit-San; Ye, Zi-Wei; Chan...   \n",
              "12  Mori, Mattia; Capasso, Clemente; Carta, Fabriz...   \n",
              "13         Liu, Yen-Chin; Kuo, Rei-Lin; Shih, Shin-Ru   \n",
              "14  Salata, Cristiano; Calistri, Arianna; Parolin,...   \n",
              "15                        Seah, Ivan; Agrawal, Rupesh   \n",
              "16  Chandra, Aman; Haynes, Richard; Burdon, Michae...   \n",
              "17                            Bloomfield, Laura S. P.   \n",
              "18  Li, Yujun; Wang, Haimin; Tang, Xiaojuan; Ma, D...   \n",
              "19  Zhou, Hong; Chen, Xing; Hu, Tao; Li, Juan; Son...   \n",
              "20  Ye, Zi-Wei; Yuan, Shuofeng; Yuen, Kit-San; Fun...   \n",
              "21  Guth, Sarah; Hanley, Kathryn; Althouse, Benjam...   \n",
              "22  Zhang, Liangsheng; Shen, Fu-ming; Chen, Fei; L...   \n",
              "23                          Kai, Hisashi; Kai, Mamiko   \n",
              "24  Ortega, Joseph Thomas; Serrano, Maria Luisa; P...   \n",
              "25  Qiu, Ye; Zhao, Yuan-Bo; Wang, Qiong; Li, Jin-Y...   \n",
              "26    Bahadur, Saraj; Long, Wenxing; Shuaib, Muhammad   \n",
              "27  Wang, Xinye; Bailey, Emily S; Qi, Xian; Yu, Hu...   \n",
              "28                    Yan, Zhen; Spaulding, Hannah R.   \n",
              "29                    Sharma, Amresh Kumar; Som, Anup   \n",
              "30  Boni, Maciej F; Lemey, Philippe; Jiang, Xiaowe...   \n",
              "31  Babadaei, Mohammad Mahdi Nejadi; Hasan, Anwaru...   \n",
              "32  Aktas, Osman Ulas; Kryzanowski, Lawrence; Zhan...   \n",
              "33  Wan, Yushun; Shang, Jian; Graham, Rachel; Bari...   \n",
              "34  Lam, Tommy Tsan-Yuk; Shum, Marcus Ho-Hin; Zhu,...   \n",
              "35  Shi, Jianzhong; Wen, Zhiyuan; Zhong, Gongxun; ...   \n",
              "36  Shi, Jianzhong; Wen, Zhiyuan; Zhong, Gongxun; ...   \n",
              "37  Peng, Qi; Peng, Ruchao; Yuan, Bin; Zhao, Jingr...   \n",
              "38  Peng, Qi; Peng, Ruchao; Yuan, Bin; Zhao, Jingr...   \n",
              "39  Hoffmann, Markus; Kleine-Weber, Hannah; Pöhlma...   \n",
              "40  Tilocca, Bruno; Soggiu, Alessio; Sanguinetti, ...   \n",
              "41  Smiley Evans, Tierra; Shi, Zhengli; Boots, Mic...   \n",
              "42  Wu, Fan; Zhao, Su; Yu, Bin; Chen, Yan-Mei; Wan...   \n",
              "43  Luan, Junwen; Jin, Xiaolu; Lu, Yue; Zhang, Lei...   \n",
              "44  Tan, Yongjun; Schneider, Theresa; Leong, Matth...   \n",
              "45  Plyusnin, Ilya; Kant, Ravi; Jääskeläinen, Anne...   \n",
              "46                                 Andreotta, Adam J.   \n",
              "47  Zhao, Xuesen; Chen, Danying; Szabla, Robert; Z...   \n",
              "48  Fischer, Heinz; Tschachler, Erwin; Eckhart, Le...   \n",
              "49                                            Shi, Yi   \n",
              "\n",
              "                                                 text      Score  \n",
              "0   J o u r n a l P r e -p r o o f 2 infected huma...  22.379831  \n",
              "1   In December 2019, there was an outbreak of pne...  15.671288  \n",
              "2   In December 2019, there was an outbreak of pne...  14.773573  \n",
              "3   because studies only examine a small subset of...  14.143423  \n",
              "4   In December 2019, a cluster of pneumonia with ...  12.735855  \n",
              "5   Is the outbreak of an infectious disease cause...  12.127306  \n",
              "6   One Health (OH) is a holistic approach which d...  12.061077  \n",
              "7   Coronaviruses (CoVs) are well-known causes of ...  11.823325  \n",
              "8   Understanding the cause of a specific disease ...  11.678412  \n",
              "9   The severe respiratory disease COVID-19 was fi...  11.136798  \n",
              "10  Infectious diseases are considered to be “emer...  11.029762  \n",
              "11  Avian infectious bronchitis virus was the firs...  11.006490  \n",
              "12  Coronaviruses (CoV) are a family of viruses th...  10.810611  \n",
              "13  Currently, people all over the world have been...  10.800136  \n",
              "14  Coronaviruses (CoVs) are important pathogens f...  10.766239  \n",
              "15  CoVs belong to the subfamily Coronavirinae, in...  10.757793  \n",
              "16  SARS-CoV-2 is the recently discovered virus wh...  10.681540  \n",
              "17  Everyone wants to know when the next pandemic ...  10.462016  \n",
              "18  Two subgroups of coronaviruses, including alph...  10.385967  \n",
              "19  novel bat coronavirus novel bat coronavirus no...  10.159164  \n",
              "20  Coronaviruses (CoVs) belong to the family Coro...  10.076111  \n",
              "21  Humans have frequently enabled pathogens to ov...   9.767405  \n",
              "22  To the Editor-The 2019 novel coronavirus disea...   9.487336  \n",
              "23  A novel coronavirus, named severe acute respir...   9.474802  \n",
              "24  Severe Acute Respiratory Syndrome Coronavirus ...   9.440461  \n",
              "25  The aa sequences of ACE2s of Homo sapiens (hum...   9.105582  \n",
              "26  Coronaviruses are positive-standard enveloped ...   9.105582  \n",
              "27  Bioaerosol samples were obtained from a small ...   9.099620  \n",
              "28  Oxygen is indispensable for life; however, und...   8.950877  \n",
              "29  Coronaviruses are single-stranded RNA virus of...   8.895972  \n",
              "30  In December 2019, a cluster of pneumonia cases...   8.855392  \n",
              "31  The new coronavirus (CoV), known as Severe Acu...   8.771147  \n",
              "32  We gratefully acknowledge financial support fr...   8.706396  \n",
              "33  A novel coronavirus (2019-nCoV) from Wuhan, Ch...   8.692931  \n",
              "34  ). We were also 72 able to successfully isolat...   8.533306  \n",
              "35  In late December 2019, an unusual pneumonia em...   8.464124  \n",
              "36  In late December 2019, an unusual pneumonia em...   8.464124  \n",
              "37  In the end of 2019, a novel coronavirus (2019-...   8.375391  \n",
              "38  In the end of 2019, a novel coronavirus (2019-...   8.324044  \n",
              "39  It is believed that the severe acute respirato...   8.240075  \n",
              "40  The virus species and accession number of the ...   8.151314  \n",
              "41  The catastrophic health and economic impacts o...   8.142541  \n",
              "42  The patient studied was a 41-year-old man with...   8.099710  \n",
              "43  Coronavirus disease 2019 (COVID‐19), which was...   8.028711  \n",
              "44  Nidoviruses are an ancient group of lipid-enve...   7.934869  \n",
              "45  Our ability to produce sequence data in the ra...   7.888766  \n",
              "46  Artificially intelligent robots, machines, and...   7.776895  \n",
              "47  human population remains mysterious, although ...   7.667037  \n",
              "48  Emerging infectious diseases represent a major...   7.666060  \n",
              "49  Yi Shi 1, * The ongoing COVID-19 pandemic has ...   7.638880  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-ab47fcb8-db55-44b4-96dc-1703e3795cf5\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>title</th>\n",
              "      <th>abstract</th>\n",
              "      <th>url</th>\n",
              "      <th>authors</th>\n",
              "      <th>text</th>\n",
              "      <th>Score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Spillover of SARS-CoV-2 into novel wild hosts ...</td>\n",
              "      <td>Abstract There is evidence that the current ou...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Franklin, Alan B.; Bevins, Sarah N.</td>\n",
              "      <td>J o u r n a l P r e -p r o o f 2 infected huma...</td>\n",
              "      <td>22.379831</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Are pangolins the intermediate host of the 201...</td>\n",
              "      <td>The outbreak of 2019-nCoV pneumonia (COVID-19)...</td>\n",
              "      <td>https://doi.org/10.1101/2020.02.18.954628</td>\n",
              "      <td>Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...</td>\n",
              "      <td>In December 2019, there was an outbreak of pne...</td>\n",
              "      <td>15.671288</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>Are pangolins the intermediate host of the 201...</td>\n",
              "      <td>The outbreak of a novel corona Virus Disease 2...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32407364/;...</td>\n",
              "      <td>Liu, Ping; Jiang, Jing-Zhe; Wan, Xiu-Feng; Hua...</td>\n",
              "      <td>In December 2019, there was an outbreak of pne...</td>\n",
              "      <td>14.773573</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Exceptional diversity and selection pressure o...</td>\n",
              "      <td>Pandemics originating from pathogen transmissi...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.20.051656</td>\n",
              "      <td>Frank, Hannah K.; Enard, David; Boyd, Scott D.</td>\n",
              "      <td>because studies only examine a small subset of...</td>\n",
              "      <td>14.143423</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>COVID-19: Epidemiology, Evolution, and Cross-D...</td>\n",
              "      <td>The recent outbreak of COVID-19 in Wuhan turne...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32359479/;...</td>\n",
              "      <td>Sun, Jiumeng; He, Wan-Ting; Wang, Lifang; Lai,...</td>\n",
              "      <td>In December 2019, a cluster of pneumonia with ...</td>\n",
              "      <td>12.735855</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>Severe acute respiratory syndrome-related coro...</td>\n",
              "      <td>The present outbreak of lower respiratory trac...</td>\n",
              "      <td>https://doi.org/10.1101/2020.02.07.937862</td>\n",
              "      <td>Gorbalenya, Alexander E.; Baker, Susan C.; Bar...</td>\n",
              "      <td>Is the outbreak of an infectious disease cause...</td>\n",
              "      <td>12.127306</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>Horses as a Crucial Part of One Health</td>\n",
              "      <td>One Health (OH) is a crucial concept, where th...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32121327/;...</td>\n",
              "      <td>Lönker, Nelly Sophie; Fechner, Kim; Abd El Wah...</td>\n",
              "      <td>One Health (OH) is a holistic approach which d...</td>\n",
              "      <td>12.061077</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Emerging novel coronavirus (2019-nCoV)—current...</td>\n",
              "      <td>Coronaviruses are the well-known cause of seve...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32036774/;...</td>\n",
              "      <td>Malik, Yashpal Singh; Sircar, Shubhankar; Bhat...</td>\n",
              "      <td>Coronaviruses (CoVs) are well-known causes of ...</td>\n",
              "      <td>11.823325</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>The species Severe acute respiratory syndrome-...</td>\n",
              "      <td>The present outbreak of a coronavirus-associat...</td>\n",
              "      <td>https://doi.org/10.1038/s41564-020-0695-z; htt...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>Understanding the cause of a specific disease ...</td>\n",
              "      <td>11.678412</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>Emergence of SARS-CoV-2 through Recombination ...</td>\n",
              "      <td>COVID-19 has become a global pandemic caused b...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.20.000885</td>\n",
              "      <td>Li, Xiaojun; Giorgi, Elena E.; Marichann, Manu...</td>\n",
              "      <td>The severe respiratory disease COVID-19 was fi...</td>\n",
              "      <td>11.136798</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>Detection of novel coronaviruses in bats in My...</td>\n",
              "      <td>The recent emergence of bat-borne zoonotic vir...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32271768/;...</td>\n",
              "      <td>Valitutto, Marc T.; Aung, Ohnmar; Tun, Kyaw Ya...</td>\n",
              "      <td>Infectious diseases are considered to be “emer...</td>\n",
              "      <td>11.029762</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>A tug-of-war between severe acute respiratory ...</td>\n",
              "      <td>World Health Organization has declared the ong...</td>\n",
              "      <td>https://doi.org/10.1080/22221751.2020.1736644;...</td>\n",
              "      <td>Fung, Sin-Yee; Yuen, Kit-San; Ye, Zi-Wei; Chan...</td>\n",
              "      <td>Avian infectious bronchitis virus was the firs...</td>\n",
              "      <td>11.006490</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>A deadly spillover: SARS-CoV-2 outbreak</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32321324/;...</td>\n",
              "      <td>Mori, Mattia; Capasso, Clemente; Carta, Fabriz...</td>\n",
              "      <td>Coronaviruses (CoV) are a family of viruses th...</td>\n",
              "      <td>10.810611</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>COVID-19: the First Documented Coronavirus Pan...</td>\n",
              "      <td>The novel human coronavirus disease COVID-19 h...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32387617/;...</td>\n",
              "      <td>Liu, Yen-Chin; Kuo, Rei-Lin; Shih, Shin-Ru</td>\n",
              "      <td>Currently, people all over the world have been...</td>\n",
              "      <td>10.800136</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>Coronaviruses: a paradigm of new emerging zoon...</td>\n",
              "      <td>A novel type of coronavirus (2019-nCoV) infect...</td>\n",
              "      <td>https://doi.org/10.1093/femspd/ftaa006; https:...</td>\n",
              "      <td>Salata, Cristiano; Calistri, Arianna; Parolin,...</td>\n",
              "      <td>Coronaviruses (CoVs) are important pathogens f...</td>\n",
              "      <td>10.766239</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>Can the Coronavirus Disease 2019 (COVID-19) Af...</td>\n",
              "      <td>In December 2019, a novel coronavirus (CoV) ep...</td>\n",
              "      <td>https://doi.org/10.1080/09273948.2020.1738501;...</td>\n",
              "      <td>Seah, Ivan; Agrawal, Rupesh</td>\n",
              "      <td>CoVs belong to the subfamily Coronavirinae, in...</td>\n",
              "      <td>10.757793</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>Personal protective equipment (PPE) for vitreo...</td>\n",
              "      <td>SARS-CoV-2 is the recently discovered virus wh...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32398849/;...</td>\n",
              "      <td>Chandra, Aman; Haynes, Richard; Burdon, Michae...</td>\n",
              "      <td>SARS-CoV-2 is the recently discovered virus wh...</td>\n",
              "      <td>10.681540</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>Global mapping of landscape fragmentation, hum...</td>\n",
              "      <td>Everyone wants to know when the next pandemic ...</td>\n",
              "      <td>https://doi.org/10.1007/s10460-020-10104-x; ht...</td>\n",
              "      <td>Bloomfield, Laura S. P.</td>\n",
              "      <td>Everyone wants to know when the next pandemic ...</td>\n",
              "      <td>10.462016</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>Potential host range of multiple SARS-like cor...</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.10.032342</td>\n",
              "      <td>Li, Yujun; Wang, Haimin; Tang, Xiaojuan; Ma, D...</td>\n",
              "      <td>Two subgroups of coronaviruses, including alph...</td>\n",
              "      <td>10.385967</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>A novel bat coronavirus closely related to SAR...</td>\n",
              "      <td>Summary The unprecedented pandemic of pneumoni...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Zhou, Hong; Chen, Xing; Hu, Tao; Li, Juan; Son...</td>\n",
              "      <td>novel bat coronavirus novel bat coronavirus no...</td>\n",
              "      <td>10.159164</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>Zoonotic origins of human coronaviruses</td>\n",
              "      <td>Mutation and adaptation have driven the co-evo...</td>\n",
              "      <td>https://doi.org/10.7150/ijbs.45472; https://ww...</td>\n",
              "      <td>Ye, Zi-Wei; Yuan, Shuofeng; Yuen, Kit-San; Fun...</td>\n",
              "      <td>Coronaviruses (CoVs) belong to the family Coro...</td>\n",
              "      <td>10.076111</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>Ecological processes underlying the emergence ...</td>\n",
              "      <td>Pathogens originating from wildlife (zoonoses)...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.24.057430</td>\n",
              "      <td>Guth, Sarah; Hanley, Kathryn; Althouse, Benjam...</td>\n",
              "      <td>Humans have frequently enabled pathogens to ov...</td>\n",
              "      <td>9.767405</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>Origin and Evolution of the 2019 Novel Coronav...</td>\n",
              "      <td>As of today, the intermediate host of 2019-nCo...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32011673/;...</td>\n",
              "      <td>Zhang, Liangsheng; Shen, Fu-ming; Chen, Fei; L...</td>\n",
              "      <td>To the Editor-The 2019 novel coronavirus disea...</td>\n",
              "      <td>9.487336</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>Interactions of coronaviruses with ACE2, angio...</td>\n",
              "      <td>The rapid spread of a novel coronavirus, sever...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32341442/;...</td>\n",
              "      <td>Kai, Hisashi; Kai, Mamiko</td>\n",
              "      <td>A novel coronavirus, named severe acute respir...</td>\n",
              "      <td>9.474802</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>Role of changes in SARS-CoV-2 spike protein in...</td>\n",
              "      <td>Many human viral diseases are a consequence of...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32210742/;...</td>\n",
              "      <td>Ortega, Joseph Thomas; Serrano, Maria Luisa; P...</td>\n",
              "      <td>Severe Acute Respiratory Syndrome Coronavirus ...</td>\n",
              "      <td>9.440461</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>Predicting the angiotensin converting enzyme 2...</td>\n",
              "      <td>Abstract SARS-CoV-2, the newly identified huma...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32199943/;...</td>\n",
              "      <td>Qiu, Ye; Zhao, Yuan-Bo; Wang, Qiong; Li, Jin-Y...</td>\n",
              "      <td>The aa sequences of ACE2s of Homo sapiens (hum...</td>\n",
              "      <td>9.105582</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>Human coronaviruses with emphasis on the COVID...</td>\n",
              "      <td>Human coronaviruses are known as respiratory p...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32399479/;...</td>\n",
              "      <td>Bahadur, Saraj; Long, Wenxing; Shuaib, Muhammad</td>\n",
              "      <td>Coronaviruses are positive-standard enveloped ...</td>\n",
              "      <td>9.105582</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>Bioaerosol Sampling at a Live Animal Market in...</td>\n",
              "      <td>From October to December 2018, periodic bioaer...</td>\n",
              "      <td>https://doi.org/10.1093/ofid/ofaa134; https://...</td>\n",
              "      <td>Wang, Xinye; Bailey, Emily S; Qi, Xian; Yu, Hu...</td>\n",
              "      <td>Bioaerosol samples were obtained from a small ...</td>\n",
              "      <td>9.099620</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>Extracellular superoxide dismutase, a molecula...</td>\n",
              "      <td>Extracellular superoxide dismutase (EcSOD) is ...</td>\n",
              "      <td>https://doi.org/10.1016/j.redox.2020.101508; h...</td>\n",
              "      <td>Yan, Zhen; Spaulding, Hannah R.</td>\n",
              "      <td>Oxygen is indispensable for life; however, und...</td>\n",
              "      <td>8.950877</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>Deep phylogenetic analysis of Orthocoronavirin...</td>\n",
              "      <td>The outbreak of severe acute respiratory syndr...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.12.091199</td>\n",
              "      <td>Sharma, Amresh Kumar; Som, Anup</td>\n",
              "      <td>Coronaviruses are single-stranded RNA virus of...</td>\n",
              "      <td>8.895972</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>Evolutionary origins of the SARS-CoV-2 sarbeco...</td>\n",
              "      <td>There are outstanding evolutionary questions o...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.30.015008</td>\n",
              "      <td>Boni, Maciej F; Lemey, Philippe; Jiang, Xiaowe...</td>\n",
              "      <td>In December 2019, a cluster of pneumonia cases...</td>\n",
              "      <td>8.855392</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>Development of remdesivir repositioning as a n...</td>\n",
              "      <td>Severe acute respiratory syndrome coronavirus ...</td>\n",
              "      <td>https://doi.org/10.1080/07391102.2020.1767210;...</td>\n",
              "      <td>Babadaei, Mohammad Mahdi Nejadi; Hasan, Anwaru...</td>\n",
              "      <td>The new coronavirus (CoV), known as Severe Acu...</td>\n",
              "      <td>8.771147</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>VOLATILITY SPILLOVER AROUND PRICE LIMITS IN AN...</td>\n",
              "      <td>Abstract The intraday volatility effects of pr...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Aktas, Osman Ulas; Kryzanowski, Lawrence; Zhan...</td>\n",
              "      <td>We gratefully acknowledge financial support fr...</td>\n",
              "      <td>8.706396</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>Receptor Recognition by the Novel Coronavirus ...</td>\n",
              "      <td>Recently, a novel coronavirus (2019-nCoV) has ...</td>\n",
              "      <td>https://doi.org/10.1128/jvi.00127-20; https://...</td>\n",
              "      <td>Wan, Yushun; Shang, Jian; Graham, Rachel; Bari...</td>\n",
              "      <td>A novel coronavirus (2019-nCoV) from Wuhan, Ch...</td>\n",
              "      <td>8.692931</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>Identification of 2019-nCoV related coronaviru...</td>\n",
              "      <td>The ongoing outbreak of viral pneumonia in Chi...</td>\n",
              "      <td>https://doi.org/10.1101/2020.02.13.945485</td>\n",
              "      <td>Lam, Tommy Tsan-Yuk; Shum, Marcus Ho-Hin; Zhu,...</td>\n",
              "      <td>). We were also 72 able to successfully isolat...</td>\n",
              "      <td>8.533306</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>Susceptibility of ferrets, cats, dogs, and oth...</td>\n",
              "      <td>Severe acute respiratory syndrome coronavirus ...</td>\n",
              "      <td>https://doi.org/10.1126/science.abb7015; https...</td>\n",
              "      <td>Shi, Jianzhong; Wen, Zhiyuan; Zhong, Gongxun; ...</td>\n",
              "      <td>In late December 2019, an unusual pneumonia em...</td>\n",
              "      <td>8.464124</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>Susceptibility of ferrets, cats, dogs, and dif...</td>\n",
              "      <td>Severe acute respiratory syndrome coronavirus ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.30.015347</td>\n",
              "      <td>Shi, Jianzhong; Wen, Zhiyuan; Zhong, Gongxun; ...</td>\n",
              "      <td>In late December 2019, an unusual pneumonia em...</td>\n",
              "      <td>8.464124</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>Structural and biochemical characterization of...</td>\n",
              "      <td>Summary The ongoing global pandemic of coronav...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Peng, Qi; Peng, Ruchao; Yuan, Bin; Zhao, Jingr...</td>\n",
              "      <td>In the end of 2019, a novel coronavirus (2019-...</td>\n",
              "      <td>8.375391</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>Structural and biochemical characterization of...</td>\n",
              "      <td>The ongoing global pandemic of coronavirus dis...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.23.057265</td>\n",
              "      <td>Peng, Qi; Peng, Ruchao; Yuan, Bin; Zhao, Jingr...</td>\n",
              "      <td>In the end of 2019, a novel coronavirus (2019-...</td>\n",
              "      <td>8.324044</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>A Multibasic Cleavage Site in the Spike Protei...</td>\n",
              "      <td>Summary The pandemic coronavirus SARS-CoV-2 th...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Hoffmann, Markus; Kleine-Weber, Hannah; Pöhlma...</td>\n",
              "      <td>It is believed that the severe acute respirato...</td>\n",
              "      <td>8.240075</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>Comparative computational analysis of SARS-CoV...</td>\n",
              "      <td>Abstract Several research lines are currently ...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32302675/;...</td>\n",
              "      <td>Tilocca, Bruno; Soggiu, Alessio; Sanguinetti, ...</td>\n",
              "      <td>The virus species and accession number of the ...</td>\n",
              "      <td>8.151314</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>Synergistic China–US Ecological Research is Es...</td>\n",
              "      <td>The risk of a zoonotic pandemic disease threat...</td>\n",
              "      <td>https://doi.org/10.1007/s10393-020-01471-2; ht...</td>\n",
              "      <td>Smiley Evans, Tierra; Shi, Zhengli; Boots, Mic...</td>\n",
              "      <td>The catastrophic health and economic impacts o...</td>\n",
              "      <td>8.142541</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>42</th>\n",
              "      <td>A new coronavirus associated with human respir...</td>\n",
              "      <td>Emerging infectious diseases, such as severe a...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32015508/;...</td>\n",
              "      <td>Wu, Fan; Zhao, Su; Yu, Bin; Chen, Yan-Mei; Wan...</td>\n",
              "      <td>The patient studied was a 41-year-old man with...</td>\n",
              "      <td>8.099710</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>SARS‐CoV‐2 spike protein favors ACE2 from Bovi...</td>\n",
              "      <td>Severe acute respiratory syndrome coronavirus ...</td>\n",
              "      <td>https://doi.org/10.1002/jmv.25817; https://www...</td>\n",
              "      <td>Luan, Junwen; Jin, Xiaolu; Lu, Yue; Zhang, Lei...</td>\n",
              "      <td>Coronavirus disease 2019 (COVID‐19), which was...</td>\n",
              "      <td>8.028711</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>Novel Immunoglobulin Domain Proteins Provide I...</td>\n",
              "      <td>A novel coronavirus (SARS-CoV-2) is the causat...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.04.977736</td>\n",
              "      <td>Tan, Yongjun; Schneider, Theresa; Leong, Matth...</td>\n",
              "      <td>Nidoviruses are an ancient group of lipid-enve...</td>\n",
              "      <td>7.934869</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>Novel NGS Pipeline for Virus Discovery from a ...</td>\n",
              "      <td>The study of the microbiome data holds great p...</td>\n",
              "      <td>https://doi.org/10.1101/2020.05.07.082107</td>\n",
              "      <td>Plyusnin, Ilya; Kant, Ravi; Jääskeläinen, Anne...</td>\n",
              "      <td>Our ability to produce sequence data in the ra...</td>\n",
              "      <td>7.888766</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>The hard problem of AI rights</td>\n",
              "      <td>In the past few years, the subject of AI right...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>Andreotta, Adam J.</td>\n",
              "      <td>Artificially intelligent robots, machines, and...</td>\n",
              "      <td>7.776895</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>47</th>\n",
              "      <td>Broad and differential animal ACE2 receptor us...</td>\n",
              "      <td>The COVID-19 pandemic has caused an unpreceden...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.19.048710</td>\n",
              "      <td>Zhao, Xuesen; Chen, Danying; Szabla, Robert; Z...</td>\n",
              "      <td>human population remains mysterious, although ...</td>\n",
              "      <td>7.667037</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>48</th>\n",
              "      <td>Pangolins Lack IFIH1/MDA5, a Cytoplasmic RNA S...</td>\n",
              "      <td>Zoonotic infections are an imminent threat to ...</td>\n",
              "      <td>https://doi.org/10.3389/fimmu.2020.00939</td>\n",
              "      <td>Fischer, Heinz; Tschachler, Erwin; Eckhart, Le...</td>\n",
              "      <td>Emerging infectious diseases represent a major...</td>\n",
              "      <td>7.666060</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>49</th>\n",
              "      <td>New Virus, New Challenge</td>\n",
              "      <td>The ongoing COVID-19 pandemic has made us to c...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Shi, Yi</td>\n",
              "      <td>Yi Shi 1, * The ongoing COVID-19 pandemic has ...</td>\n",
              "      <td>7.638880</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-ab47fcb8-db55-44b4-96dc-1703e3795cf5')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-ab47fcb8-db55-44b4-96dc-1703e3795cf5 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-ab47fcb8-db55-44b4-96dc-1703e3795cf5');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-7f63a4cf-67f1-411b-b07c-9586e0fe82dd\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-7f63a4cf-67f1-411b-b07c-9586e0fe82dd')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-7f63a4cf-67f1-411b-b07c-9586e0fe82dd button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "res",
              "summary": "{\n  \"name\": \"res\",\n  \"rows\": 50,\n  \"fields\": [\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"COVID-19: the First Documented Coronavirus Pandemic in History\",\n          \"A Multibasic Cleavage Site in the Spike Protein of SARS-CoV-2 Is Essential for Infection of Human Lung Cells\",\n          \"Evolutionary origins of the SARS-CoV-2 sarbecovirus lineage responsible for the COVID-19 pandemic\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 49,\n        \"samples\": [\n          \"A novel type of coronavirus (2019-nCoV) infecting humans appeared in Wuhan, China, at the end of December 2019. Since the identification of the outbreak the infection quickly spread involving in one month more than 31,000 confirmed cases with 638 death. Molecular analysis suggest that 2019-nCoV could be originated from bats after passaging in intermediate hosts, highlighting the high zoonotic potential of coronaviruses.\",\n          \"In the past few years, the subject of AI rights\\u2014the thesis that AIs, robots, and other artefacts (hereafter, simply \\u2018AIs\\u2019) ought to be included in the sphere of moral concern\\u2014has started to receive serious attention from scholars. In this paper, I argue that the AI rights research program is beset by an epistemic problem that threatens to impede its progress\\u2014namely, a lack of a solution to the \\u2018Hard Problem\\u2019 of consciousness: the problem of explaining why certain brain states give rise to experience. To motivate this claim, I consider three ways in which to ground AI rights\\u2014namely: superintelligence, empathy, and a capacity for consciousness. I argue that appeals to superintelligence and empathy are problematic, and that consciousness should be our central focus, as in the case of animal rights. However, I also argue that AI rights is disanalogous from animal rights in an important respect: animal rights can proceed without a solution to the \\u2018Hard Problem\\u2019 of consciousness. Not so with AI rights, I argue. There we cannot make the same kinds of assumptions that we do about animal consciousness, since we still do not understand why brain states give rise to conscious mental states in humans.\",\n          \"Zoonotic infections are an imminent threat to human health. Pangolins were recently identified as carriers and intermediate hosts of coronaviruses. Previous research has shown that infection with coronaviruses activates an innate immune response upon sensing of viral RNA by interferon-induced with helicase C domain 1 (IFIH1), also known as MDA5. Here, we performed a comparative genomics study of RNA sensor genes in three species of pangolins. DDX58/RIG-I, a sensor of cytoplasmic viral RNA and toll-like receptors (TLR) 3, 7, and 8, which bind RNA in endosomes, are conserved in pangolins. By contrast, IFIH1 a sensor of intracellular double-stranded RNA, has been inactivated by mutations in pangolins. Likewise, Z-DNA-binding protein (ZBP1), which senses both Z-DNA and Z-RNA, has been lost during the evolution of pangolins. These results suggest that the innate immune response to viruses differs significantly between pangolins and other mammals, including humans. We put forward the hypothesis that loss of IFIH1 and ZBP1 provided an evolutionary advantage by reducing inflammation-induced damage to host tissues and thereby contributed to a switch from resistance to tolerance of viral infections in pangolins.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"https://www.ncbi.nlm.nih.gov/pubmed/32387617/; https://www.sciencedirect.com/science/article/pii/S2319417020300445?v=s5; https://doi.org/10.1016/j.bj.2020.04.007; https://api.elsevier.com/content/article/pii/S2319417020300445\",\n          \"https://www.sciencedirect.com/science/article/pii/S1097276520302641; https://api.elsevier.com/content/article/pii/S1097276520302641; https://www.ncbi.nlm.nih.gov/pubmed/32362314/; https://doi.org/10.1016/j.molcel.2020.04.022\",\n          \"https://doi.org/10.1101/2020.03.30.015008\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 48,\n        \"samples\": [\n          \"Yan, Zhen; Spaulding, Hannah R.\",\n          \"Wu, Fan; Zhao, Su; Yu, Bin; Chen, Yan-Mei; Wang, Wen; Song, Zhi-Gang; Hu, Yi; Tao, Zhao-Wu; Tian, Jun-Hua; Pei, Yuan-Yuan; Yuan, Ming-Li; Zhang, Yu-Ling; Dai, Fa-Hui; Liu, Yi; Wang, Qi-Min; Zheng, Jiao-Jiao; Xu, Lin; Holmes, Edward C.; Zhang, Yong-Zhen\",\n          \"Wang, Xinye; Bailey, Emily S; Qi, Xian; Yu, Huiyan; Bao, Changjun; Gray, Gregory C\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"Currently, people all over the world have been affected by coronavirus disease 2019 (COVID-19), which is the fifth pandemic after the 1918 flu pandemic. As of now, we can trace the first report and subsequent outbreak from a cluster of novel human pneumonia cases in Wuhan City, China, since late December 2019. The earliest date of symptom onset was 1 December 2019. The symptomatology of these patients, including fever, malaise, dry cough, and dyspnea, was diagnosed as viral pneumonia [1, 2]. Initially, the disease was called Wuhan pneumonia by the press because of the area and pneumonia symptoms. Whole-genome sequencing results showed that the causative agent is a novel coronavirus. Therefore, this virus is the seventh member of the coronavirus family to infect humans [3]. The World Health Organization (WHO) temporarily termed the new virus 2019 novel coronavirus (2019-nCoV) on 12 January 2020 and then officially named this infectious disease coronavirus disease 2019 (COVID-19) on 12 February 2020. Later, the International Committee on Taxonomy of Viruses (ICTV) officially designated the virus as SARS-CoV-2 based on phylogeny, taxonomy and established practice [4]. Subsequently, human-to-human transmission of COVID-19 occurring within Hong Kong has been shown in clinical data [5]. Since COVID-19 initially emerged in China, the virus has evolved for four months and rapidly spread to other countries worldwide as a global threat. On 11 March 2020, the WHO finally made the assessment that COVID-19 can be characterized as a pandemic, following 1918 Spanish flu (H1N1), 1957 Asian flu (H2N2), 1968 Hong Kong flu (H3N2), and 2009 Pandemic flu (H1N1), which caused an estimated 50 million, 1.5 million, 1 million, and 300,000 human deaths, respectively [[6], [7], [8], [9]] [Fig. 1\\n].\\nSARS-CoV-2 is an enveloped and spherical particle approximately 120 nm in diameter containing a positive-sense single-stranded RNA genome. It belongs to the subfamily Coronavirinae, family Coronavirdiae, and order Nidovirales. The RNA genome of SARS-CoV-2 contains a 5' methyl-guanosine cap, poly (A)-tail, and 29,903 nucleotides according to WH-Human 1 coronavirus (WHCV) [3, 10]. It is classified as a beta-coronavirus (\\u03b2CoV) [lineage B] and is the seventh coronavirus to infect humans, following 2 \\u03b1CoV (HCoV-229E and HKU-NL63) and 4 \\u03b2CoV (HCoV-OC43 [lineage A], HCoV-HKU1 [lineage A], severe acute respiratory syndrome SARS-CoV [lineage B] and Middle East respiratory syndrome MERS-CoV [lineage C]) [[11], [12], [13], [14]]. It has been shown that bats and rodents are the gene sources of most \\u03b1CoVs and \\u03b2CoVs, however, avian species are the gene sources of most \\u03b4CoVs and \\u03b3CoVs by evolutionary analyses. The human coronavirus (HCoV) strains HCoV-NL63, HCoV-229E, HCoV-HKU1, and HCoV-OC43 usually cause mild, self-limiting upper respiratory tract infections, such as the common cold [15, 16]. However, SARS-CoV, MERS-CoV, and SARS-CoV-2 can cause severe acute respiratory syndrome and result in life-threatening disease [[17], [18], [19]] [Table 1\\n].\\nSARS-CoV-2 transcribes nine subgenomic RNAs, and its genome comprises a 5' untranslated region including a 5' leader sequence; an open reading frame (ORF) 1a/ab encoding nonstructural proteins (nsp) for replication; four structural proteins including spike (S), envelope (E), membrane (M) and nucleocapsid (N); several accessory proteins such as ORF 3a, 6, 7a/b, and 8; and a 3' untranslated region. The replicase polyprotein pp1a/ab encodes and is proteolytically cleaved into 16 putative nsps, including nsp3 (papain-like protease), nsp5 (3C-like protease), nsp12 (RNA-dependent RNA polymerase [RdRp]), nsp13 (helicase), and other nsps [10, 13, 20]. The spike glycoprotein of SARS-CoV-2 binds to angiotensin-converting enzyme 2 (ACE2) in human and Chinese horseshoe bats, civet for cell entry, that is also dependent on S protein priming by the serine protease TMPRSS2. A similar panel of mammalian cell lines can be infected with SARS-CoV-2-S and SARS-CoV-S [[21], [22], [23], [24]]. The spike protein could be cleaved by host proteases into the S1 and S2 subunits, which are responsible for receptor recognition and membrane fusion, respectively. S1 also can be divided into an N-terminal domain (NTD) and a C-terminal domain (CTD). The S1 CTD of SARS-CoV-2, but not the NTD, showed strong affinity for human ACE2 (hACE2). The receptor-binding domain (RBD) within SARS-CoV-2 CTD is the key region that interacts with the hACE2 receptor with higher affinity than the RBD of SARS-CoV by 10- to 20-fold with kinetic quantification [23, 25]. The putative life cycle of SARS-CoV-2 in host cells begins from spike protein and hACE2 receptor binding. The conformational change in the S protein after receptor binding facilitates viral envelope fusion with the cell membrane through the endosomal pathway. The viral RNA genome is then released into the cytoplasm and translated into viral replicase polyproteins pp1a and 1ab, which can be cleaved into small products by virus-encoded proteinases. The polymerase transcribes a series of subgenomic mRNAs by discontinuous transcription. The subgenomic mRNAs are finally translated into viral structural proteins. The S, E and M proteins enter the endoplasmic reticulum (ER) and Golgi apparatus, and the N protein is combined with the positive-stranded genomic RNA to form a nucleoprotein complex. The structural proteins and nucleoprotein complex are assembled with the viral envelope at the ER\\u2013Golgi intermediate compartment. The newly assembled viral particles are then released from the infected cell [Fig. 2\\n].\\nAll human coronaviruses have animal origins, namely, natural hosts. Bats may be the natural hosts of HCoV-229E, SARS-CoV, HCoV-NL63, and MERS-CoV. Furthermore, HCoV-OC43 and HKU1 probably originated from rodents [[26], [27], [28]]. Bats are undoubtedly important and the major natural reservoirs of alpha-coronaviruses and beta-coronaviruses [29]. Domestic animals can suffer from disease as intermediate hosts that cause virus transmission from natural hosts to humans; for example, SARS-CoV and MERS-CoV crossed the species barriers into masked palm civets and camels, respectively [30, 31] [Table 1]. SARS-CoV-2 sequenced at the early stage of the COVID-19 outbreak only shares 79.6% sequence identity with SARS-CoV through early full-length genomic comparisons. However, it is highly identical (96.2%) at the whole-genome level to Bat-CoV RaTG13, which was previously detected in Rhinolophus affinis from Yunnan Province, over 1,500 km from Wuhan [21]. Bats are likely reservoir hosts for SARS-CoV-2; however, whether Bat-CoV RaTG13 directly jumped to humans or transmits to intermediate hosts to facilitate animal-to-human transmission remains inconclusive. No intermediate host sample was obtained by scientists in an initial cluster of infections of the Huanan Seafood and Wildlife Market in Wuhan, where the sale of wild animals may be the source of zoonotic infection. Furthermore, the earliest three patients with symptom onset had no known history of exposure to the Huanan market [1]. Therefore, there may be multiple sources of COVID-19 in the beginning. According to previous studies by metagenomic sequencing for the samples from Malayan pangolins (Manis javanica) in Guangxi and Guangdong, China, it has been suggested that pangolins might be the intermediate hosts between bats and humans because of the similarity of the pangolin coronavirus to SARS-CoV-2 [32, 33]. The additional analysis of phylogenetic networks can be effective and useful to trace undocumented COVID-19 infection sources [34]. In addition to the zoonotic origins of SARS-CoV-2 by natural evolution, there are still some disputes about the origin of the virus because its spike protein seems to perfectly interact with the human receptor in contributing to human-to-human transmission after evolution in a short period. Nevertheless, more direct evidence is required to clarify the arguments.\\nReplication of RNA viruses could generate mutations due to the low proofreading ability of their RdRP. The genome variations generated by viral RdRP could be beneficial for an emerging virus to adapt to new hosts. However, previous studies have shown that the mutation rates could vary in RNA viruses [35]. The synonymous substitution rate for coronaviruses might be approximately 1 \\u00d7 10-3/synonymous site/year, which is lower than some other RNA viruses. The mutation rate during coronavirus replication could be partially controlled by the viral exoribonuclease nsp14 [36, 37]. Nevertheless, SARS-CoV-2 has been continuously evolving to different groups worldwide during the pandemic.\\nAccording to the information of nCoV-19 (SARS-CoV-2) sequences submitted to the GISAID database in January 2020, the virus was first collected in late December 2019 from Wuhan, China. However, those viral sequences varied from the latest submitted sequence collected in early April 2020 from North America. Since the viral sequences continuously change, the construction of a phylogenetic network is crucial to investigate the adaption of the virus in different human populations and environments. Although the virus keeps evolving within humans who could also be susceptible to other human coronaviruses, recombination between SARS-CoV-2 and old human coronaviruses, such as HCoV-229E, OC43, NL63, and HKU1, has not been found. Nevertheless, a recent study revealed that three genetic types, A, B, and C, of the virus have been circulating globally [34]. The study demonstrated that the genotypes could also correlate to the geographic locations. This indicates that the evolution of SARS-CoV-2 may be affected by replication environments, such as genetic and immunological restrictions in different human populations. With evolutionary pressure, the selection of SARS-CoV-2 mutations will be ongoing. The investigation of the geographic patterns of SARS-CoV-2 variations will provide information on vaccine development for different populations.\\nHuman coronaviruses usually cause mild upper respiratory diseases. However, in the past two decades, two coronaviruses transmitted from animals, SARS-CoV and MERS-CoV, have caused severe pneumonia and death in humans. In addition, since late December 2019, the COVID-19 pandemic has spread globally and consequently resulted in at least 190,549 deaths worldwide as of April 24, 2020 [38]. Due to the high sequence homology with a coronavirus isolated from bats, SARS-CoV-2 is considered a zoonotic origin coronavirus. Undoubtedly, SARS-CoV-2 has become the fifth human coronavirus, and it is possible that this virus will continuously circulate in the human population in the future. Because specific antiviral treatments and vaccines are still under development, testing, quarantine, and social distancing are encouraged to prevent virus spread. Nonetheless, since the virus keeps mutating and evolving during the pandemic, studies on viral pathogenicity, treatments and prophylactic vaccines should closely consider the genetic characteristics of the virus.\\n\",\n          \"It is believed that the severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2, previously termed nCoV-2019) was introduced into the human population from a poorly characterized animal reservoir in late 2019 (Ge et al., 2013, Wang et al., 2020, Zhou et al., 2020b, Zhu et al., 2020). The epicenter of the subsequent SARS-CoV-2 spread was Wuhan, Hubei province, China, with more than 65,000 cases occurring in this area (WHO, 2020a). However, infections have now been detected in more than 110 countries and massive outbreaks are currently ongoing in the United States, Italy, and Spain (WHO, 2020a, WHO, 2020b). Understanding which features of SARS-CoV-2 are essential for infection of human cells should provide insights into viral transmissibility and pathogenesis and might reveal targets for intervention.\\nThe spike protein of coronaviruses is incorporated into the viral envelope and facilitates viral entry into target cells. For this, the surface unit S1 binds to a cellular receptor while the transmembrane unit S2 facilitates fusion of the viral membrane with a cellular membrane (Hoffmann et al., 2018, Hulswit et al., 2016, Millet and Whittaker, 2018). Membrane fusion depends on S protein cleavage by host cell proteases at the S1/S2 and the S2\\u2032 site (Figure 1\\nA), which results in S protein activation (Hoffmann et al., 2018, Hulswit et al., 2016, Millet and Whittaker, 2018). Cleavage of the S protein can occur in the constitutive secretory pathway of infected cells or during viral entry into target cells and is essential for viral infectivity. Therefore, the responsible enzymes constitute potential targets for antiviral intervention.\\nOur previous work revealed that the activity of the cellular serine protease TMPRSS2, which activates several coronaviruses (Bertram et al., 2013, Gierer et al., 2013, Glowacka et al., 2011, Matsuyama et al., 2010, Shirato et al., 2013, Shirato et al., 2016, Shulla et al., 2011), is also required for robust SARS-CoV-2 infection of human lung cells (Hoffmann et al., 2020). However, it is conceivable that the activity of other cellular proteases is also necessary. Thus, the Middle East respiratory syndrome coronavirus spike protein (MERS-S) is activated by a two-step process: MERS-S is first cleaved by furin at the S1/S2 site in infected cells, which is required for subsequent TMPRSS2-mediated cleavage at the S2\\u2032 site (Figure 1A) during viral entry into lung cells (Kleine-Weber et al., 2018, Park et al., 2016, Millet and Whittaker, 2014). A cathepsin B/L-dependent auxiliary activation pathway is operative in many TMPRSS2\\u2212 cell lines but seems not to be available in viral target cells in the lung because TMPRSS2-dependent activation of the S protein is essential for robust MERS-CoV and SARS-CoV spread and pathogenesis in the infected host (Iwata-Yoshikawa et al., 2019, Simmons et al., 2005, Zhou et al., 2015).\\nThe S1/S2 site in SARS-CoV-2 forms an exposed loop (Figure 1B) that harbors multiple arginine residues (multibasic) (Walls et al., 2020, Wrapp et al., 2020) that are not found in SARS-CoV-related coronaviruses (SARSr-CoV) but are present in the human coronaviruses OC43, HKU1, and MERS-CoV (Figure 1C). However, the contribution of this multibasic cleavage site to SARS-CoV-2 infection of human cells is unknown and was in the focus of the present study.\\nIn order to address the role of the multibasic S1/S2 cleavage site in SARS-CoV-2 infection, we generated S protein mutants with altered S1/S2 cleavage sites (Figure 2\\nA). In particular, we exchanged the multibasic cleavage site against its monobasic counterparts present in SARS-S or RaTG13-S (Figure 2A, RaTG13 is a bat coronavirus closely related to SARS-CoV-2 [Zhou et al., 2020b]). This resulted in mutants SARS-2-S (SARS) and SARS-2-S (RaTG). Moreover, we either deleted all arginines in the S1/S2 site of SARS-2-S or inserted an additional arginine residue (jointly with an alanine to lysine exchange), giving rise to mutants SARS-2-S (delta) and SARS-2-S (opt), respectively. Finally, we introduced the S1/S2 sites of SARS-2-S and RaTG13-S into the background of SARS-S (Figure 2A), which yielded the mutants SARS-S (SARS-2) and SARS-S (RaTG).\\nThe effects of the above described S1/S2 mutations on viral entry were examined using vesicular stomatitis virus (VSV) particles bearing S proteins because these particles are safe and adequately reflect coronavirus entry into target cells. Immunoblot of VSV particles bearing S proteins with a C-terminal antigenic tag revealed that all S proteins were readily incorporated into VSV particles. SARS-2-S WT (wild type) was efficiently cleaved at the S1/S2 site (Figure 2B), in keeping with published data (Hoffmann et al., 2020, Walls et al., 2020). Exchange of the S1/S2 site of SARS-2-S against those of SARS-S and RaTG13-S abrogated cleavage, and this effect was also seen when the multibasic motif was deleted (Figure 2B). Moreover, insertion of an additional arginine residue jointly with an alanine-to-lysine exchange at the S1/S2 site did not appreciably increase cleavability. Finally, insertion of the S1/S2 site of SARS-2-S into SARS-S increased S protein cleavability while insertion of the RaTG13 S1/S2 site did not (Figure 2B). These results indicate that the presence of several arginine residues at the S1/S2 site is required for efficient SARS-2-S proteolytic processing in human cells and also confers high cleavability to SARS-S.\\nWe next investigated which protease is required for S protein processing at the S1/S2 site. The S1/S2 motif matches the minimal furin sequence RXXR and is closely related to the furin consensus sequence RX[K/R]R. Therefore, we analyzed whether decanoyl-RVKR-CMK, a furin inhibitor, blocks SARS-2-S processing at the S1/S2 site. Decanoyl-RVKR-CMK inhibited processing of MERS-S, which is known to depend on furin (Gierer et al., 2015, Millet and Whittaker, 2014), in a concentration-dependent manner and had no effect on SARS-S expression (Figure 2C), as expected. Processing of SARS-2-S was also inhibited, indicating that furin cleaves SARS-2-S at the S1/S2 site. In order to determine whether cleavage at the S1/S2 site is required for SARS-2-S-driven cell-cell fusion, we studied S-protein-dependent formation of multinucleated giant cells (syncytia). No syncytia were observed in the absence of S protein expression while MERS-S WT expression resulted in syncytium formation, which was increased upon addition of trypsin or expression of TMPRSS2 (Figure 2D). Expression of SARS-S WT or SARS-S harboring the S1/S2 site of RaTG13-S did not induce syncytium formation in the absence of protease, but modest multikaryon formation was detected in the presence of trypsin or TMPRSS2. In contrast, SARS-S harboring the SARS-2-S S1/S2 site induced syncytia in the absence of protease, and syncytium formation was markedly increased by trypsin and, particularly, TMPRSS2. SARS-2-S expression triggered syncytium formation that was strongly increased by trypsin and TMPRSS2. Syncytium formation was clearly less prominent and required the presence of trypsin or TMPRSS2 when the SARS-2-S S1/S2 site was replaced by that of SARS-S or RaTG13-S. Moreover, deletion of the multibasic motif resulted in a spike protein that was no longer able to induce syncytium formation even in the presence of trypsin or TMPRSS2. Finally, the addition of an arginine residue to the S1/S2 site of SARS-2-S jointly with alanine-to-lysine exchange strongly increased syncytium formation, indicating that viral variants with optimized S1/S2 sites might show augmented cell-cell spread and potentially altered pathogenicity. Thus, the S1/S2 site of SARS-2-S is required for cell-cell fusion, and this process can be augmented by adding basic residues to the S1/S2 site.\\nWe finally examined the importance of the S1/S2 site for S-protein-mediated virus-cell fusion. Blockade of SARS-2-S cleavage at the S1/S2 site (mutants SARS-2-S (SARS), SARS-2-S (RaTG), and SARS-2-S (delta)) abrogated entry into the TMPRSS2+ human lung cell line Calu-3 (Figure 2E), in which the cathepsin B/L-dependent S protein activation pathway is not sufficiently available (Park et al., 2016). In contrast, entry into TMPRSS2\\u2212 Vero cells, which is known to be cathepsin B/L dependent, was not affected by these mutations (Figure 1E), in keeping with results reported by Walls and colleagues (Walls et al., 2020). Optimization of the S1/S2 site did not increase entry into the cell lines tested; it slightly decreased entry into both Vero and Calu-3 cells, for, at present, unclear reasons. Finally, alterations of the S1/S2 site of SARS-S did not augment entry efficiency. Collectively, these results demonstrate that a multibasic S1/S2 site is essential for SARS-2-S-driven entry into human lung cells while a monobasic site is sufficient for SARS-S.\\nOur results demonstrate that the multibasic S1/S2 cleavage site is essential for SARS-2-S-driven entry into TMPRSS2+ lung cells. It will be interesting to extend these studies to primary human respiratory epithelial cells and to authentic SARS-CoV-2, which requires a reverse genetics system not available to the present study.\\nFurther information and requests for resources and reagents should be directed to and will be fulfilled by the Lead Contact, Stefan P\\u00f6hlmann (spoehlmann@dpz.eu).\\nAll unique/stable reagents generated in this study are available from the Lead Contact with a completed Materials Transfer Agreement.\\nThe study did not generate unique datasets or code.\\n293T (human, kidney) and Vero (African green monkey, kidney) cells were cultivated in Dulbecco\\u2019s Modified Eagle Medium (PAN-Biotech) supplemented with 10% fetal bovine serum (Biochrom), 100 U/mL of penicillin and 0.1 mg/mL of streptomycin (PAN-Biotech). Vero cells that stably express human TMPRSS2 have been described previously (Hoffmann et al., 2020) and were cultivated in the presence of 10 \\u03bcg/mL blasticidin (Invivogen). Calu-3 (human, lung; kindly provided by Stephan Ludwig, Westf\\u00e4lische Wilhelms-Universit\\u00e4t, Muenster/Germany) cells were cultivated in Minimum Essential Medium (Thermo Fisher Scientific) supplemented with 10% fetal bovine serum (Biochrom), 100 U/mL of penicillin and 0.1 mg/mL of streptomycin (PAN-Biotech), 1x non-essential amino acid solution (from 100x stock, PAA) and 10 mM sodium pyruvate (Thermo Fisher Scientific). All cell lines were incubated at 37\\u00b0C and 5% CO2 in a humidified atmosphere.\\nExpression plasmids for full-length vesicular stomatitis virus (VSV) glycoprotein (VSV-G), Middle-East respiratory syndrome coronavirus spike glycoprotein (MERS-S) containing a C-terminal V5 epitope tag, severe acute respiratory syndrome coronavirus spike glycoprotein (SARS-S) and severe acute respiratory syndrome coronavirus 2 spike glycoprotein (SARS-2-S) both equipped with a C-terminal hemagglutinin (HA) epitope tag have been described previously (Brinkmann et al., 2017, Hoffmann et al., 2020). Empty pCG1 expression vector was kindly provided by Roberto Cattaneo, Mayo Clinic, Rochester, MN/USA). Based on the SARS-S and SARS-2-S expression plasmids we cloned mutated versions with alterations at the S1/S2 cleavage site: We generated SARS-S containing the cleavage site of SARS-2-S, SARS-S (SARS-2), or BetaCoV/bat/Yunnan/RaTG13/2013 (RaTG; GISAID: EPI_ISL_402131), SARS-S (RaTG). Further, we generated SARS-2-S harboring the S1/S2 cleavage site of SARS-S, SARS-2-S (SARS) or RaTG-S, SARS-2-S (RaTG). Finally, we constructed SARS-2-S variants in which either the multibasic motif was deleted, SARS-2-S (delta), or in which the proline residue preceding the multibasic motif was mutated to arginine and the alanine residue within the minimal furin motif was changed to lysine in order to increase the basic environment at the S1/S2 site, SARS-2-S (opt). All newly cloned spike protein constructs further contained a deletion of 18 amino acids at their respective C terminus as this has been shown to improve coronavirus spike protein incorporation into VSV particles and thus transduction (Schwegmann-We\\u00dfels et al., 2009). Further, for each construct an untagged variant as well as a version containing a C-terminal V5 epitope tag was constructed.\\nA previously published protocol was employed to produce VSV pseudotype particles (VSVpp) carrying foreign viral glycoproteins in their envelope (Berger Rentsch and Zimmer, 2011, Kleine-Weber et al., 2019). First, 293T cells were transfected with expression plasmid for the respective spike glycoprotein or VSV-G or empty expression vector by calcium-phosphate precipitation. At 16 h posttransfection, the cells were inoculated with VSV\\u2217\\u0394G-fLuc (kindly provided by Gert Zimmer, Institute of Virology and Immunology, Mittelh\\u00e4usern/Switzerland), a replication-deficient VSV vector that lacks the genetic information for VSV-G and encodes for eGFP and firefly luciferase (fLuc), at a multiplicity of infection of 3. After 1 h of incubation, the inoculum was removed and cells were washed with phosphate-buffered saline (PBS) before medium containing anti-VSV-G antibody (I1, mouse hybridoma supernatant from CRL-2700; ATCC) was added to all cells except for those expressing VSV-G (here, medium without antibody was added). Cells were further incubated for 16 h, before the VSVpp containing supernatants were harvested, freed from cellular debris by centrifugation and used for experiments.\\nFor transduction, target cells were grown in 96-well plates until they reached 50%\\u201380% confluency. The culture supernatant was removed by aspiration and 100 \\u03bcl/well of the respective pseudotype were added (quadruplicate samples). At 16 h posttransduction, culture supernatants were aspirated and cells lysed in 1x cell culture lysis reagent (prepared from 5x stock, Promega) for 20 min at room temperature. The lysates were then transferred to white, opaque-walled 96-well plates and luciferase activity was quantified by measuring luminescence upon addition of a substrate (PJK) using a Hidex Sense plate luminometer (Hidex).\\nFor the analysis of S protein processing, we subjected VSVpp harboring V5- or HA-tagged S proteins to SDS-PAGE and western blot analysis. For this, we loaded 1 mL VSVpp onto 50 \\u03bcl of a 20% (w/v) sucrose cushion and performed high-speed centrifugation (25.000 g for 120 min at 4\\u00b0C). Next, we removed 1 mL of supernatant, added 50 \\u03bcl of 2x SDS-sample buffer and incubated the samples for 15 min at 96\\u00b0C. Thereafter, the samples were subjected to SDS-PAGE and protein transfer to nitrocellulose membranes by western blot. The membranes were subsequently blocked in 5% skim milk solution (PBS containing 0.05% Tween-20 [PBS-T] and 5% skim milk powder) for 1 h at room temperature. The blots were then incubated over night at 4\\u00b0C with primary antibody solution (all antibodies were diluted in PBS-T containing 5% skim milk; mouse anti-HA tag [Sigma-Aldrich, H3663, 1:2,500], mouse anti-V5 tag [Thermo Fisher Scientific, R960-25, 1:2,500] or VSV matrix protein [Kerafast, EB0011, 1:2,500]). Following this incubation, the blots were washed 3x with PBS-T before they were incubated for 1 h at room temperature with peroxidase-coupled goat anti-mouse antibody (Dianova, 115-035-003, 1:10,000). Finally, the blots were again washed and imaged. For this, an in house-prepared enhanced chemiluminescent solution (0.1 M Tris-HCl [pH 8.6], 250 \\u03bcg/mL luminol, 1 mg/mL para-hydroxycoumaric acid, 0.3% H2O2) and the ChemoCam imaging system along with the ChemoStar Professional software (Intas Science Imaging Instruments GmbH) were used.\\nVero or Vero-TMPRSS2 cells were grown on coverslips seeded in 24-well plates and transfected with S protein expression plasmids (1 \\u03bcg/well) using Lipofectamine 2000 LTX with Plus reagent (Thermo Fisher Scientific) and OptiMEM medium (GIBCO). After 6 h the transfection solutions were aspirated and the cells further incubated for 24 h in standard culture medium. Next, the medium was changed to serum free medium \\u00b1 1 \\u03bcg/mL bovine trypsin (Sigma-Aldrich) and the cells were incubated for additional 24 h. Then, the cells were washed with PBS, fixed with 4% paraformaldehyde solution for 20 min at room temperature, washed again, air-dried and incubated for 30 min with May-Gruenwald solution (Sigma-Aldrich). Thereafter, the cells were washed three times with deionized water, air-dried and incubated for 30 min with 1:10 diluted Giemsa solution (Sigma-Aldrich). After an additional washing interval with deionized water, the samples were air-dried and analyzed by bright-field microscopy using a Zeiss LSM800 confocal laser scanning microscope and the ZEN imaging software (both from Zeiss).\\nSequence alignments were performed using the Clustal Omega online tool (https://www.ebi.ac.uk/Tools/msa/clustalo/). Protein models were designed using the YASARA software (http://www.yasara.org/index.html). For the generation of the SARS-2-S protein model the protein sequence was first modeled on a SARS-S template (PDB: 5X5B, (Yuan et al., 2017)) using the SWISS-MODEL online tool (https://swissmodel.expasy.org/). The following sequences information were obtained from National Center for Biotechnology Information (NCBI) database: SARS-CoV BJ01 (GenBank: AY278488.2), SARS-CoV CUHK-W1 (GenBank: AY278554.2), SARS-CoV Frankfurt-1 (GenBank: AY291315.1), SARS-CoV Tor2 (GenBank: CS050815.1), SARS-CoV Urbani (GenBank: AY278741.1), civet SARS-CoV SZ16 (GenBank: AY304488.1), civet SARS-CoV civet020 (GenBank: AY572038.1), raccoon dog SARS-CoV A030 (GenBank: AY687357.1), bat SARSr-CoV BtKY72/KEN (GenBank: KY352407.1), bat SARSr-CoV BM48-31/BGR/2008 (GenBank: GU190215.1), bat SARSr-CoV Rs4231 (GenBank: KY417146.1), bat SARSr-CoV WIV16 (GenBank: KT444582.1), bat SARSr-CoV Rs4874 (GenBank: KY417150.1), bat SARSr-CoV SL-CoVZC45 (GenBank: MG772933.1), bat SARSr-CoV SL-CoVZXC21 (GenBank: MG772934.1), bat SARSr-CoV LYRa11 (GenBank: KF569996.1), bat SARSr-CoV LYRa3 (GenBank: KF569997.1), bat SARSr-CoV WIV1 (GenBank: KF367457.1), bat SARSr-CoV RsSHC014 (GenBank: KC881005.1), bat SARSr-CoV Rs3367 (GenBank: KC881006.1), bat SARSr-CoV Cp/Yunnan2011 (GenBank: JX993988.1), bat SARSr-CoV Rp/Shaanxi2011 (GenBank: JX993987.1), bat SARSr-CoV HKU3-1 (GenBank: DQ022305.2), bat SARSr-CoV Rm1 (GenBank: DQ412043.1), bat SARSr-CoV Rp3 (GenBank: DQ071615.1), bat SARSr-CoV Rf1 (GenBank: DQ412042.1), bat SARSr-CoV 279 (GenBank: DQ648857.1), bat SARSr-CoV 273 (GenBank: DQ648856.1), bat SARSr-CoV YN2013 (GenBank: KJ473816.1), bat SARSr-CoV Rs/HuB2013 (GenBank: KJ473814.1), bat SARSr-CoV Rs/GX2013 (GenBank: KJ473815.1), bat SARSr-CoV Rf/SX2013 (GenBank: KJ473813.1), bat SARSr-CoV Rf/JL2012 (GenBank: KJ473811.1), bat SARSr-CoV Rf/HeB2013 (GenBank: KJ473812.1), bat SARSr-CoV YNLF/34C (GenBank: KP886809.1), bat SARSr-CoV YNLF/31C (GenBank: KP886808.1), bat SARSr-CoV Rs672 (GenBank: FJ588686.1), bat SARSr-CoV Rs7327 (GenBank: KY417151.1), bat SARSr-CoV Rs4084 (GenBank: KY417144.1), bat SARSr-CoV Rs9401 (GenBank: KY417152.1), bat SARSr-CoV Rs4247 (GenBank: KY417148.1), bat SARSr-CoV Rs4255 (GenBank: KY417149.1), bat SARSr-CoV Rs4081 (GenBank: KY417143.1), bat SARSr-CoV Rs4237 (GenBank: KY417147.1), bat SARSr-CoV As6526 (GenBank: KY417142.1), bat SARSr-CoV Rf4092 (GenBank: KY417145.1), bat SARSr-CoV Longquan-140 (GenBank: KF294457.1), bat SARSr-CoV Rs806 (GenBank: FJ588692.1), bat SARSr-CoV Anlong-103 (GenBank: KY770858.1), bat SARSr-CoV JTMC15 (GenBank: KU182964.1), bat SARSr-CoV 16BO133 (GenBank: KY938558.1), bat SARSr-CoV B15-21 (GenBank: KU528591.1), pangolin coronavirus MP789 (GenBank: MT084071.1). In addition the following sequences information were obtained from the Global Initiative on Sharing All Influenza Data (GISAID) database: SARS-CoV-2 (GISAID: EPI_ISL_404895), bat SARSr-CoV RaTG13 (GISAID: EPI_ISL_402131).\\nIf not stated otherwise, statistical significance was tested by one-way analysis of variance with Dunnet\\u2019s posttest (GraphPad Prism 7.03). Only p values of 0.05 or lower were considered statistically significant (p > 0.05 [ns, not significant], p \\u2264 0.05 [\\u2217], p \\u2264 0.01 [\\u2217\\u2217], p \\u2264 0.001 [\\u2217\\u2217\\u2217]). For all statistical analyses, the GraphPad Prism 7 software package was used (GraphPad Software).\\n\",\n          \"In December 2019, a cluster of pneumonia cases epidemiologically linked to an open-air wet market in city of Wuhan (Hubei Province), China Zhou et al., 2020b) led local health officials to issue an epidemiological alert to the Chinese Center for Disease Control and Prevention (China CDC) and the World Health Organization's (WHO) China Country Office. In early January, the etiological agent of the pneumonia cases was found to be a coronavirus (World Health Organization, 2020a) , subsequently named SARS-CoV-2 by an International Committee on Taxonomy of Viruses (ICTV) Study Group (Gorbalenya et al., 2020) , (also named hCoV-19 by (Wu et al., 2020b) ). The first available sequence data placed this novel human pathogen in the Sarbecovirus subgenus of Coronaviridae (Lu et al., 2020) , the same sub-genus as the SARS virus that caused a global outbreak of nearly 8000 cases in [2002] [2003] . By mid-January, the virus was spreading widely inside Hubei province and by early March SARS-CoV-2 reached pandemic status (World Health Organization, 2020b) .\\nIn outbreaks of zoonotic pathogens, identification of the infection source is crucial, as this may allow health authorities to separate human populations from the wild-life or domestic-animal reservoirs posing the zoonotic risk (Stegeman et al., 2004; Yu et al., 2013) . If controlling an outbreak in its early stages is not possible -as was the case for the COVID-19 epidemic in Hubei province -identification of origins and point sources is nevertheless important for containment purposes in other provinces and prevention of future outbreaks. When the first genome sequence of SARS-CoV-2, Wuhan-Hu-1, was released on January 10 2020 on Virological.org by a consortium led by Yong-Zhen Zhang it enabled the immediate analyses of its ancestry. Across a large region of its genome, corresponding approximately to ORF1b , it did not cluster with any of the known bat coronaviruses indicating that recombination likely played a role in the evolutionary history of these viruses (Lu et al., 2020; Wu et al. 2020b ). Subsequently, a bat sarbecovirus -RaTG13 sampled from a Rhinolophus affinis horseshoe bat in 2013 in Yunnan province -was reported that clusters with SARS-CoV-2 in almost all genomic regions with approximately 96% genome sequence identity (Zhou et al., 2020b) . Zhou et al. (2020b) concluded from the genetic proximity of SARS-CoV-2 to a bat virus that a bat origin for the current COVID-19 outbreak is probable. Recent evidence has also identified pangolins as potential intermediate species for SARS-CoV-2 emergence or as a potential reservoir species themselves (Lam et al., 2020; Xiao et al., 2020) .\\nUnlike other viruses that have emerged in the past two decades, coronaviruses are highly recombinogenic (Forni et al., 2017; Hon et al., 2008; Lam et al., 2018) . Influenza viruses reassort (Webster et al., 1992) but they do not undergo homologous recombination within RNA segments (Boni et al., 2008 (Boni et al., , 2010 , meaning that 'origins' questions for influenza outbreaks can always be reduced to origins questions for each of influenza's eight RNA segments. For coronaviruses however, recombination means that small genomic sub-regions can have independent origins, identifiable if sufficient sampling has been done in the animal reservoirs that support the endemic circulation, coinfection, and recombination that appear to be common.\\nHere, we analyse the evolutionary history of SARS-CoV-2, using available genomic data on sarbecoviruses.\\nWe demonstrate that the sarbecoviruses circulating in horseshoe bats have complex recombination histories 5 / 25 permitting generation of novel viral variants, as reported by others (He et al., 2014; Hon et al., 2008; Hu et al., 2017; Li et al., 2020b; Lin et al., 2017; Wang et al., 2017; Yuan et al., 2010; Zhou et al., 2020a) .\\nInterestingly, despite the lineage leading to SARS-CoV-2 acquiring residues in its Spike (S) protein's receptorbinding domain (RBD) permitting use of human ACE2 (Wan et al., 2020) and being closer to the pangolin virus than RaTG13 in this region (Lam et al., 2020 ) -a signal that suggests recombination -the divergence patterns in the S protein show no evidence of recombination between SARS-CoV-2 and known sarbecoviruses. Our results indicate the presence of a single lineage circulating in bats with properties that allowed it to infect human cells, as previously described for the bat sarbecoviruses related to the first SARS-CoV lineage (Ge et al., 2013; Menachery et al., 2015 Menachery et al., , 2016 .\\nTo gauge the length of time this lineage has circulated in bats we estimate the time to most recent common ancestor (tMRCA) of SARS-CoV-2 and RaTG13. We use three bioinformatic approaches to remove the effects of recombination, and we combine these approaches to identify putative non-recombinant regions that can be used for reliable phylogenetic reconstruction and dating. Collectively our analyses point to bats being the primary reservoir for the SARS-CoV-2 lineage. While it is possible that pangolins may have acted as an intermediate species facilitating transmission to humans, the evidence is consistent with the virus having evolved in bats resulting in bat sarbecoviruses that can replicate in the upper respiratory tract of both humans and pangolins (Zhang and Holmes, 2020; Zhou et al., 2020a) .\\nOf the 68 sequences in the aligned sarbecovirus sequence set, 67 show evidence of mosaicism, indicating involvement in homologous recombination either directly with identifiable parentals or in their deeper shared evolutionary history, i.e., due to shared ancestral recombination events (all Dunn-Sidak corrected p < 4 \\u00d7 10 -4 , 3SEQ (Lam et al., 2018) ). This is evidence for numerous recombination events occurring in the evolutionary history of the sarbecoviruses ; however, identifying all past events and their temporal order (Eden et al., 2013) is challenging. Figure 1 (top) shows the distribution of all identified breakpoints (using 3SEQ's exhaustive triplet search) by the number of candidate recombinant sequences supporting them. The histogram allows for the identification of non-recombining regions by revealing regions with no breakpoints. Sorting these breakpoint-free regions (BFRs) by length results in two segments longer than 5kb: an ORF1a sub-region spanning nucleotides 3625-9150 and the first half of ORF1b spanning nucleotides 13291-19628. Eight other BFRs longer than 500nt were identified. Of the nine breakpoints (defining these ten BFRs), four showed phylogenetic incongruence (PI) signals with bootstrap support higher than 80%, adopting previously published criteria on using a combination of mosaic and PIsignals to show evidence of past recombination events (Boni et al., 2010) . All four of these breakpoints were also identified with the tree-based recombination detection method GARD (Kosakovsky Pond et al., 2006) .\\nThe extent of the sarbecoviruses' recombination history can be demonstrated by five phylogenetic trees inferred from BFRs or concatenated adjacent BFRs (Figure 1, bottom) . BFRs were concatenated if no 6 / 25 phylogenetic incongruence signal could be identified between them. When viewing the last 7kb of the genome a clade of viruses from northeast (NE) China appears to cluster with sequences from south-central (SC) Chinese provinces, but when inspecting trees from different parts of ORF1ab the NE China clade is phylogenetically separated from the SC China clade. Individual sequences such as RpShaanxi2011, Guangxi GX2013, and two sequences from Zhejiang province (CoVZXC21/CoVZC45), as previously shown Zhou et al., 2020a) , show strong phylogenetic recombination signals, as they fall on different evolutionary lineages (with bootstrap support > 80%) depending what region of the genome is being examined.\\nDespite this high frequency of recombination among bat viruses, the 'block-like' nature of the genome permits retrieval of a clean sub-alignment for phylogenetic analysis. Conservatively, we can combine three >2kb BFRs identified above into a putative non-recombining region 1 (NRR1), after removing five sequences that appear to be recombinants and two small sub-regions of the longer region (nucleotides 13291-19628).\\nAlternatively, combining 3SEQ-inferred breakpoints with GARD-inferred breakpoints and the necessity of PI-signals for inferring recombination, we can use the 9.9kb region spanning nucleotides 11885-21753 (NRR2) as a putative non-recombining region; this approach is conservative in identifying breakpoints but not conservative in identifying non-recombining regions. Using a third consensus-based approach for identifying recombinant regions in individual sequences -with six different recombination detection methods in RDP5 (Martin et al., 2015) -gives a putative recombination-free alignment that we call non-recombinant alignment 3 (NRA3) (see Methods). Figure 2 , with phylogenies reconstructed for two major sub-regions of NRR1.\\nAs the SARS-CoV-2 S protein has been implicated in past recombination events or possibly convergent evolution (Lam et al., 2020) , we specifically investigated several sub-regions of the S proteinthe N-terminal domain of S1, the C-terminal domain of S1, the variable loop region of the C-terminal domain, and S2. The variable loop region in SARS-CoV-2 shows closer identity to the 2019 pangolin coronavirus sequence than to the RaTG13 bat virus, supported by phylogenetic inference (Figure 3 ). On first examination this would suggest that that SARS-CoV-2 is a recombinant of an ancestor of Pangolin-2019 and RaTG13.\\nHowever, on closer inspection, the relative divergences in the phylogenetic tree (Figure 3 , bottom) show that SARS-CoV-2 is unlikely to have acquired the variable loop from an ancestor of Pangolin-2019 as these two sequences are approximately 10% divergent throughout the entire S protein (excluding the NTD). It is RaTG13 that is divergent in the variable loop region and is the likely product of recombination, acquiring a divergent 7 / 25 variable loop from an as yet unsampled bat sarbecovirus. This is notable because the variable loop region contains the six key contact residues in the receptor binding domain (RBD) that give SARS-CoV-2 its ACE2 binding specificity Wan et al., 2020) . These residues are also in the 2019 pangolin coronavirus sequence. The most parsimonious explanation for these shared residues is a single bat virus lineage (including the common ancestor of SARS-CoV-2, RaTG13, and Pangolin Guangdong 2019) has the ACE2 specific residues, rather than the SARS-CoV-2 being recombinant. This provides compelling support for the SARS-CoV-2 lineage being the consequence of a direct or nearly-direct zoonotic jump from bats because the key ACE2 binding residues were present in viruses circulating in bats.\\nUsing the most conservative approach to identifying a non-recombinant genomic region (NRR1), SARS-CoV-2 forms a sister lineage with RaTG13, with genetically related cousin lineages of coronavirus sampled in pangolins in Guangdong and Guangxi provinces ( Figure 2 ). Given that these pangolin viruses are ancestral to the progenitor of the RaTG13/SARS-CoV-2 lineage, it is more likely that they are also picking up viruses from bats. While pangolins could be acting as intermediate hosts for bat viruses to get into humansthey develop severe respiratory disease (Lie et al., 2019) and commonly come into contact with people as they are trafficked in large numbers for consumption and use in Chinese medicine -pangolin infection is not a requirement for bat viruses to cross into humans.\\nPhylogenies of sub-regions of NRR1 depict an appreciable degree of spatial structuring of the sarbecoviruses population across different regions ( Figure 2 ). One cluster consists of viruses from provinces in southcentral (SC) China (Guangxi, Yunnan, Guizhou) as well as one sequence from Sichuan province. The major cluster in its sister lineage is almost exclusively occupied by viruses from provinces in northeast (NE) and central China (Hubei, Shaanxi, Shanxi, Henan, Hebei, Zheijang and Jilin). showing two trimmed segments to yield Region A' (nt 13291-14932, 15405-17162, 18009-19628) . Region B spans nucleotides 3625-9150, and region C spans 9261-11795. To avoid artefacts due to recombination, we focused on the non-recombining regions NRR1, NRR2, and the recombination-masked alignment NRA3 for inferring time-measured evolutionary histories. Visual exploration using TempEst indicates there is no evidence for temporal signal in these data sets ( Figure S1 ). This is not surprising for diverse viral populations with relatively deep evolutionary histories. In such cases, even moderate rate variation among long deep phylogenetic branches will significantly impact expected root-to-tip divergences over a sampling time range that represents only a small fraction of the evolutionary history (Trova et al., 2015) . However, formal testing using marginal likelihood estimation (Duchene et al., 2019) does not reject the absence of temporal signal in all three data sets (Table S1) , albeit without strong support in favor of temporal signal (log Bayes factor support of 3, 10, and 3 for NRR1, NRR2, and NRA3 respectively).\\nIn the absence of a strong temporal signal, we sought to identify a suitable prior rate distribution to calibrate the time-measured trees by examining several coronaviruses sampled through time, including HCoV-OC43, MERS-CoV, and SARS-CoV virus genomes. These data sets were subjected to the same recombination masking approach as NRA3 and were characterized by a strong temporal signal (Figure 4 ), but also by markedly different evolutionary rates. Specifically, using a formal Bayesian approach (Suchard et al., 2018) (see Methods), we estimate a fast evolutionary rate (0.00169 subst/site/yr, 95% highest posterior density We infer time-measured evolutionary histories using a Bayesian phylogenetic approach while incorporating rate priors based on the mean MERS-CoV and HCoV-OC43 rates and with standard deviations that allow for more uncertainty than the empirical estimates for both viruses (see Methods). Using both prior distributions, this results in six highly similar posterior rate estimates for NRR1, NRR2, and NRA3, centered around 0.00055 subst/site/yr. The fact that these estimates lie in between the MERS-CoV and HCoV-OC43 14 / 25 rate is consistent with the intermediate sampling time range of about 18 years ( Figure 5 ). The consistency of the posterior rates for the different prior means also implies that the data do contribute to the evolutionary rate estimate despite the fact that a temporal signal was visually not apparent ( Figure S1 ). Below, we report divergence time estimates based on the HCoV-OC43-centred rate priors, but also summarize similar corresponding estimates for the MERS-CoV-centred rate priors in Supplementary Figure S3 . are in general agreement with estimates using NRR2 and NRA3, which result in divergence times of 1982 [1948,2009] and 1948 [1879-1999] , respectively, for SARS-CoV-2, and estimates of 1952 [1906, 1989] and 1970 [1932-1996] , respectively, for the divergence time of SARS-CoV from its closest known bat ancestor.\\nThe SARS-CoV divergence times are somewhat earlier than previously estimated dates (Hon et al., 2008) as previous estimates were obtained using a collection of SARS-CoV genomes from human and civet hosts (as well as a few closely related bat genomes) which implies that the evolutionary rates were predominantly informed by the short-term SARS outbreak scale and likely biased upwards for the time scale under investigation. Indeed, the rates reported by these studies are in line with the short-term SARS rates we estimate ( Figure 4) . The estimated divergence times for the pangolin virus most closely related to SARS-CoV-2/RaTG13 range from 1851 [1730, 1958] to 1877 [1746, 1986] , indicating that these pangolin lineages were acquired from bat viruses divergent to those that gave rise to SARS-CoV-2. Current sampling of pangolins does not implicate them as an intermediate host.\\nIdentifying the origins of an emerging pathogen can be critical during the early stages of an outbreak as it may allow for containment measures to be precisely targeted at a stage when the number of daily new infections is still low. Early detection via genomics and sequencing was not possible during SE Asia's initial outbreaks of avian influenza H5N1 (1997, (2003) (2004) (Ge et al., 2013; Menachery et al., 2015 Menachery et al., , 2016 , appropriate measures should have been in place for outbreaks of novel coronaviruses. The key to successful surveillance is that we know which human-adapted viral phenotypes to look for (Holmes et al., 2018) .\\nThe key difficulty in inferring reliable evolutionary histories for coronaviruses is that their high recombination rate (Graham and Baric, 2010; Su et al., 2016) violates the assumption of standard phylogenetic approaches because different parts of the genome will have different origins. In order to begin characterizing any ancestral relationships for SARS-CoV-2, non-recombinant regions of the genome must be identified, so that reliable phylogenetic reconstruction and dating can be performed. Evolutionary rate estimation can be profoundly affected by the presence of recombination (Schierup and Hein, 1999) . As there is no single accepted method of inferring breakpoints and identifying clean sub-regions with high certainty, we implemented several approaches to identifying three classic statistical signals of recombination: mosaicism, 16 / 25 Green boxplots show the TMRCA estimate for the RaTG13/SARS-CoV-2 lineage and its most closely related pangolin lineage (Guangdong 2020). Transparent boxes of the interquartile range width and with the same colours are superimposed to highlight the overlap between the estimates. In Supplementary Figure S3 , we compare these divergence time estimates to those obtained using the MERS-CoV-centred rate priors for NRR1, NRR2 and NRA3.\\nphylogenetic incongruence and excessive homoplasy (Posada et al., 2002) . Our most conservative approach attempted to ensure that putative non-recombining regions had no mosaic or phylogenetic incongruence signals. A second approach was conservative with respect to breakpoint identification, while a third approach attempted to minimize the number of regions removed while still minimizing signals of mosaicism and homoplasy. The origins we present in Figure 5 (NRR1) are conservative in the sense that NRR1 is more likely to be non-recombinant than NRR2 or NRA3. As the estimated rates and divergence dates were highly similar in the three data sets analyzed, we conclude that our estimates are robust to the method of identifying the genome's non-recombinant regions.\\nDue to the absence of temporal signal in the sarbecovirus data sets, we used informative prior distributions on the evolutionary rate to estimate divergence dates. We show that such prior calibration information can be obtained from other coronaviruses (SARS-CoV, MERS-CoV, and HCoV-OC43), but their rates vary with the time scale of sample collection. In the presence of time-dependent rate variation, a widely observed phenomenon for viruses (Duchene et al., 2014; Aiewsakun and Katzourakis, 2016; Membrebe et al., 2019) , the slower rates appear more appropriate for sarbecoviruses that currently encompass a sampling time range of about 18 years. We therefore centered the prior rate distributions on the mean HCoV-OC43 and MERS-CoV rate, but permitted the distributions a greater variance than the empirical rate estimates for these lineages. This approach resulted in similar posterior rates despite the different prior means, implying that the sarbecovirus data do inform the rate estimate even though root-to-top temporal signal was not apparent.\\nThe relatively fast evolutionary rate means that it is most appropriate to estimate shallow nodes in the sarbecovirus evolutionary history including the divergence times of SARS-CoV and SARS-CoV-2 from their most closely related bat viruses. Accurately estimating deeper nodes would require adequately accommodating time-dependent rate variation. While such models have recently been made available, we lack the information to calibrate the rate decline through time (e.g. through internal node calibrations (Membrebe et al., 2019) ). As a proxy, it would be possible to model the long-term purifying selection dynamics as a major source of time-dependent rates rates (Duchene et al., 2014; Aiewsakun and Katzourakis, 2016; Membrebe et al., 2019) , but this is beyond the scope of the current study. The assumption of long-term purifying selection would imply that coronaviruses are at endemic equilibrium with their natural host species, horseshoe bats, to which they are presumably well-adapted. Although there is currently little evidence supporting or contradicting strong positive selection in the sarbecovirus lineage, if multiple host species were to be identified that are able to support endemic viral transmission, then the positive selection that would be associated with new species adaptations would need to be considered when inferring evolutionary rate variation through time.\\nOf importance for future emergence events is the appreciation that SARS-CoV-2 has emerged from the same horseshoe bat subgenus that harbours SARS-like coronaviruses. Another similarity between SARS-CoV and SARS-CoV-2 is their divergence time (40-70 years ago) from currently known extant bat-virus lineages ( Figure 5 ). This long divergence period indicates there are unsampled viruses circulating in horseshoe bats that have zoonotic potential . While there is clear involvement of other mammalian species -specifically pangolins for SARS-CoV-2 -as a plausible conduit for transmission to humans, there 18 / 25 is no evidence pangolins are facilitating adaptation to humans. A hypothesis of snakes as intermediate hosts of SARS-CoV-2 was posited during the early epidemic phase (Ji et al., 2020) , but we found no evidence of this (Anderson, 2020; Robertson, 2020) ; see Supplementary Section 3.\\nWith horseshoe bats currently the most plausible source for SARS-CoV-2, it is important to consider that sarbecoviruses circulate in a variety of horseshoe bat species with often widely overlapping species ranges (Wong et al., 2019 ). Yet, the viral population is largely spatially structured according to provinces in the south and southeast on one lineage, and provinces in the center, east, and northeast on another (Figure 2) . This boundary appears to be rarely crossed. Two exceptions can be seen in the relatively close relationship of viruses from Hong Kong bats to viruses from Zheijang (with two of the latter, CoVZC45 and CoVZXC21, identified as recombinants) and a recombinant virus from Sichuan for which part of the genome (region B of SC2018 in Figure 2 ) clusters with viruses from provinces in the center, east, and northeast of China. SARS-CoV-2 and RaTG13 are also exceptions as they were sampled from Hubei and Yunnan, respectively. The fact that they are geographically relatively distant is in agreement with their somewhat distant tMRCA because the spatial structure suggests that migration between their locations may be uncommon. From this perspective, it may be useful to perform surveillance for more closely related virus to SARS-CoV-2 along the gradient from Yunnan to Hubei.\\nIt is clear from our analysis that viruses closely related to SARS-CoV-2 have been circulating in horseshoe bats for many decades. The substantial unsampled diversity on the SARS-CoV-2/RaTG13 lineage suggests that there is a major clade of bat sarbecoviruses with generalist properties -with respect to their ability to infect a range of mammalian cells -that facilitated its jump to humans and may do so again. Although the human ACE2-compatible receptor binding domain was very likely to have been present in a bat sarbecovirus lineage that ultimately lead to SARS-CoV-2, this RBD sequence has thus far only been found in a few pangolin viruses. Furthermore, the other key feature thought to be instrumental to SARS-CoV-2's ability to infect humans -a polybasic cleavage site insertion in the Spike protein -has been seen in another close bat relative of the SARS-CoV-2 virus ; however, the sequence is different and it is likely an independent event.\\nThe existing diversity and dynamic process of recombination amongst lineages in the bat reservoir demonstrate how difficult it will be to identify viruses with potential to cause significant human outbreaks before they emerge. This underscores the need for a global network of real-time human disease surveillance systems like that which identified the unusual cluster of pneumonia in Wuhan in December 2019 and the rapid deployment of functional studies and genomic tools for pathogen identification and characterization.\\nSarbecovirus data. Complete genome sequence data were downloaded from GenBank and ViPIR; accession numbers available in Supplementary Section 4. Sequences were aligned by MAFTT (Katoh et al., 2009) , with a final alignment length of 30927, and used in the analyses below.\\nHCoV-OC43. We compiled a data set including 27 human coronavirus OC43 virus genomes and 10 related animal virus genomes (6 bovine, 3 white-tailed deer and one canine virus). The canine viral genome was excluded from the Bayesian phylogenetic analyses because temporal signal analyses (see below) indicated that it was an outlier.\\nMERS-CoV. We extracted a similar number (n = 35) of genomes from a MERS-CoV data set analysed by Dudas et al (2018) using the phylogenetic diversity analyser tool (Chernomor et al., 2015) .\\nSARS-CoV. We compiled a set of 69 SARS-CoV genomes including 58 genomes sampled from human and 11 sampled from civets and raccoon dogs. This data set comprises an updated version of the one used in Hon et al (2008) .\\nAs coronaviruses are known to be highly recombinant, we used three different approaches to identify nonrecombinant regions to be used in our Bayesian time-calibrated phylogenetic inference.\\nFirst, we took an approach that relies on identifying mosaic regions (via 3SEQ) that are also supported by phylogenetic incongruence (PI) signals (Boni et al., 2010) . As 3SEQ is the most powerful the mosaic methods (Boni et al., 2007) , we used 3SEQ to identify the best supported breakpoint history for each potential child (recombinant) sequence in the data set. A single 3SEQ run on the genome alignment resulted in 67 out of 68 sequences supporting some recombination in the past, with multiple candidate breakpoint ranges listed for each putative recombinant. Then, we (a) collected all breakpoints into a single set, (b) complemented this set to generate a set of non-breakpoints, (c) grouped the non-breakpoints into contiguous breakpoint-free regions (BFRs), and (d) sorted these regions by length. A phylogenetic tree -using RAxML v8.2.8 (Stamatakis, 2014 (Stamatakis, , 2006 , GTR+\\u0393 model, 100 bootstrap replicates -was inferred for each BFR longer than 500nt.\\nWe considered both (1) the possibility that BFRs could be combined into larger non-recombinant regions, and (2) Section 5), region D and shorter BFRs were not further included into a combined putative non-recombinant regions.\\nRegions A, B, and C were further examined for mosaic signals by 3SEQ; all showed signs of mosaicism. In region A, we removed sub-region A1 (nt positions 3872-4716 within region A) and sub-region A4 (nt 1642-2113) as both of these showed PI-signals with other sub-regions of region A. After removal of A1 and A4, we named the new region A\\u2032. In addition, sequences NC_014470 (Bulgaria 2008), CoVZXC21, CoVZC45, and DQ412042 (Hubei-Yichang) needed to be removed to maintain a clean non-recombinant signal in A\\u2032. Region B showed no PI-signals within the region, except one including sequence SC2018 (Sichuan), thus this sequence was also removed from the set. Region C showed no PI-signals within it.\\nCombining regions A\\u2032, B, and C, and removing the five mentioned sequences gives us putative nonrecombining region 1, or NRR1, as an alignment of 63 sequences.\\nSecond, we wanted to construct non-recombinant regions where our approach to breakpoint identification was as conservative as possible. In this approach, we considered a breakpoint supported only if it had three types of statistical support from (1) mosaic signals identified by 3SEQ, (2) PI-signals identified by building trees around 3SEQ's breakpoints, (3) the GARD algorithm (Kosakovsky Pond et al., 2006) which identifies breakpoints by identifying PI-signals across proposed breakpoints. Since 3SEQ identified 10 BFRs longer than 500nt, we used GARD's inference on 10, 11, and 12 breakpoints. A reduced sequence set of 25 sequences chosen to capture the breadth of diversity in the sarbecoviruses (obvious recombinants not involving the SARS-CoV-2 lineage were also excluded) was used as GARD is computationally intensive. GARD identified eight breakpoints that were also within 50nt of breakpoints identified by 3SEQ. PI-signals were identified (with bootstrap support >80%) for seven of these eight breakpoints: positions 1684, 3046, 9237, 11885, 21753, 22773, and 24628 . Using these breakpoints, the longest putative non-recombining segment (nt 11885-21753) is 9.9kb long, and we call this region NRR2. Because this approach to breakpoint identification is conservative, the approach to identifying non-recombinant regions is not conservative.\\nOur third approach involved identifying breakpoints and masking minor recombinant regions (with gaps, which are treated as unobserved characters in probabilistic phylogenetic approaches). Specifically, we used a combination of six methods implemented in RDP5 (RDP, GENECONV, MaxChi, Bootscan SisScan and 3SEQ) and considered recombination signals detected by more than two methods for breakpoint identification. Except for specifying that sequences are linear, all settings were kept to their defaults. Based on the identified breakpoints in each genome, only the major non-recombinant region is kept in each genome while other regions are masked. To evaluate the performance procedure, we confirmed that the recombination masking resulted in (i) a significantly different outcome of the PHI-test (Bruen et al., 2006) , (ii) a removal of well-supported (bootstrap value >95%) incompatible splits in Neighbor-Nets (Bryant and Moulton, 2004) , and (iii) a near-complete reduction of mosaic signal as identified by 3SEQ. If the latter still identified nonnegligible recombination signal, we removed additional genomes that were identified as the major contributors to the remaining signal. This produced non-recombining alignment NRA3, which included 63 of the 68 genomes.\\nWe focused on these three non-recombining regions/alignments for divergence time estimation. This avoids inappropriately modelling evolutionary processes with recombination on strictly bifurcating trees, which can result in different artefacts such as homoplasies that inflate branch lengths and result in apparently longer evolutionary divergence times. In order to examine temporal signal in the sequenced data, we plotted root-totip divergence against sampling time using TempEst based on a maximum likelihood tree. The latter was reconstructed using IQTREE (Nguyen et al., 2014 ) under a General-Time Reversible (GTR) model with a discrete gamma distribution to model among-site rate variation.\\nTime-measured phylogenetic reconstruction was performed using a Bayesian approach implemented in BEAST (Suchard et al., 2018) . When the genomic data included both coding and noncoding regions we used a single GTR+\\u0393 substitution model; for concatenated coding genes we partitioned the alignment by codon positions and specified an independent GTR+\\u0393 model for each partition with a separate gamma model to accommodate among-site rate variation. We used an uncorrelated relaxed clock model with a lognormal distribution for all data sets, except for the low-diversity SARS data for which we specified a strict molecular clock model. For the HCoV-OC43, MERS-CoV, and SARS data sets we specified flexible skygrid coalescent tree priors. In the absence of any reasonable prior knowledge on the tMRCA of the sarbecovirus data sets (which is required for the grid specification in a skygrid model), we specified a simpler constant size population prior. As informative rate priors for the analysis of the sarbecovirus data sets, we used two different normal prior distributions: one with a mean of 0.00078 and a standard deviation of 0.0003 and one with a mean of 0.00024 and standard deviation of 0.0001. These means are based on the mean rates estimated for MERS-CoV and HCoV-OC43 respectively, while the standard deviations are set ten times larger than the empirical standard deviations to allow more prior uncertainty and avoid strong bias ( Supplementary Fig. 2 ).\\nIn our analyses of the sarbecovirus data sets, we incorporated the uncertainty of the sampling dates when exact dates were not available. To estimate nonsynonymous over synonymous rate ratios for the concatenated coding genes, we used the empirical Bayes 'Renaissance counting' procedure (Lemey et al., 2012) . Temporal signal was tested using a recently developed marginal likelihood estimation procedure (Duchene et al., 2019) ( Table S1) .\\nPosterior distributions were approximated through Markov chain Monte Carlo sampling, which were run sufficiently long to ensure effective sampling sizes > 100. BEAST inferences made use of the BEAGLEv3 library (Yres et al., 2019) for efficient likelihood computations. We used TreeAnnotator to summarize the posterior tree distributions and annotated the estimated to a maximum clade credibility tree, which was visualized using FigTree.\\nAll authors contributed to analyses and interpretations. DLR and XJ performed recombination and phylogenetic analysis, and annotated the virus names with geographical and sampling dates. AR performed the S recombination analysis. PB and TC performed the codon-usage analysis. MFB performed the recombination analysis for non-recombining regions 1 and 2, breakpoint analysis, and phylogenetic inference on recombinant segments. PL performed recombination analysis for non-recombining alignment 3, calibration of rate of evolution, and phylogenetic reconstruction and dating. TT-YL collected SARS-CoV data. PL, MFB, and DLR wrote the first draft of the manuscript, and all authors contributed to manuscript editing.\\n\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2.57935637188917,\n        \"min\": 7.638880361847612,\n        \"max\": 22.37983104725695,\n        \"num_unique_values\": 48,\n        \"samples\": [\n          8.950876801902982,\n          8.09971027613814,\n          9.099620231180763\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 100
        }
      ],
      "source": [
        "p = '''\n",
        "Animal host(s) and any evidence of continued spill-over to humans\n",
        "'''\n",
        "res = searchAndGetAnswer(p)\n",
        "res"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "NicMnPNeYSsA"
      },
      "source": [
        "### Q5: *Socioeconomic and behavioral risk factors for this spill-over*"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 101,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "XGlFcBiRYSsA",
        "outputId": "827211f0-cdc3-4f37-9c3c-ff6f63e5dd66"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "No valid answers found.\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                                title  \\\n",
              "0   Household financial vulnerability in Indonesia...   \n",
              "1   BigO: A public health decision support system ...   \n",
              "2             A deadly spillover: SARS-CoV-2 outbreak   \n",
              "3   Is Brazil prepared for the new era of infectio...   \n",
              "4   On the dynamics emerging from pandemics and in...   \n",
              "5   Exposure to air pollution and COVID-19 mortali...   \n",
              "6   COVID-19: the First Documented Coronavirus Pan...   \n",
              "7   Invasion Science and the Global Spread of SARS...   \n",
              "8              Horses as a Crucial Part of One Health   \n",
              "9   Climate affects global patterns of COVID-19 ea...   \n",
              "10  Synergistic China–US Ecological Research is Es...   \n",
              "11  Behavioral and Emotional Disorders in Children...   \n",
              "12  Racial/ethnic and socioeconomic disparities of...   \n",
              "13  Behavioral Food Addiction During Lockdown: Tim...   \n",
              "14  Impacts of social and economic factors on the ...   \n",
              "15  Spillover of SARS-CoV-2 into novel wild hosts ...   \n",
              "16  County-level factors influence the trajectory ...   \n",
              "17  Spillovers and diversification potential of ba...   \n",
              "18  Impacts of social and economic factors on the ...   \n",
              "19  Estimated inequities in COVID-19 infection fat...   \n",
              "20  Behavioral considerations and impact on person...   \n",
              "21  Understanding the Socio-Economic Disruption in...   \n",
              "22  Race, Socioeconomic Deprivation, and Hospitali...   \n",
              "23  The socio-economic determinants of the coronav...   \n",
              "24  A multi-group SEIRA model for the spread of CO...   \n",
              "25  Meta-regression of COVID-19 prevalence/fatalit...   \n",
              "26  Social and behavioral health responses to COVI...   \n",
              "27  Investigating spatial variability in COVID-19 ...   \n",
              "28                           New Virus, New Challenge   \n",
              "29  Globalized low-income countries may experience...   \n",
              "30  Why people failed to adhere to COVID-19 preven...   \n",
              "31        Impact of COVID-19 on Indian Sugar Industry   \n",
              "32  COVID-19 in patients with diabetes: risk facto...   \n",
              "33  Racial, Economic and Health Inequality and COV...   \n",
              "34  Stroke in patients with SARS-CoV-2 infection: ...   \n",
              "35  Determining risk factors for mortality in live...   \n",
              "36  Critical patients with coronavirus disease 201...   \n",
              "37  Causal Inference on Networks under Continuous ...   \n",
              "38  Outcomes from COVID-19 across the range of fra...   \n",
              "39  Risk factors for severe COVID-19 illness in he...   \n",
              "40  Coronavirus Disease (COVID-19) in Italy: Analy...   \n",
              "41  Prevalence, risk factors, and clinical correla...   \n",
              "42  Reply to “COVID-19 in persons with haematologi...   \n",
              "43  Early forecasting of the potential risk zones ...   \n",
              "44  Epidemiological and clinical characteristics o...   \n",
              "45  Symptom profiles and risk factors for hospital...   \n",
              "46  Initial Cluster of Novel Coronavirus (2019-nCo...   \n",
              "47  Applying the SEIR Model in Forecasting The COV...   \n",
              "48  Modifiable and non-modifiable risk factors for...   \n",
              "49  Letter to the Editor on ‘Risk Factors for Unpl...   \n",
              "\n",
              "                                             abstract  \\\n",
              "0   This study assesses the level of financial vul...   \n",
              "1   Obesity is a complex disease and its prevalenc...   \n",
              "2                                                 NaN   \n",
              "3   Brazil must maintain a focus on enhancing the ...   \n",
              "4   This position paper discusses emerging behavio...   \n",
              "5   Objectives: United States government scientist...   \n",
              "6   The novel human coronavirus disease COVID-19 h...   \n",
              "7   Abstract Emerging infectious diseases like COV...   \n",
              "8   One Health (OH) is a crucial concept, where th...   \n",
              "9   Environmental factors, including seasonal clim...   \n",
              "10  The risk of a zoonotic pandemic disease threat...   \n",
              "11                                                NaN   \n",
              "12  We investigated the dependence of Covid-19 att...   \n",
              "13                                                NaN   \n",
              "14  This study models local and cross-city transmi...   \n",
              "15  Abstract There is evidence that the current ou...   \n",
              "16  With new cases of Covid-19 surging in the Unit...   \n",
              "17  Abstract We examine the network spillovers, po...   \n",
              "18  This paper examines the role of various socioe...   \n",
              "19  There is limited evidence as to how COVID-19 i...   \n",
              "20                                                NaN   \n",
              "21  In this paper, we collect and study Twitter co...   \n",
              "22  Preliminary reports suggest that the Coronavir...   \n",
              "23  The magnitude of the coronavirus disease (COVI...   \n",
              "24  The outbreak and propagation of COVID-19 have ...   \n",
              "25  To screen potential risk and protective socioe...   \n",
              "26  Our public health approaches to addressing COV...   \n",
              "27  Background: Spain has been disproportionately ...   \n",
              "28  The ongoing COVID-19 pandemic has made us to c...   \n",
              "29  Understanding the factors underpinning COVID-1...   \n",
              "30  These violations might endanger the community ...   \n",
              "31  The Indian sugar industry, a significant playe...   \n",
              "32                                                NaN   \n",
              "33  Abstract Background: There is preliminary evid...   \n",
              "34  BACKGROUND: Italy is one of the most affected ...   \n",
              "35                                                NaN   \n",
              "36                                                NaN   \n",
              "37  This paper presents a methodology to draw caus...   \n",
              "38  Purpose Our aim was to quantify the mortality ...   \n",
              "39                                                NaN   \n",
              "40                                                NaN   \n",
              "41                                                NaN   \n",
              "42                                                NaN   \n",
              "43  Recently, the coronavirus disease 2019 (COVID-...   \n",
              "44  Background: The first case of COVID-19 was det...   \n",
              "45                                                NaN   \n",
              "46  Reanalysis of the epidemic curve from the init...   \n",
              "47  On March 18, 2020 the Malaysian government imp...   \n",
              "48  Background Information on risk factors for COV...   \n",
              "49                                                NaN   \n",
              "\n",
              "                                                  url  \\\n",
              "0   https://api.elsevier.com/content/article/pii/S...   \n",
              "1              https://arxiv.org/pdf/2005.02928v1.pdf   \n",
              "2   https://www.ncbi.nlm.nih.gov/pubmed/32321324/;...   \n",
              "3   https://www.ncbi.nlm.nih.gov/pubmed/32374800/;...   \n",
              "4              https://arxiv.org/pdf/2004.08917v1.pdf   \n",
              "5         https://doi.org/10.1101/2020.04.05.20054502   \n",
              "6   https://www.ncbi.nlm.nih.gov/pubmed/32387617/;...   \n",
              "7   https://www.sciencedirect.com/science/article/...   \n",
              "8   https://www.ncbi.nlm.nih.gov/pubmed/32121327/;...   \n",
              "9         https://doi.org/10.1101/2020.03.23.20040501   \n",
              "10  https://doi.org/10.1007/s10393-020-01471-2; ht...   \n",
              "11  https://www.ncbi.nlm.nih.gov/pubmed/32248989/;...   \n",
              "12             https://arxiv.org/pdf/2004.12175v1.pdf   \n",
              "13  https://doi.org/10.1007/s11695-020-04649-3; ht...   \n",
              "14  https://www.ncbi.nlm.nih.gov/pubmed/32395017/;...   \n",
              "15  https://api.elsevier.com/content/article/pii/S...   \n",
              "16  http://medrxiv.org/cgi/content/short/2020.05.0...   \n",
              "17  https://api.elsevier.com/content/article/pii/S...   \n",
              "18        https://doi.org/10.1101/2020.03.13.20035238   \n",
              "19  http://medrxiv.org/cgi/content/short/2020.04.2...   \n",
              "20  https://doi.org/10.1016/j.jaad.2020.03.013; ht...   \n",
              "21             https://arxiv.org/pdf/2004.05451v1.pdf   \n",
              "22  http://medrxiv.org/cgi/content/short/2020.04.2...   \n",
              "23  http://medrxiv.org/cgi/content/short/2020.04.1...   \n",
              "24  https://www.sciencedirect.com/science/article/...   \n",
              "25  http://medrxiv.org/cgi/content/short/2020.05.2...   \n",
              "26  https://doi.org/10.1007/s10865-020-00157-y; ht...   \n",
              "27        https://doi.org/10.1101/2020.04.14.20065524   \n",
              "28  https://www.sciencedirect.com/science/article/...   \n",
              "29        https://doi.org/10.1101/2020.03.31.20049122   \n",
              "30  https://doi.org/10.1017/ice.2020.245; https://...   \n",
              "31         https://doi.org/10.1007/s12355-020-00846-7   \n",
              "32  https://doi.org/10.1016/j.metabol.2020.154224;...   \n",
              "33  http://medrxiv.org/cgi/content/short/2020.04.2...   \n",
              "34  https://www.ncbi.nlm.nih.gov/pubmed/32436105/;...   \n",
              "35  https://api.elsevier.com/content/article/pii/S...   \n",
              "36  https://api.elsevier.com/content/article/pii/S...   \n",
              "37             https://arxiv.org/pdf/2004.13459v1.pdf   \n",
              "38  http://medrxiv.org/cgi/content/short/2020.05.2...   \n",
              "39  https://doi.org/10.1017/ice.2020.178; https://...   \n",
              "40  https://doi.org/10.3389/fmed.2020.00140; https...   \n",
              "41  https://www.ncbi.nlm.nih.gov/pubmed/32380272/;...   \n",
              "42  https://doi.org/10.1038/s41375-020-0877-y; htt...   \n",
              "43  https://api.elsevier.com/content/article/pii/S...   \n",
              "44  http://medrxiv.org/cgi/content/short/2020.04.2...   \n",
              "45  https://www.ncbi.nlm.nih.gov/pubmed/32437750/;...   \n",
              "46  https://doi.org/10.3390/jcm9020488; https://ww...   \n",
              "47  http://medrxiv.org/cgi/content/short/2020.04.1...   \n",
              "48  http://medrxiv.org/cgi/content/short/2020.04.2...   \n",
              "49  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   \n",
              "\n",
              "                                              authors  \\\n",
              "0   Noerhidajati, Sri; Purwoko, Agung Bayu; Werdan...   \n",
              "1   Diou, Christos; Sarafis, Ioannis; Papapanagiot...   \n",
              "2   Mori, Mattia; Capasso, Clemente; Carta, Fabriz...   \n",
              "3                              Vicente, Creuza Rachel   \n",
              "4                                    Leitner, Stephan   \n",
              "5   Wu, Xiao; Nethery, Rachel C.; Sabath, Benjamin...   \n",
              "6          Liu, Yen-Chin; Kuo, Rei-Lin; Shih, Shin-Ru   \n",
              "7           Nuñez, M. A.; Pauchard, A.; Ricciardi, A.   \n",
              "8   Lönker, Nelly Sophie; Fechner, Kim; Abd El Wah...   \n",
              "9        Ficetola, Gentile Francesco; Rubolini, Diego   \n",
              "10  Smiley Evans, Tierra; Shi, Zhengli; Boots, Mic...   \n",
              "11  Jiao, Wen Yan; Wang, Lin Na; Liu, Juan; Fang, ...   \n",
              "12  Dobin, Daniel; School, Alexander Dobin Candlew...   \n",
              "13  Cherikh, Faredj; Frey, Sébastien; Bel, Corali;...   \n",
              "14                       Qiu, Yun; Chen, Xi; Shi, Wei   \n",
              "15                Franklin, Alan B.; Bevins, Sarah N.   \n",
              "16                        Kranjac, A. W.; Kranjac, D.   \n",
              "17  Arreola Hernandez, Jose; Hoon Kang, Sang; Huss...   \n",
              "18                       Qiu, Yun; Chen, Xi; Shi, Wei   \n",
              "19  Steyn, Nicholas; Binny, Rachelle N; Hannah, Ka...   \n",
              "20                                   Kantor, Jonathan   \n",
              "21  Shanthakumar, Swaroop Gowdra; Seetharam, Anand...   \n",
              "22  Patel, A. P.; Paranjpe, M. D.; Kathiresan, N. ...   \n",
              "23  Stojkoski, V.; Utkovski, Z.; Jolakoski, P.; Te...   \n",
              "24  Contreras, Sebastián; Villavicencio, H. Andrés...   \n",
              "25  Takagi, H.; Kuno, T.; Yokoyama, Y.; Ueyama, H....   \n",
              "26                 Eaton, Lisa A.; Kalichman, Seth C.   \n",
              "27  Dahal, Sushma; Mizumoto, Kenji; Rothenberg, Ri...   \n",
              "28                                            Shi, Yi   \n",
              "29  Jaffe, Rodolfo; Ortiz Vera, Mabel Patricia; Ja...   \n",
              "30  Chan, Derwin K. C.; Zhang, Chun-Qing; Weman-Jo...   \n",
              "31        Solomon, S.; Rao, Govind Pratap; Swapna, M.   \n",
              "32         Klonoff, David C.; Umpierrez, Guillermo E.   \n",
              "33  Abedi, V.; Olulana, O.; Avula, V.; Chaudhary, ...   \n",
              "34  Morassi, Mauro; Bagatto, Daniele; Cobelli, Mil...   \n",
              "35  Webb, Gwilym J; Moon, Andrew M; Barnes, Eleano...   \n",
              "36  Li, Lingzhi; Zhang, Binghong; He, Bing; Gong, ...   \n",
              "37  Prete, Davide Del; Forastiere, Laura; Sciabola...   \n",
              "38  Miles, A.; Webb, T. E.; Mcloughlin, B.; Mannan...   \n",
              "39  Wander, Pandora L.; Orlov, Marika; Merel, Susa...   \n",
              "40          Di Lorenzo, Giuseppe; Di Trolio, Rossella   \n",
              "41  Li, Xingxing; Yu, Haihang; Bian, Guolin; Hu, Z...   \n",
              "42  Fattizzo, Bruno; Giannotta, Juri Alessandro; S...   \n",
              "43  Ren, Hongyan; Zhao, Lu; Zhang, An; Song, Liuyi...   \n",
              "44  Souza, William Marciel de; Buss, Lewis Fletche...   \n",
              "45  Díaz, Luis Antonio; García-Salum, Tamara; Fuen...   \n",
              "46  Nishiura, Hiroshi; Linton, Natalie M.; Akhmetz...   \n",
              "47                             Mahmud, A.; Lim, P. Y.   \n",
              "48  Ho, F. K.; Celis-Morales, C. A.; Gray, S. R.; ...   \n",
              "49                Kamath, Atul F.; Piuzzi, Nicolas S.   \n",
              "\n",
              "                                                 text      Score  \n",
              "0   The crisis in 2008, which was caused by securi...  12.580977  \n",
              "1   Obesity prevalence has been continuously risin...  11.940364  \n",
              "2   Coronaviruses (CoV) are a family of viruses th...  10.810611  \n",
              "3   Dear Editor:\\nThe world has been facing a new ...  10.182449  \n",
              "4   in the direct aftermath of a pandemic are key ...  10.110361  \n",
              "5   The scale of the COVID-19 public health emerge...   9.560090  \n",
              "6   Currently, people all over the world have been...   9.036929  \n",
              "7   A sinister combination of ecosystem alteration...   8.917510  \n",
              "8   One Health (OH) is a holistic approach which d...   8.913372  \n",
              "9   1 Abstract 1 2 Environmental factors, includin...   8.739939  \n",
              "10  The catastrophic health and economic impacts o...   8.630575  \n",
              "11  The COVID-19 outbreak first erupted in the cit...   8.540292  \n",
              "12  The Covid-19 disease first emerged in late 201...   8.373394  \n",
              "13  The world is currently facing the COVID-19 pan...   8.356851  \n",
              "14  The first pneumonia case of unknown cause was ...   8.263538  \n",
              "15  J o u r n a l P r e -p r o o f 2 infected huma...   8.193293  \n",
              "16  The presence of severe acute respiratory syndr...   8.164640  \n",
              "17  The banking sector plays an important role in ...   8.139073  \n",
              "18  Several clusters of patients with pneumonia of...   8.116073  \n",
              "19  The COVID-19 outbreak originated in Wuhan Chin...   8.020653  \n",
              "20  In this issue of the JAAD, Lan et al1 provide ...   8.012635  \n",
              "21  COVID-19 (also known as the novel coronavirus)...   8.004584  \n",
              "22  The Coronavirus Disease 2019 (COVID-19) pandem...   7.973779  \n",
              "23  The coronavirus pandemic began as a simple out...   7.949000  \n",
              "24  Since the outbreak of novel SARS-CoV2 in late ...   7.915906  \n",
              "25  Reuters reported \"The coronavirus mortality ra...   7.791638  \n",
              "26  Sustained, individual-level, behavioral change...   7.759786  \n",
              "27  The case fatality ratio is a useful metric to ...   7.747395  \n",
              "28  Yi Shi 1, * The ongoing COVID-19 pandemic has ...   7.638880  \n",
              "29  Understanding the factors underpinning COVID-1...   7.513772  \n",
              "30  The integrated model of self-determination the...   7.415810  \n",
              "31  Sugarcane is a long duration labour-intensive ...   7.404040  \n",
              "32  We agree with the conclusion by Dr. Hill and c...   7.260640  \n",
              "33  The complexity of managing patients with the C...   7.253174  \n",
              "34  Italy was the first country in Europe to witne...   7.194565  \n",
              "35  We read with great interest the Correspondence...   7.182662  \n",
              "36  Dear Editor,\\nWe read Yang's paper with great ...   7.182662  \n",
              "37  Policies often involve interventions that spil...   7.130043  \n",
              "38  While it is clear that mortality from SARS-CoV...   7.123729  \n",
              "39  \\nTo the Editor—We were very interested to rea...   7.106342  \n",
              "40  The lethality rate is determined as follows: t...   7.031626  \n",
              "41  Since December 2019, a large-scale outbreak of...   6.958465  \n",
              "42  We read with interest the article by He et al....   6.958465  \n",
              "43  Involving six continents, the outbreaks of the...   6.921999  \n",
              "44  The first case of COVID-19 was detected in Bra...   6.887059  \n",
              "45  San Carlos for their support to this project r...   6.886811  \n",
              "46  The clinical summary of the earliest cases of ...   6.875602  \n",
              "47  Universiti Putra Malaysia. CC-BY-NC-ND 4.0 Int...   6.856776  \n",
              "48  COVID-19, caused by SARS-CoV-2 infection, incl...   6.852470  \n",
              "49  We read with great interest the article by Suk...   6.816618  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-b9840663-304d-40e8-bc8d-2922516bf951\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>title</th>\n",
              "      <th>abstract</th>\n",
              "      <th>url</th>\n",
              "      <th>authors</th>\n",
              "      <th>text</th>\n",
              "      <th>Score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Household financial vulnerability in Indonesia...</td>\n",
              "      <td>This study assesses the level of financial vul...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Noerhidajati, Sri; Purwoko, Agung Bayu; Werdan...</td>\n",
              "      <td>The crisis in 2008, which was caused by securi...</td>\n",
              "      <td>12.580977</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>BigO: A public health decision support system ...</td>\n",
              "      <td>Obesity is a complex disease and its prevalenc...</td>\n",
              "      <td>https://arxiv.org/pdf/2005.02928v1.pdf</td>\n",
              "      <td>Diou, Christos; Sarafis, Ioannis; Papapanagiot...</td>\n",
              "      <td>Obesity prevalence has been continuously risin...</td>\n",
              "      <td>11.940364</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>A deadly spillover: SARS-CoV-2 outbreak</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32321324/;...</td>\n",
              "      <td>Mori, Mattia; Capasso, Clemente; Carta, Fabriz...</td>\n",
              "      <td>Coronaviruses (CoV) are a family of viruses th...</td>\n",
              "      <td>10.810611</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>Is Brazil prepared for the new era of infectio...</td>\n",
              "      <td>Brazil must maintain a focus on enhancing the ...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32374800/;...</td>\n",
              "      <td>Vicente, Creuza Rachel</td>\n",
              "      <td>Dear Editor:\\nThe world has been facing a new ...</td>\n",
              "      <td>10.182449</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>On the dynamics emerging from pandemics and in...</td>\n",
              "      <td>This position paper discusses emerging behavio...</td>\n",
              "      <td>https://arxiv.org/pdf/2004.08917v1.pdf</td>\n",
              "      <td>Leitner, Stephan</td>\n",
              "      <td>in the direct aftermath of a pandemic are key ...</td>\n",
              "      <td>10.110361</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>Exposure to air pollution and COVID-19 mortali...</td>\n",
              "      <td>Objectives: United States government scientist...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.05.20054502</td>\n",
              "      <td>Wu, Xiao; Nethery, Rachel C.; Sabath, Benjamin...</td>\n",
              "      <td>The scale of the COVID-19 public health emerge...</td>\n",
              "      <td>9.560090</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>COVID-19: the First Documented Coronavirus Pan...</td>\n",
              "      <td>The novel human coronavirus disease COVID-19 h...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32387617/;...</td>\n",
              "      <td>Liu, Yen-Chin; Kuo, Rei-Lin; Shih, Shin-Ru</td>\n",
              "      <td>Currently, people all over the world have been...</td>\n",
              "      <td>9.036929</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Invasion Science and the Global Spread of SARS...</td>\n",
              "      <td>Abstract Emerging infectious diseases like COV...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Nuñez, M. A.; Pauchard, A.; Ricciardi, A.</td>\n",
              "      <td>A sinister combination of ecosystem alteration...</td>\n",
              "      <td>8.917510</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>Horses as a Crucial Part of One Health</td>\n",
              "      <td>One Health (OH) is a crucial concept, where th...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32121327/;...</td>\n",
              "      <td>Lönker, Nelly Sophie; Fechner, Kim; Abd El Wah...</td>\n",
              "      <td>One Health (OH) is a holistic approach which d...</td>\n",
              "      <td>8.913372</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>Climate affects global patterns of COVID-19 ea...</td>\n",
              "      <td>Environmental factors, including seasonal clim...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.23.20040501</td>\n",
              "      <td>Ficetola, Gentile Francesco; Rubolini, Diego</td>\n",
              "      <td>1 Abstract 1 2 Environmental factors, includin...</td>\n",
              "      <td>8.739939</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>Synergistic China–US Ecological Research is Es...</td>\n",
              "      <td>The risk of a zoonotic pandemic disease threat...</td>\n",
              "      <td>https://doi.org/10.1007/s10393-020-01471-2; ht...</td>\n",
              "      <td>Smiley Evans, Tierra; Shi, Zhengli; Boots, Mic...</td>\n",
              "      <td>The catastrophic health and economic impacts o...</td>\n",
              "      <td>8.630575</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>Behavioral and Emotional Disorders in Children...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32248989/;...</td>\n",
              "      <td>Jiao, Wen Yan; Wang, Lin Na; Liu, Juan; Fang, ...</td>\n",
              "      <td>The COVID-19 outbreak first erupted in the cit...</td>\n",
              "      <td>8.540292</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>Racial/ethnic and socioeconomic disparities of...</td>\n",
              "      <td>We investigated the dependence of Covid-19 att...</td>\n",
              "      <td>https://arxiv.org/pdf/2004.12175v1.pdf</td>\n",
              "      <td>Dobin, Daniel; School, Alexander Dobin Candlew...</td>\n",
              "      <td>The Covid-19 disease first emerged in late 201...</td>\n",
              "      <td>8.373394</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>Behavioral Food Addiction During Lockdown: Tim...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://doi.org/10.1007/s11695-020-04649-3; ht...</td>\n",
              "      <td>Cherikh, Faredj; Frey, Sébastien; Bel, Corali;...</td>\n",
              "      <td>The world is currently facing the COVID-19 pan...</td>\n",
              "      <td>8.356851</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>Impacts of social and economic factors on the ...</td>\n",
              "      <td>This study models local and cross-city transmi...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32395017/;...</td>\n",
              "      <td>Qiu, Yun; Chen, Xi; Shi, Wei</td>\n",
              "      <td>The first pneumonia case of unknown cause was ...</td>\n",
              "      <td>8.263538</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>Spillover of SARS-CoV-2 into novel wild hosts ...</td>\n",
              "      <td>Abstract There is evidence that the current ou...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Franklin, Alan B.; Bevins, Sarah N.</td>\n",
              "      <td>J o u r n a l P r e -p r o o f 2 infected huma...</td>\n",
              "      <td>8.193293</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>County-level factors influence the trajectory ...</td>\n",
              "      <td>With new cases of Covid-19 surging in the Unit...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.0...</td>\n",
              "      <td>Kranjac, A. W.; Kranjac, D.</td>\n",
              "      <td>The presence of severe acute respiratory syndr...</td>\n",
              "      <td>8.164640</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>Spillovers and diversification potential of ba...</td>\n",
              "      <td>Abstract We examine the network spillovers, po...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Arreola Hernandez, Jose; Hoon Kang, Sang; Huss...</td>\n",
              "      <td>The banking sector plays an important role in ...</td>\n",
              "      <td>8.139073</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>Impacts of social and economic factors on the ...</td>\n",
              "      <td>This paper examines the role of various socioe...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.13.20035238</td>\n",
              "      <td>Qiu, Yun; Chen, Xi; Shi, Wei</td>\n",
              "      <td>Several clusters of patients with pneumonia of...</td>\n",
              "      <td>8.116073</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>Estimated inequities in COVID-19 infection fat...</td>\n",
              "      <td>There is limited evidence as to how COVID-19 i...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.2...</td>\n",
              "      <td>Steyn, Nicholas; Binny, Rachelle N; Hannah, Ka...</td>\n",
              "      <td>The COVID-19 outbreak originated in Wuhan Chin...</td>\n",
              "      <td>8.020653</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>Behavioral considerations and impact on person...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://doi.org/10.1016/j.jaad.2020.03.013; ht...</td>\n",
              "      <td>Kantor, Jonathan</td>\n",
              "      <td>In this issue of the JAAD, Lan et al1 provide ...</td>\n",
              "      <td>8.012635</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>Understanding the Socio-Economic Disruption in...</td>\n",
              "      <td>In this paper, we collect and study Twitter co...</td>\n",
              "      <td>https://arxiv.org/pdf/2004.05451v1.pdf</td>\n",
              "      <td>Shanthakumar, Swaroop Gowdra; Seetharam, Anand...</td>\n",
              "      <td>COVID-19 (also known as the novel coronavirus)...</td>\n",
              "      <td>8.004584</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>Race, Socioeconomic Deprivation, and Hospitali...</td>\n",
              "      <td>Preliminary reports suggest that the Coronavir...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.2...</td>\n",
              "      <td>Patel, A. P.; Paranjpe, M. D.; Kathiresan, N. ...</td>\n",
              "      <td>The Coronavirus Disease 2019 (COVID-19) pandem...</td>\n",
              "      <td>7.973779</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>The socio-economic determinants of the coronav...</td>\n",
              "      <td>The magnitude of the coronavirus disease (COVI...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.1...</td>\n",
              "      <td>Stojkoski, V.; Utkovski, Z.; Jolakoski, P.; Te...</td>\n",
              "      <td>The coronavirus pandemic began as a simple out...</td>\n",
              "      <td>7.949000</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>A multi-group SEIRA model for the spread of CO...</td>\n",
              "      <td>The outbreak and propagation of COVID-19 have ...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Contreras, Sebastián; Villavicencio, H. Andrés...</td>\n",
              "      <td>Since the outbreak of novel SARS-CoV2 in late ...</td>\n",
              "      <td>7.915906</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>Meta-regression of COVID-19 prevalence/fatalit...</td>\n",
              "      <td>To screen potential risk and protective socioe...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.2...</td>\n",
              "      <td>Takagi, H.; Kuno, T.; Yokoyama, Y.; Ueyama, H....</td>\n",
              "      <td>Reuters reported \"The coronavirus mortality ra...</td>\n",
              "      <td>7.791638</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>Social and behavioral health responses to COVI...</td>\n",
              "      <td>Our public health approaches to addressing COV...</td>\n",
              "      <td>https://doi.org/10.1007/s10865-020-00157-y; ht...</td>\n",
              "      <td>Eaton, Lisa A.; Kalichman, Seth C.</td>\n",
              "      <td>Sustained, individual-level, behavioral change...</td>\n",
              "      <td>7.759786</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>Investigating spatial variability in COVID-19 ...</td>\n",
              "      <td>Background: Spain has been disproportionately ...</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.14.20065524</td>\n",
              "      <td>Dahal, Sushma; Mizumoto, Kenji; Rothenberg, Ri...</td>\n",
              "      <td>The case fatality ratio is a useful metric to ...</td>\n",
              "      <td>7.747395</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>New Virus, New Challenge</td>\n",
              "      <td>The ongoing COVID-19 pandemic has made us to c...</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Shi, Yi</td>\n",
              "      <td>Yi Shi 1, * The ongoing COVID-19 pandemic has ...</td>\n",
              "      <td>7.638880</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>Globalized low-income countries may experience...</td>\n",
              "      <td>Understanding the factors underpinning COVID-1...</td>\n",
              "      <td>https://doi.org/10.1101/2020.03.31.20049122</td>\n",
              "      <td>Jaffe, Rodolfo; Ortiz Vera, Mabel Patricia; Ja...</td>\n",
              "      <td>Understanding the factors underpinning COVID-1...</td>\n",
              "      <td>7.513772</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>Why people failed to adhere to COVID-19 preven...</td>\n",
              "      <td>These violations might endanger the community ...</td>\n",
              "      <td>https://doi.org/10.1017/ice.2020.245; https://...</td>\n",
              "      <td>Chan, Derwin K. C.; Zhang, Chun-Qing; Weman-Jo...</td>\n",
              "      <td>The integrated model of self-determination the...</td>\n",
              "      <td>7.415810</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>Impact of COVID-19 on Indian Sugar Industry</td>\n",
              "      <td>The Indian sugar industry, a significant playe...</td>\n",
              "      <td>https://doi.org/10.1007/s12355-020-00846-7</td>\n",
              "      <td>Solomon, S.; Rao, Govind Pratap; Swapna, M.</td>\n",
              "      <td>Sugarcane is a long duration labour-intensive ...</td>\n",
              "      <td>7.404040</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>COVID-19 in patients with diabetes: risk facto...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://doi.org/10.1016/j.metabol.2020.154224;...</td>\n",
              "      <td>Klonoff, David C.; Umpierrez, Guillermo E.</td>\n",
              "      <td>We agree with the conclusion by Dr. Hill and c...</td>\n",
              "      <td>7.260640</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>Racial, Economic and Health Inequality and COV...</td>\n",
              "      <td>Abstract Background: There is preliminary evid...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.2...</td>\n",
              "      <td>Abedi, V.; Olulana, O.; Avula, V.; Chaudhary, ...</td>\n",
              "      <td>The complexity of managing patients with the C...</td>\n",
              "      <td>7.253174</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>34</th>\n",
              "      <td>Stroke in patients with SARS-CoV-2 infection: ...</td>\n",
              "      <td>BACKGROUND: Italy is one of the most affected ...</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32436105/;...</td>\n",
              "      <td>Morassi, Mauro; Bagatto, Daniele; Cobelli, Mil...</td>\n",
              "      <td>Italy was the first country in Europe to witne...</td>\n",
              "      <td>7.194565</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>Determining risk factors for mortality in live...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Webb, Gwilym J; Moon, Andrew M; Barnes, Eleano...</td>\n",
              "      <td>We read with great interest the Correspondence...</td>\n",
              "      <td>7.182662</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>Critical patients with coronavirus disease 201...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Li, Lingzhi; Zhang, Binghong; He, Bing; Gong, ...</td>\n",
              "      <td>Dear Editor,\\nWe read Yang's paper with great ...</td>\n",
              "      <td>7.182662</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>37</th>\n",
              "      <td>Causal Inference on Networks under Continuous ...</td>\n",
              "      <td>This paper presents a methodology to draw caus...</td>\n",
              "      <td>https://arxiv.org/pdf/2004.13459v1.pdf</td>\n",
              "      <td>Prete, Davide Del; Forastiere, Laura; Sciabola...</td>\n",
              "      <td>Policies often involve interventions that spil...</td>\n",
              "      <td>7.130043</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>Outcomes from COVID-19 across the range of fra...</td>\n",
              "      <td>Purpose Our aim was to quantify the mortality ...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.05.2...</td>\n",
              "      <td>Miles, A.; Webb, T. E.; Mcloughlin, B.; Mannan...</td>\n",
              "      <td>While it is clear that mortality from SARS-CoV...</td>\n",
              "      <td>7.123729</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>Risk factors for severe COVID-19 illness in he...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://doi.org/10.1017/ice.2020.178; https://...</td>\n",
              "      <td>Wander, Pandora L.; Orlov, Marika; Merel, Susa...</td>\n",
              "      <td>\\nTo the Editor—We were very interested to rea...</td>\n",
              "      <td>7.106342</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>40</th>\n",
              "      <td>Coronavirus Disease (COVID-19) in Italy: Analy...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://doi.org/10.3389/fmed.2020.00140; https...</td>\n",
              "      <td>Di Lorenzo, Giuseppe; Di Trolio, Rossella</td>\n",
              "      <td>The lethality rate is determined as follows: t...</td>\n",
              "      <td>7.031626</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>Prevalence, risk factors, and clinical correla...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32380272/;...</td>\n",
              "      <td>Li, Xingxing; Yu, Haihang; Bian, Guolin; Hu, Z...</td>\n",
              "      <td>Since December 2019, a large-scale outbreak of...</td>\n",
              "      <td>6.958465</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>42</th>\n",
              "      <td>Reply to “COVID-19 in persons with haematologi...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://doi.org/10.1038/s41375-020-0877-y; htt...</td>\n",
              "      <td>Fattizzo, Bruno; Giannotta, Juri Alessandro; S...</td>\n",
              "      <td>We read with interest the article by He et al....</td>\n",
              "      <td>6.958465</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>Early forecasting of the potential risk zones ...</td>\n",
              "      <td>Recently, the coronavirus disease 2019 (COVID-...</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Ren, Hongyan; Zhao, Lu; Zhang, An; Song, Liuyi...</td>\n",
              "      <td>Involving six continents, the outbreaks of the...</td>\n",
              "      <td>6.921999</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>Epidemiological and clinical characteristics o...</td>\n",
              "      <td>Background: The first case of COVID-19 was det...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.2...</td>\n",
              "      <td>Souza, William Marciel de; Buss, Lewis Fletche...</td>\n",
              "      <td>The first case of COVID-19 was detected in Bra...</td>\n",
              "      <td>6.887059</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>Symptom profiles and risk factors for hospital...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32437750/;...</td>\n",
              "      <td>Díaz, Luis Antonio; García-Salum, Tamara; Fuen...</td>\n",
              "      <td>San Carlos for their support to this project r...</td>\n",
              "      <td>6.886811</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>46</th>\n",
              "      <td>Initial Cluster of Novel Coronavirus (2019-nCo...</td>\n",
              "      <td>Reanalysis of the epidemic curve from the init...</td>\n",
              "      <td>https://doi.org/10.3390/jcm9020488; https://ww...</td>\n",
              "      <td>Nishiura, Hiroshi; Linton, Natalie M.; Akhmetz...</td>\n",
              "      <td>The clinical summary of the earliest cases of ...</td>\n",
              "      <td>6.875602</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>47</th>\n",
              "      <td>Applying the SEIR Model in Forecasting The COV...</td>\n",
              "      <td>On March 18, 2020 the Malaysian government imp...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.1...</td>\n",
              "      <td>Mahmud, A.; Lim, P. Y.</td>\n",
              "      <td>Universiti Putra Malaysia. CC-BY-NC-ND 4.0 Int...</td>\n",
              "      <td>6.856776</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>48</th>\n",
              "      <td>Modifiable and non-modifiable risk factors for...</td>\n",
              "      <td>Background Information on risk factors for COV...</td>\n",
              "      <td>http://medrxiv.org/cgi/content/short/2020.04.2...</td>\n",
              "      <td>Ho, F. K.; Celis-Morales, C. A.; Gray, S. R.; ...</td>\n",
              "      <td>COVID-19, caused by SARS-CoV-2 infection, incl...</td>\n",
              "      <td>6.852470</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>49</th>\n",
              "      <td>Letter to the Editor on ‘Risk Factors for Unpl...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>Kamath, Atul F.; Piuzzi, Nicolas S.</td>\n",
              "      <td>We read with great interest the article by Suk...</td>\n",
              "      <td>6.816618</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-b9840663-304d-40e8-bc8d-2922516bf951')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-b9840663-304d-40e8-bc8d-2922516bf951 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-b9840663-304d-40e8-bc8d-2922516bf951');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-1d9d411a-59c6-4c8b-996a-b38b88c06eb1\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-1d9d411a-59c6-4c8b-996a-b38b88c06eb1')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-1d9d411a-59c6-4c8b-996a-b38b88c06eb1 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "res",
              "summary": "{\n  \"name\": \"res\",\n  \"rows\": 50,\n  \"fields\": [\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"Behavioral Food Addiction During Lockdown: Time for Awareness, Time to Prepare the Aftermath\",\n          \"Risk factors for severe COVID-19 illness in healthcare workers: Too many unknowns\",\n          \"Why people failed to adhere to COVID-19 preventive behaviors? Perspectives from an integrated behavior change model\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 37,\n        \"samples\": [\n          \"In this paper, we collect and study Twitter communications to understand the socio-economic impact of COVID-19 in the United States during the early days of the pandemic. Our analysis reveals that COVID-19 gripped the nation during this time as is evidenced by the significant number of trending hashtags. With infections soaring rapidly, users took to Twitter asking people to self isolate and quarantine themselves. Users also demanded closure of schools, bars, and restaurants as well as lockdown of cities and states. The communications reveal the ensuing panic buying and the unavailability of some essential goods, in particular toilet paper. We also observe users express their frustration in their communications as the virus spread continued. We methodically collect a total of 530,206 tweets by identifying and tracking trending COVID-related hashtags. We then group the hashtags into six main categories, namely 1) General COVID, 2) Quarantine, 3) Panic Buying, 4) School Closures, 5) Lockdowns, and 6) Frustration and Hope, and study the temporal evolution of tweets in these hashtags. We conduct a linguistic analysis of words common to all the hashtag groups and specific to each hashtag group. Our preliminary study presents a succinct and aggregated picture of people's response to the pandemic and lays the groundwork for future fine-grained linguistic and behavioral analysis.\",\n          \"With new cases of Covid-19 surging in the United States, we need to better understand how the spread of novel coronavirus varies across all segments of the population. We use hierarchical exponential growth curve modeling techniques to examine whether community social and economic characteristics uniquely influence the incidence of Covid-19 cases in the urban built environment. We show that, as of May 3, 2020, confirmed coronavirus infections are concentrated along demographic and socioeconomic lines in New York City and surrounding areas, the epicenter of the Covid-19 pandemic in the United States. Furthermore, we see evidence that, after the onset of the pandemic, timely enactment of physical distancing measures such as school closures is imperative in order to limit the extent of the coronavirus spread in the population. Public health authorities must impose nonpharmaceutical measures early on in the pandemic and consider community-level factors that associate with a greater risk of viral transmission.\",\n          \"Objectives: United States government scientists estimate that COVID-19 may kill tens of thousands of Americans. Many of the pre-existing conditions that increase the risk of death in those with COVID-19 are the same diseases that are affected by long-term exposure to air pollution. We investigated whether long-term average exposure to fine particulate matter (PM2.5) is associated with an increased risk of COVID-19 death in the United States. Design: A nationwide, cross-sectional study using county-level data. Data sources: COVID-19 death counts were collected for more than 3,000 counties in the United States (representing 98% of the population) up to April 22, 2020 from Johns Hopkins University, Center for Systems Science and Engineering Coronavirus Resource Center. Main outcome measures: We fit negative binomial mixed models using county-level COVID-19 deaths as the outcome and county-level long-term average of PM2.5 as the exposure. In the main analysis, we adjusted by 20 potential confounding factors including population size, age distribution, population density, time since the beginning of the outbreak, time since state issuance of the stay-at-home order, hospital beds, number of individuals tested, weather, and socioeconomic and behavioral variables such as obesity and smoking. We included a random intercept by state to account for potential correlation in counties within the same state. We conducted more than 68 additional sensitivity analyses. Results: We found that an increase of only 1 \\u03bcg/m3 in PM2.5 is associated with an 8% increase in the COVID-19 death rate (95% confidence interval [CI]: 2%, 15%). The results were statistically significant and robust to secondary and sensitivity analyses. Conclusions: A small increase in long-term exposure to PM2.5 leads to a large increase in the COVID-19 death rate. Despite the inherent limitations of the ecological study design, our results underscore the importance of continuing to enforce existing air pollution regulations to protect human health both during and after the COVID-19 crisis. The data and code are publicly available so our analyses can be updated routinely.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"https://doi.org/10.1007/s11695-020-04649-3; https://www.ncbi.nlm.nih.gov/pubmed/32405909/\",\n          \"https://doi.org/10.1017/ice.2020.178; https://www.ncbi.nlm.nih.gov/pubmed/32336303/\",\n          \"https://doi.org/10.1017/ice.2020.245; https://www.ncbi.nlm.nih.gov/pubmed/32408917/\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 49,\n        \"samples\": [\n          \"Cherikh, Faredj; Frey, S\\u00e9bastien; Bel, Corali; Attanasi, Giuseppe; Alifano, Marco; Iannelli, Antonio\",\n          \"Nishiura, Hiroshi; Linton, Natalie M.; Akhmetzhanov, Andrei R.\",\n          \"Ho, F. K.; Celis-Morales, C. A.; Gray, S. R.; Katikireddi, S. V.; Niedzwiedz, C. L.; Hastie, C.; Lyall, D. M.; Ferguson, L. D.; Berry, C.; Mackay, D. F.; Gill, J. M. R.; Pell, J. P.; Sattar, N.; Welsh, P. I.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"text\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 50,\n        \"samples\": [\n          \"The world is currently facing the COVID-19 pandemic and countries are one-by-one increasing restrictive measures on their populations. To avoid rapid transmission of the virus, governmental authorities have had to impose nationwide lockdowns. Humans were asked to be patient and stay inside their home, imposing social distancing and limiting drastically any human contact. France was not an exception, and such constrain inflicted inhabitants to change their habits, without notice, which eventually altered their perception of health risks. It would not take long to foresee the consequences of prolonged physical inactivity, social isolation, behavioral addiction, impatience, and so on [1]. While as a bariatric multidisciplinary team, we could not pursue our surgical activity in the matter, we began to be aware of the fallouts, through our teleconsultation service. To illustrate the common awareness we shared, we conducted a global survey on the behavioral modification towards food consumption during the third week of confinement.\\nAn automated questionnaire including five simple and straight-forward questions was diffused through social media in France. The population of the study was unspecific. A thousand ninety-two individuals replied to the questionnaires in no more than 7 days. Results are available in Fig. 1. More than a quarter of the respondents have currently no physical activity, corresponding to another quarter feeling that they lost control of their usual alimentation habits. Additionally, stress, feeling of emptiness, and boredom are managed with food consumption in 37.4, 36.7, and 43% of individuals, respectively. Furthermore, and more importantly, the latter three emotional eating habits show a low though significant positive correlation with lower physical activity (lowest Spearman\\u2019s Rho = 0.10, p value < 0.001) and a high positive correlation with loss of control of usual alimentation (lowest Spearman\\u2019s Rho = 0.32, p value < 0.001).\\nAlthough such survey does not have the same merits of traditional scientific comparative studies, the results are striking as they were obtained only 3 weeks after the beginning of mass quarantine. From the viewpoint of the psychiatrist, it is clear that any addictions, from tobacco consumption to online shopping, would dramatically increase during such period. Quarantine and isolation lead to impatience and risk-seeking behavior that in turn have been recently found to increase the likelihood to become obese [2]. In line with those intuitions, our poll shows that a considerable part of the population finds comfort and consolation through food consumption. Doing so, we can anticipate that unhealthy dietary might settle, leading to substantial weight gain and worsening of eating disorders such as bulimia or bing-eating. Similarly, following a prolonged shelter-in-place, doctors might encounter an increase in the incidence of obesity-related comorbidities. Thus, a group of individuals require our attention now. If not, patient care might be more complex as eating disorders often come along with depression, anxiety, or even psychotic decompensation in the vulnerable ones.\\nTo provide sufficient health caregivers, authorities have logically mandated hospitals to cancel any surgical procedures, outside emergencies, and cancer surgeries that cannot be postponed. The exclusion of bariatric patients did not curtail their desire to be treated; instead, their concerns for bariatric surgery have grown consequently [3]. The waiting list has and will expand. In the meantime, multidisciplinary bariatric teams should not rest on their feet. Rather, the mobilization through alternative methods should intensify, with each speciality involved. Psychiatrists and nutritionists should follow up and carefully target sensitive patients, vulnerable to food misconduct. Endocrinologists and bariatric surgeons should work on temporary medical management and draw up action plans to efficiently provide care once routine establishes again.\\nLast but not least, it might be time for governmental prevention programs to boost their fight against obesity. Without mentioning once more the epidemic of obesity worldwide and its downfall, this disease finds itself to be a risk factor for other ones too often. Coronavirus infection did not differ: obese patients have been associated with a higher risk of morbidity and mortality. Several reasons for the increased lethality of COVID-19 infection have been advanced going from the uncontrolled inflammatory cascade leading to and/or aggravating acute respiratory failure, the overexpression of ACE2\\u2014at the heart of the debate on the COVID-19 pandemic [4]\\u2014 in adipose tissue, and the metabolic obesity-related comorbidities, namely type 2 diabetes and hypertension that have been shown to be associated with increased disease severity and mortality [5, 6]. However, deregulation of the renin/angiotensin system in obese patients warrants some consideration. Angiotensinogen is mostly produced by the liver in normal physiology, whereas in obese individuals, white adipose tissues take over a major part of its production, with a subsequent abnormal increase of local and circulating levels of the vasoconstrictor Angiotensin II [7]. In case of COVID-19 infection, the renin/angiotensin counterregulatory axis, also known as ACE2/Ang-(1\\u20137)/Mas, cannot counteract the deleterious effect of Angiotensin II on pulmonary and cardiac structures and functionalities, due to a loss of function of ACE2 by massive viral binding and internalization. In comparison to this physiopathology understood by doctors, global awareness of obesity and its risks starts to expand among the population.\\nIn conclusion, concerns for food disorder and weight gain are slowly gaining ground in our isolated populations. Awareness among multidisciplinary bariatric teams has never been more needed and effective health measures must be taken, to avoid uncontrolled obesity and obesity-related comorbidity growth, and to prepare the aftermath following the resumption of our routine. Once more, bariatric surgeons must be in the center of the battle and stand ready to offer the right care.\\n\",\n          \"\\nTo the Editor\\u2014We were very interested to read the recent letter by Zhou et al, \\u201cProtecting Chinese Healthcare Workers While Combating the 2019 Novel Coronavirus.\\u201d1 We agree that everything possible should be done to protect healthcare workers (HCWs) from developing COVID-19. We agree with the recommendations of Zhou et al including the importance of increasing production of personal protection equipment (PPE), training HCWs in proper use, and maintaining a high clinical suspicion for COVID-19 even in patients without respiratory symptoms. In addition, we call for more research into the risk factors leading to severe illness among HCWs, defined as COVID-19 requiring hospitalization or admission to the intensive care unit.2 Research in this area is sorely lacking, limiting implementation of evidence-based practices. Despite being younger and healthier than the general population with COVID-19, HCWs have similar rates of severe illness. In China, the proportion of HCWs with severe illness decreased from 45% in early January 2020 to 9% after February 1,3 likely reflecting more consistent adoption of appropriate infection-control practices, including the use of PPE. However, severe illness among HCWs continues to be reported, suggesting that the use of currently approved infection control processes do not entirely prevent severe COVID-19 among HCWs. To explore risk factors for severe COVID-19 in HCWs, we performed structured searches using a Twitter analytics tool (Tweet Archivist, Seattle, WA) to identify news stories reported before March 16, 2020, about HCWs with severe COVID-19\\u2013related illness (n = 6 cases) (Table 1). In most of these articles, neither high-risk host factors nor a clear high-inoculum exposure was evident, but in a few reports, potential exposure to inocula containing a high viral load was reported, including potential exposures to virus in stool.\\nIn 10% of cases, gastrointestinal symptoms precede fever or respiratory symptoms by 1\\u20132 days.4 Furthermore, 60% of samples from the toilet, sink and door handles of an individual with SARS-CoV-2 were positive for viral RNA, even though the individual reported respiratory symptoms but not diarrhea.5 For HCWs, contact with surfaces and/or patients with these symptoms could represent opportunities for high-inoculum exposure. During the SARS epidemic, SARS-CoV RNA was detected in stool in greater quantities than any other site,6 leading the World Health Organization to conclude that, \\u201cdiarrhoea could still remain important for infectivity, regardless of its cause.\\u201d Although SARS-CoV-2 RNA is readily found in stool,7 whether replicating virus is present is less clear. However, ACE2 receptors, which are used by the virus to infect cells, are present in the GI tract,7 making it plausible that the GI tract is an active site of viral replication. We therefore postulate that exposure to virus from high-viral load sites such as stool should be formally evaluated as an ongoing risk factor for severe COVID-19\\u2013related illness in HCWs. To facilitate research in this area and to ensure adequate power, we suggest that deidentified information about HCW cases be shared in national data repositories so that these and other risk factors can be assessed and the workforce can be adequately protected. In the meantime, institutions, if not already doing so, should screen for diarrheal symptoms.\\n\",\n          \"The integrated model of self-determination theory3 and the theory of planned behavior4 is a behavior change model that utilizes the concepts of 2 widely used psychological theories. The integrated model outlines the processes by which psychological need support, and motivations directly and indirectly link to the social cognition beliefs, intention, and behavior (Fig. 1).5 In the integrated model,5 when social environments are supportive to individuals\\u2019 basic psychological needs of autonomy, competence, and relatedness, individuals are more likely to endorse autonomous motivation (ie, acting for inherent interest, satisfaction, personal goals, and values) than controlled motivation (ie, acting due to external contingencies, internal pressure, or sense of ego). They are also more likely to have more favorable social cognition beliefs (ie, attitude, subjective norm, and perceived behavioral control) and intentions and to demonstrate behavioral adherence in health behaviors. The psychological pathways illustrated in the integrated model have been supported by evidence from various health contexts and cultures,5,6 including preventing H1N1 transmission during a pandemic.2 We believe that the integrated model can explain why some people have failed to adhere to the recommended behaviors for COVID-19 prevention.\\nA number of countries have set up legislation regarding social distance measures (eg, stay-home restriction), quarantine, and lockdown/travel ban. These legislative actions are classic examples of external factors that foster the development of controlled motivation. According to the integrated model,5 individuals who are driven by controlled motivation (ie, acting due to external contingencies, internal pressure, or sense of ego) may adhere to the advisory behavior as soon as the external factors (eg, contingencies of following COVID-19 preventive behaviors or not) are present, but they are more vulnerable to nonadherence in the long term than those who hold autonomous motivation (ie, acting for inherent interest, satisfaction, personal goals, and values) for the action. Individuals driven by controlled motivation alone might consider violations of health legislations when they perceive that the risks of getting caught or negative health consequences are low.7 At present, the enforcement of some new COVID-19 prevention legislations (eg, social distancing measures) could be extremely challenging when the surveillance involves a large geographical area or population. Governments or public health organizations should consider noncoercive strategies that are aligned with basic psychological needs to foster individuals\\u2019 autonomous motivation of COVID-19 prevention.\\nIn addition to law enforcement, other social situations and environmental factors are supportive or detrimental to the motivational and social cognition factors affecting COVID-19 prevention. In support to the psychological factors in the integrated model5, there are social situations or personal beliefs that facilitate autonomous motivation (eg, \\u201cpreventing COVID-19 is what I want to do because I am responsible for my own health\\u201d), attitude (eg, accessible online information about the values of COVID-19 prevention), subjective norms (eg, family or friends who are following the COVID-19 preventive strategies say I should do the same), and perceived behavioral control (eg, training resources that make it easier for me to correctly apply COVID-19 preventive behavior such as hand hygiene). In contrast, some social circumstances are detrimental to motivational and social cognition factors. For instance, advice on the necessity of wearing face masks in community settings has been inconsistent across different nations and health organizations,8 which might discourage individual autonomous motivations (eg, \\u201cDo I really want to prevent COVID-19 in this way?\\u201d) and attitudes (eg, \\u201cAre there any points to wearing a face mask for the prevention of COVID-19?\\u201d). The shortage of personal protective equipment (PPE)9 might impair an individual\\u2019s sense of competence and perceived behavioral control (eg, \\u201clack of PPE has made the prevention of COVID-19 challenging and uncontrollable\\u201d). Discrimination toward, alienation of, and labeling of individuals who wear face masks in public areas10 or social groups that encourage the ignorance of social distance measures1 might undermine an individual\\u2019s relatedness and subjective norms in the context of COVID-19 prevention. Governments and health organizations should be aware of these factors and should implement policies and social strategies that facilitate the motivational and social cognition factors affecting COVID-19 prevention.\\nIn conclusion, the integrated model of self-determination theory and the theory of planned behavior explains why some individuals fail to adhere to the preventive behaviors of COVID-19. We hope our discussion may raise the awareness of governing bodies and public health sectors regarding the importance of considering individuals\\u2019 motivation and social cognition beliefs when implementing COVID-19 preventivon measures in the community.\\n\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 1.2694738447336886,\n        \"min\": 6.816617612328175,\n        \"max\": 12.580976745113897,\n        \"num_unique_values\": 48,\n        \"samples\": [\n          7.747394938240275,\n          6.958465293743185,\n          7.759786047477437\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 101
        }
      ],
      "source": [
        "p = '''\n",
        "Socioeconomic and behavioral risk factors for this spill-over\n",
        "'''\n",
        "res = searchAndGetAnswer(p)\n",
        "res"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9Tc2PcUsYSsA"
      },
      "source": [
        "### Q6: *Sustainable risk reduction strategies*"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 102,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 710
        },
        "id": "do0svmydYSsE",
        "outputId": "dadd305f-cfb5-41b4-eca9-04d278e8a560"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "1/10\n",
            "2/10\n",
            "3/10\n",
            "4/10\n",
            "5/10\n",
            "6/10\n",
            "7/10\n",
            "8/10\n",
            "9/10\n",
            "10/10\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "                                               title  \\\n",
              "0    Indirect effects of COVID-19 on the environment   \n",
              "1  Post COVID 19 and food pathways to sustainable...   \n",
              "2  A new vehicle to accelerate the UN Sustainable...   \n",
              "3  A Brave New World: Lessons from the COVID-19 P...   \n",
              "4  Editorial: Impacts of COVID-19 on agricultural...   \n",
              "5  From crisis to utopia: crafting new public–pri...   \n",
              "6  COVID-19 lockdowns cause global air pollution ...   \n",
              "7  Further analysis of the impact of distancing u...   \n",
              "8  A cloth mask for under-resourced healthcare se...   \n",
              "9  Preparing your intensive care unit for the COV...   \n",
              "\n",
              "                                            abstract Answer     Score  \\\n",
              "0  Abstract This research aims to show the positi...  【  】   9.350980   \n",
              "1                                                NaN  【  】   9.074676   \n",
              "2                                                NaN  【  】   8.976158   \n",
              "3                                                NaN  【  】   8.693033   \n",
              "4                                                NaN  【  】   8.602586   \n",
              "5                                                NaN  【  】   8.514002   \n",
              "6  The lockdown response to COVID-19 has caused a...  【  】   8.450780   \n",
              "7  This paper questions various claims from the p...  【  】   8.254299   \n",
              "8  INTRODUCTION: COVID19 pandemic poses a global ...  【  】   8.101378   \n",
              "9  The coronavirus disease 2019 (COVID-19) has ra...  【  】   8.032679   \n",
              "\n",
              "                                                 url  \\\n",
              "0  https://api.elsevier.com/content/article/pii/S...   \n",
              "1  https://doi.org/10.1007/s10460-020-10051-7; ht...   \n",
              "2  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...   \n",
              "3  https://www.ncbi.nlm.nih.gov/pubmed/32313383/;...   \n",
              "4  https://www.sciencedirect.com/science/article/...   \n",
              "5  https://www.ncbi.nlm.nih.gov/pubmed/32395003/;...   \n",
              "6        https://doi.org/10.1101/2020.04.10.20060673   \n",
              "7        https://doi.org/10.1101/2020.04.14.20048025   \n",
              "8  https://www.ncbi.nlm.nih.gov/pubmed/32394153/;...   \n",
              "9  https://doi.org/10.1186/s13054-020-02916-4; ht...   \n",
              "\n",
              "                                             authors  \n",
              "0  Zambrano-Monserrate, Manuel A.; Ruano, María A...  \n",
              "1  Blay-Palmer, Alison; Carey, Rachel; Valette, E...  \n",
              "2  Sherr, Lorraine; Cluver, Lucie; Desmond, Chris...  \n",
              "3  Sarkis, Joseph; Cohen, Maurie J.; Dewick, Paul...  \n",
              "4  Stephens, Emma; Martin, Guillaume; van Wijk, M...  \n",
              "5                                     Caron, Patrick  \n",
              "6  Venter, Zander S; Aunan, Kristin; Chowdhury, S...  \n",
              "7                               Bernstein, Daniel J.  \n",
              "8  Sugrue, Michael; O’Keeffe, Derek; Sugrue, Ryan...  \n",
              "9  Goh, Ken Junyang; Wong, Jolin; Tien, Jong-Chie...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-0277190f-5e92-4524-a2d6-8ed09655e676\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>title</th>\n",
              "      <th>abstract</th>\n",
              "      <th>Answer</th>\n",
              "      <th>Score</th>\n",
              "      <th>url</th>\n",
              "      <th>authors</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>Indirect effects of COVID-19 on the environment</td>\n",
              "      <td>Abstract This research aims to show the positi...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>9.350980</td>\n",
              "      <td>https://api.elsevier.com/content/article/pii/S...</td>\n",
              "      <td>Zambrano-Monserrate, Manuel A.; Ruano, María A...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>Post COVID 19 and food pathways to sustainable...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>【  】</td>\n",
              "      <td>9.074676</td>\n",
              "      <td>https://doi.org/10.1007/s10460-020-10051-7; ht...</td>\n",
              "      <td>Blay-Palmer, Alison; Carey, Rachel; Valette, E...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>A new vehicle to accelerate the UN Sustainable...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.976158</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7...</td>\n",
              "      <td>Sherr, Lorraine; Cluver, Lucie; Desmond, Chris...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>A Brave New World: Lessons from the COVID-19 P...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.693033</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32313383/;...</td>\n",
              "      <td>Sarkis, Joseph; Cohen, Maurie J.; Dewick, Paul...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>Editorial: Impacts of COVID-19 on agricultural...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.602586</td>\n",
              "      <td>https://www.sciencedirect.com/science/article/...</td>\n",
              "      <td>Stephens, Emma; Martin, Guillaume; van Wijk, M...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>From crisis to utopia: crafting new public–pri...</td>\n",
              "      <td>NaN</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.514002</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32395003/;...</td>\n",
              "      <td>Caron, Patrick</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>COVID-19 lockdowns cause global air pollution ...</td>\n",
              "      <td>The lockdown response to COVID-19 has caused a...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.450780</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.10.20060673</td>\n",
              "      <td>Venter, Zander S; Aunan, Kristin; Chowdhury, S...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>Further analysis of the impact of distancing u...</td>\n",
              "      <td>This paper questions various claims from the p...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.254299</td>\n",
              "      <td>https://doi.org/10.1101/2020.04.14.20048025</td>\n",
              "      <td>Bernstein, Daniel J.</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>A cloth mask for under-resourced healthcare se...</td>\n",
              "      <td>INTRODUCTION: COVID19 pandemic poses a global ...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.101378</td>\n",
              "      <td>https://www.ncbi.nlm.nih.gov/pubmed/32394153/;...</td>\n",
              "      <td>Sugrue, Michael; O’Keeffe, Derek; Sugrue, Ryan...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>Preparing your intensive care unit for the COV...</td>\n",
              "      <td>The coronavirus disease 2019 (COVID-19) has ra...</td>\n",
              "      <td>【  】</td>\n",
              "      <td>8.032679</td>\n",
              "      <td>https://doi.org/10.1186/s13054-020-02916-4; ht...</td>\n",
              "      <td>Goh, Ken Junyang; Wong, Jolin; Tien, Jong-Chie...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-0277190f-5e92-4524-a2d6-8ed09655e676')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-0277190f-5e92-4524-a2d6-8ed09655e676 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-0277190f-5e92-4524-a2d6-8ed09655e676');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-784ca8e3-b8db-4d9e-8db8-40fc88e04d60\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-784ca8e3-b8db-4d9e-8db8-40fc88e04d60')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-784ca8e3-b8db-4d9e-8db8-40fc88e04d60 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "res",
              "summary": "{\n  \"name\": \"res\",\n  \"rows\": 10,\n  \"fields\": [\n    {\n      \"column\": \"title\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"A cloth mask for under-resourced healthcare settings in the COVID19 pandemic\",\n          \"Post COVID 19 and food pathways to sustainable transformation\",\n          \"From crisis to utopia: crafting new public\\u2013private articulation at territorial level to design sustainable food systems\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"abstract\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 5,\n        \"samples\": [\n          \"The lockdown response to COVID-19 has caused an unprecedented reduction in global economic activity. We test the hypothesis that this has reduced tropospheric and ground-level air pollution concentrations using satellite data and a network of >10,000 air quality stations. After accounting for the effects of meteorological variability, we find remarkable declines in ground-level nitrogen dioxide (NO2: -29 % with 95% confidence interval -44% to -13%), ozone (O3: -11%; -20% to -2%) and fine particulate matter (PM2.5: -9%; -28% to 10%) during the first two weeks of lockdown (n = 27 countries). These results are largely mirrored by satellite measures of the troposphere although long-distance transport of PM2.5 resulted in more heterogeneous changes relative to NO2. Pollutant anomalies were related to short-term health outcomes using empirical exposure-response functions. We estimate that there was a net total of 7400 (340 to 14600) premature deaths and 6600 (4900 to 7900) pediatric asthma cases avoided during two weeks post-lockdown. In China and India alone, the PM2.5-related avoided premature mortality was 1400 (1100 to 1700) and 5300 (1000 to 11700), respectively. Assuming that the lockdown-induced deviations in pollutant concentrations are maintained for the duration of 2020, we estimate 0.78 (0.09 to 1.5) million premature deaths and 1.6 (0.8 to 2) million pediatric asthma cases could be avoided globally. While the state of global lockdown is not sustainable, these findings illustrate the potential health benefits gained from reducing 'business as usual' air pollutant emissions from economic activities.\",\n          \"The coronavirus disease 2019 (COVID-19) has rapidly evolved into a worldwide pandemic. Preparing intensive care units (ICU) is an integral part of any pandemic response. In this review, we discuss the key principles and strategies for ICU preparedness. We also describe our initial outbreak measures and share some of the challenges faced. To achieve sustainable ICU services, we propose the need to 1) prepare and implement rapid identification and isolation protocols, and a surge in ICU bed capacity; (2) provide a sustainable workforce with a focus on infection control; (3) ensure adequate supplies to equip ICUs and protect healthcare workers; and (4) maintain quality clinical management, as well as effective communication.\",\n          \"This paper questions various claims from the paper \\\"Social distancing strategies for curbing the COVID-19 epidemic\\\" by Kissler, Tedijanto, Lipsitch, and Grad: most importantly, the claim that China's \\\"intense\\\" distancing measures achieved only a 60% reduction in R0.\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Answer\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 1,\n        \"samples\": [\n          \"\\u3010  \\u3011 \"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Score\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 0.4293196249071596,\n        \"min\": 8.032678597628074,\n        \"max\": 9.350980246007547,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          8.10137838410534\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"url\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"https://www.ncbi.nlm.nih.gov/pubmed/32394153/; https://doi.org/10.1007/s11845-020-02241-3\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"authors\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 10,\n        \"samples\": [\n          \"Sugrue, Michael; O\\u2019Keeffe, Derek; Sugrue, Ryan; MacLean, Lorraine; Varzgalis, Manvydas\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 102
        }
      ],
      "source": [
        "p = '''\n",
        "Sustainable risk reduction strategies\n",
        "'''\n",
        "res = searchAndGetAnswer(p)\n",
        "res"
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "8cCYN908NK56"
      },
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.12.3"
    },
    "colab": {
      "provenance": []
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "3c6fbd30273d4553ae106af589a3419d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_60564d4481b94a05bf47aba45ca668c7",
              "IPY_MODEL_a3dc95c83f764fec90f65384d08b75df",
              "IPY_MODEL_90049ac4ba384a3ea039ae1ab4bbee80"
            ],
            "layout": "IPY_MODEL_8138efe3620048ff94265c671b7d2b1d"
          }
        },
        "60564d4481b94a05bf47aba45ca668c7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_3a25dc5c597f4879ab09341ce335975d",
            "placeholder": "​",
            "style": "IPY_MODEL_008f8b0703c74a668a679319b798e2a5",
            "value": "tokenizer_config.json: 100%"
          }
        },
        "a3dc95c83f764fec90f65384d08b75df": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_df687ca3da864c64862111254cedd54c",
            "max": 48,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_8b6a0b1925b74f878f08c5fd2830619d",
            "value": 48
          }
        },
        "90049ac4ba384a3ea039ae1ab4bbee80": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_0c43033fcabf4d36b7d7fe6a8abcd86e",
            "placeholder": "​",
            "style": "IPY_MODEL_304a4909b3f844c2b41812257347ddb7",
            "value": " 48.0/48.0 [00:00&lt;00:00, 3.70kB/s]"
          }
        },
        "8138efe3620048ff94265c671b7d2b1d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "3a25dc5c597f4879ab09341ce335975d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "008f8b0703c74a668a679319b798e2a5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "df687ca3da864c64862111254cedd54c": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8b6a0b1925b74f878f08c5fd2830619d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "0c43033fcabf4d36b7d7fe6a8abcd86e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "304a4909b3f844c2b41812257347ddb7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "1ccdf56406194717bde2687919967c43": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_4e4a163366834a3787d9018c76540473",
              "IPY_MODEL_2d7cac9eff8b463c8c6a6dc40d1ae84c",
              "IPY_MODEL_4c277cf53ac640b2a6c4d87582bdddc8"
            ],
            "layout": "IPY_MODEL_fb54ab2fadfb41158a7c2eb99a0de4de"
          }
        },
        "4e4a163366834a3787d9018c76540473": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_659d23c615ab4edba8222f4da2a7c0ed",
            "placeholder": "​",
            "style": "IPY_MODEL_8217318c1ded4f1daf137b6d1c17da96",
            "value": "vocab.txt: 100%"
          }
        },
        "2d7cac9eff8b463c8c6a6dc40d1ae84c": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_1a2264de8a404bbf9cff8b9842ef647a",
            "max": 231508,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_da354a16760b49e385749cf2e3f402a4",
            "value": 231508
          }
        },
        "4c277cf53ac640b2a6c4d87582bdddc8": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_34bdb07c091f416ba769884597daf78d",
            "placeholder": "​",
            "style": "IPY_MODEL_75c7a41b98084c1cba7f4a309ecc6622",
            "value": " 232k/232k [00:00&lt;00:00, 5.72MB/s]"
          }
        },
        "fb54ab2fadfb41158a7c2eb99a0de4de": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "659d23c615ab4edba8222f4da2a7c0ed": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8217318c1ded4f1daf137b6d1c17da96": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "1a2264de8a404bbf9cff8b9842ef647a": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "da354a16760b49e385749cf2e3f402a4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "34bdb07c091f416ba769884597daf78d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "75c7a41b98084c1cba7f4a309ecc6622": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "c337bb8619964924835c99744efa3240": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_368f8f920e944852be18cada1bfd549e",
              "IPY_MODEL_f33f65c2f84647d994427a86cbc8b00e",
              "IPY_MODEL_770363869a4a43e899c8ae628102e031"
            ],
            "layout": "IPY_MODEL_04f48f7b7b754d17a8946745c672338e"
          }
        },
        "368f8f920e944852be18cada1bfd549e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_3e57fe4104e44deb84cb7c1de76ec212",
            "placeholder": "​",
            "style": "IPY_MODEL_b9919b88af1545278e13b05914542227",
            "value": "tokenizer.json: 100%"
          }
        },
        "f33f65c2f84647d994427a86cbc8b00e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_abacb2cfc4c744b7a170af68278a4e19",
            "max": 466062,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_fb0e0740e00c4c27bdca2c6cbeafee07",
            "value": 466062
          }
        },
        "770363869a4a43e899c8ae628102e031": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_bf266f0cda1c4345be7b0a4c5757c35b",
            "placeholder": "​",
            "style": "IPY_MODEL_dd30000507bd4b918a14822f36e95be6",
            "value": " 466k/466k [00:00&lt;00:00, 1.69MB/s]"
          }
        },
        "04f48f7b7b754d17a8946745c672338e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "3e57fe4104e44deb84cb7c1de76ec212": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "b9919b88af1545278e13b05914542227": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "abacb2cfc4c744b7a170af68278a4e19": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "fb0e0740e00c4c27bdca2c6cbeafee07": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "bf266f0cda1c4345be7b0a4c5757c35b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "dd30000507bd4b918a14822f36e95be6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "e5c0b47ee8164423b8885addab72fed9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_b52e7f99bd0c4b2785c8bf8026dd7c9d",
              "IPY_MODEL_138da3d243bb46d8b4353213384e80d5",
              "IPY_MODEL_a99e9ce48fe642aabd9682a5227b3e8f"
            ],
            "layout": "IPY_MODEL_12cee6cefa954965bb01e3961996ffd4"
          }
        },
        "b52e7f99bd0c4b2785c8bf8026dd7c9d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_81988520ee4b43e69d566deaaf70d9ce",
            "placeholder": "​",
            "style": "IPY_MODEL_934efc5c653c4b5f97db0ac87389a87f",
            "value": "config.json: 100%"
          }
        },
        "138da3d243bb46d8b4353213384e80d5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_1211c83210b0494d857777b9b33d6e26",
            "max": 570,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_24e3ad870e18400ca3111b3e304c81e4",
            "value": 570
          }
        },
        "a99e9ce48fe642aabd9682a5227b3e8f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8e0bbbc666db4fd18ff968b66f8583de",
            "placeholder": "​",
            "style": "IPY_MODEL_a9481c7ba0244215a9ac7aec04560519",
            "value": " 570/570 [00:00&lt;00:00, 33.6kB/s]"
          }
        },
        "12cee6cefa954965bb01e3961996ffd4": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "81988520ee4b43e69d566deaaf70d9ce": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "934efc5c653c4b5f97db0ac87389a87f": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "1211c83210b0494d857777b9b33d6e26": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "24e3ad870e18400ca3111b3e304c81e4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "8e0bbbc666db4fd18ff968b66f8583de": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a9481c7ba0244215a9ac7aec04560519": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "9527c1603ad34fb392a9dfca276b1607": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_9fad40517da54635a78242d92e483bbc",
              "IPY_MODEL_cfbf0c8439e64e82a9e4356aa7c48d5d",
              "IPY_MODEL_79b57a556f3a4d6095be76c61afbbb59"
            ],
            "layout": "IPY_MODEL_69bbf095ac64458aa968ea200e30cda8"
          }
        },
        "9fad40517da54635a78242d92e483bbc": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_7a00990118494b50970505d484d25fb6",
            "placeholder": "​",
            "style": "IPY_MODEL_a7f13ffcafb0493eb7614ef218b388e6",
            "value": "model.safetensors: 100%"
          }
        },
        "cfbf0c8439e64e82a9e4356aa7c48d5d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_f9895f853d0449b6af4ec2d7358ba22e",
            "max": 440449768,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_adc6500a77904717852cd78d7d64cb21",
            "value": 440449768
          }
        },
        "79b57a556f3a4d6095be76c61afbbb59": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_aadb941603f1422ab485a5287ea1aa7a",
            "placeholder": "​",
            "style": "IPY_MODEL_d4a03a9ba3aa4d33bad070d014e572b7",
            "value": " 440M/440M [00:04&lt;00:00, 106MB/s]"
          }
        },
        "69bbf095ac64458aa968ea200e30cda8": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "7a00990118494b50970505d484d25fb6": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a7f13ffcafb0493eb7614ef218b388e6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "f9895f853d0449b6af4ec2d7358ba22e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "adc6500a77904717852cd78d7d64cb21": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "aadb941603f1422ab485a5287ea1aa7a": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d4a03a9ba3aa4d33bad070d014e572b7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        }
      }
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}